// transferzero-sdk/dist/Api/PaymentMethodsApi.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports["default"] = void 0;













var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _PaymentMethodListResponse = _interopRequireDefault(require("../Model/PaymentMethodListResponse"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };}function _typeof(o) {"@babel/helpers - typeof";return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) {return typeof o;} : function (o) {return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o;}, _typeof(o);}function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError("Cannot call a class as a function");}}function _defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ("value" in descriptor) descriptor.writable = true;Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);}}function _createClass(Constructor, protoProps, staticProps) {if (protoProps) _defineProperties(Constructor.prototype, protoProps);if (staticProps) _defineProperties(Constructor, staticProps);Object.defineProperty(Constructor, "prototype", { writable: false });return Constructor;}function _toPropertyKey(t) {var i = _toPrimitive(t, "string");return "symbol" == _typeof(i) ? i : i + "";}function _toPrimitive(t, r) {if ("object" != _typeof(t) || !t) return t;var e = t[Symbol.toPrimitive];if (void 0 !== e) {var i = e.call(t, r || "default");if ("object" != _typeof(i)) return i;throw new TypeError("@@toPrimitive must return a primitive value.");}return ("string" === r ? String : Number)(t);} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
* PaymentMethods service.
* @module Api/PaymentMethodsApi
* @version 1.35.1
*/var PaymentMethodsApi = exports["default"] = /*#__PURE__*/function () {/**
  * Constructs a new PaymentMethodsApi. 
  * @alias module:Api/PaymentMethodsApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */function PaymentMethodsApi(apiClient) {_classCallCheck(this, PaymentMethodsApi);this.apiClient = apiClient || _ApiClient["default"].instance;}
  /**
   * This method returns possible payin methods.
   * Fetching possible payin methods. 
   * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:Model/PaymentMethodListResponse} and HTTP response
   */return _createClass(PaymentMethodsApi, [{ key: "paymentMethodsInWithHttpInfo", value:
    function paymentMethodsInWithHttpInfo() {
      var postBody = null;

      var pathParams = {
      };
      var queryParams = {
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = ['AuthorizationKey', 'AuthorizationNonce', 'AuthorizationSecret', 'AuthorizationSignature'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _PaymentMethodListResponse["default"];
      return this.apiClient.callApi(
        '/info/payment_methods/in', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null
      );
    }

    /**
     * This method returns possible payin methods.
     * Fetching possible payin methods. 
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:Model/PaymentMethodListResponse}
     */ }, { key: "paymentMethodsIn", value:
    function paymentMethodsIn() {
      return this.paymentMethodsInWithHttpInfo().
      then(function (response_and_data) {
        return response_and_data.data;
      });
    }


    /**
     * This method returns possible payout methods.
     * Fetching possible payout methods. 
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:Model/PaymentMethodListResponse} and HTTP response
     */ }, { key: "paymentMethodsOutWithHttpInfo", value:
    function paymentMethodsOutWithHttpInfo() {
      var postBody = null;

      var pathParams = {
      };
      var queryParams = {
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = ['AuthorizationKey', 'AuthorizationNonce', 'AuthorizationSecret', 'AuthorizationSignature'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _PaymentMethodListResponse["default"];
      return this.apiClient.callApi(
        '/info/payment_methods/out', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null
      );
    }

    /**
     * This method returns possible payout methods.
     * Fetching possible payout methods. 
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:Model/PaymentMethodListResponse}
     */ }, { key: "paymentMethodsOut", value:
    function paymentMethodsOut() {
      return this.paymentMethodsOutWithHttpInfo().
      then(function (response_and_data) {
        return response_and_data.data;
      });
    } }]);}();
