// transferzero-sdk/dist/index.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });Object.defineProperty(exports, "APILogsApi", { enumerable: true, get: function get() {return _APILogsApi["default"];} });Object.defineProperty(exports, "Account", { enumerable: true, get: function get() {return _Account["default"];} });Object.defineProperty(exports, "AccountDebitsApi", { enumerable: true, get: function get() {return _AccountDebitsApi["default"];} });Object.defineProperty(exports, "AccountListResponse", { enumerable: true, get: function get() {return _AccountListResponse["default"];} });Object.defineProperty(exports, "AccountMeta", { enumerable: true, get: function get() {return _AccountMeta["default"];} });Object.defineProperty(exports, "AccountResponse", { enumerable: true, get: function get() {return _AccountResponse["default"];} });Object.defineProperty(exports, "AccountValidationApi", { enumerable: true, get: function get() {return _AccountValidationApi["default"];} });Object.defineProperty(exports, "AccountValidationError", { enumerable: true, get: function get() {return _AccountValidationError["default"];} });Object.defineProperty(exports, "AccountValidationRequest", { enumerable: true, get: function get() {return _AccountValidationRequest["default"];} });Object.defineProperty(exports, "AccountValidationResponse", { enumerable: true, get: function get() {return _AccountValidationResponse["default"];} });Object.defineProperty(exports, "AccountValidationResult", { enumerable: true, get: function get() {return _AccountValidationResult["default"];} });Object.defineProperty(exports, "AccountsApi", { enumerable: true, get: function get() {return _AccountsApi["default"];} });Object.defineProperty(exports, "ApiClient", { enumerable: true, get: function get() {return _ApiClient["default"];} });Object.defineProperty(exports, "ApiLog", { enumerable: true, get: function get() {return _ApiLog["default"];} });Object.defineProperty(exports, "ApiLogListResponse", { enumerable: true, get: function get() {return _ApiLogListResponse["default"];} });Object.defineProperty(exports, "ApiLogResponse", { enumerable: true, get: function get() {return _ApiLogResponse["default"];} });Object.defineProperty(exports, "Currency", { enumerable: true, get: function get() {return _Currency["default"];} });Object.defineProperty(exports, "CurrencyExchange", { enumerable: true, get: function get() {return _CurrencyExchange["default"];} });Object.defineProperty(exports, "CurrencyExchangeAllOf", { enumerable: true, get: function get() {return _CurrencyExchangeAllOf["default"];} });Object.defineProperty(exports, "CurrencyExchangeListResponse", { enumerable: true, get: function get() {return _CurrencyExchangeListResponse["default"];} });Object.defineProperty(exports, "CurrencyInfoApi", { enumerable: true, get: function get() {return _CurrencyInfoApi["default"];} });Object.defineProperty(exports, "CurrencyListResponse", { enumerable: true, get: function get() {return _CurrencyListResponse["default"];} });Object.defineProperty(exports, "CurrencyOpposite", { enumerable: true, get: function get() {return _CurrencyOpposite["default"];} });Object.defineProperty(exports, "CurrencyOppositeAllOf", { enumerable: true, get: function get() {return _CurrencyOppositeAllOf["default"];} });Object.defineProperty(exports, "Debit", { enumerable: true, get: function get() {return _Debit["default"];} });Object.defineProperty(exports, "DebitListRequest", { enumerable: true, get: function get() {return _DebitListRequest["default"];} });Object.defineProperty(exports, "DebitListResponse", { enumerable: true, get: function get() {return _DebitListResponse["default"];} });Object.defineProperty(exports, "DebitRequest", { enumerable: true, get: function get() {return _DebitRequest["default"];} });Object.defineProperty(exports, "DebitRequestWrapper", { enumerable: true, get: function get() {return _DebitRequestWrapper["default"];} });Object.defineProperty(exports, "Document", { enumerable: true, get: function get() {return _Document["default"];} });Object.defineProperty(exports, "DocumentListResponse", { enumerable: true, get: function get() {return _DocumentListResponse["default"];} });Object.defineProperty(exports, "DocumentRequest", { enumerable: true, get: function get() {return _DocumentRequest["default"];} });Object.defineProperty(exports, "DocumentResponse", { enumerable: true, get: function get() {return _DocumentResponse["default"];} });Object.defineProperty(exports, "DocumentWebhook", { enumerable: true, get: function get() {return _DocumentWebhook["default"];} });Object.defineProperty(exports, "DocumentWebhookAllOf", { enumerable: true, get: function get() {return _DocumentWebhookAllOf["default"];} });Object.defineProperty(exports, "DocumentsApi", { enumerable: true, get: function get() {return _DocumentsApi["default"];} });Object.defineProperty(exports, "ErrorStatus", { enumerable: true, get: function get() {return _ErrorStatus["default"];} });Object.defineProperty(exports, "FieldDescription", { enumerable: true, get: function get() {return _FieldDescription["default"];} });Object.defineProperty(exports, "FieldSelectValidation", { enumerable: true, get: function get() {return _FieldSelectValidation["default"];} });Object.defineProperty(exports, "FieldValidation", { enumerable: true, get: function get() {return _FieldValidation["default"];} });Object.defineProperty(exports, "LogsApi", { enumerable: true, get: function get() {return _LogsApi["default"];} });Object.defineProperty(exports, "Pagination", { enumerable: true, get: function get() {return _Pagination["default"];} });Object.defineProperty(exports, "PaginationMeta", { enumerable: true, get: function get() {return _PaginationMeta["default"];} });Object.defineProperty(exports, "PayinMethod", { enumerable: true, get: function get() {return _PayinMethod["default"];} });Object.defineProperty(exports, "PayinMethodDetails", { enumerable: true, get: function get() {return _PayinMethodDetails["default"];} });Object.defineProperty(exports, "PayinMethodDetailsBTC", { enumerable: true, get: function get() {return _PayinMethodDetailsBTC["default"];} });Object.defineProperty(exports, "PayinMethodDetailsMobile", { enumerable: true, get: function get() {return _PayinMethodDetailsMobile["default"];} });Object.defineProperty(exports, "PayinMethodDetailsNGNBank", { enumerable: true, get: function get() {return _PayinMethodDetailsNGNBank["default"];} });Object.defineProperty(exports, "PayinMethodRequest", { enumerable: true, get: function get() {return _PayinMethodRequest["default"];} });Object.defineProperty(exports, "PayinMethodResponse", { enumerable: true, get: function get() {return _PayinMethodResponse["default"];} });Object.defineProperty(exports, "PayinMethodState", { enumerable: true, get: function get() {return _PayinMethodState["default"];} });Object.defineProperty(exports, "PayinMethodUxFlow", { enumerable: true, get: function get() {return _PayinMethodUxFlow["default"];} });Object.defineProperty(exports, "PayinMethodsApi", { enumerable: true, get: function get() {return _PayinMethodsApi["default"];} });Object.defineProperty(exports, "PaymentMethod", { enumerable: true, get: function get() {return _PaymentMethod["default"];} });Object.defineProperty(exports, "PaymentMethodListResponse", { enumerable: true, get: function get() {return _PaymentMethodListResponse["default"];} });Object.defineProperty(exports, "PaymentMethodOpposite", { enumerable: true, get: function get() {return _PaymentMethodOpposite["default"];} });Object.defineProperty(exports, "PaymentMethodOppositeAllOf", { enumerable: true, get: function get() {return _PaymentMethodOppositeAllOf["default"];} });Object.defineProperty(exports, "PaymentMethodsApi", { enumerable: true, get: function get() {return _PaymentMethodsApi["default"];} });Object.defineProperty(exports, "PayoutMethod", { enumerable: true, get: function get() {return _PayoutMethod["default"];} });Object.defineProperty(exports, "PayoutMethodBankAccountTypeEnum", { enumerable: true, get: function get() {return _PayoutMethodBankAccountTypeEnum["default"];} });Object.defineProperty(exports, "PayoutMethodCashProviderEnum", { enumerable: true, get: function get() {return _PayoutMethodCashProviderEnum["default"];} });Object.defineProperty(exports, "PayoutMethodCountryEnum", { enumerable: true, get: function get() {return _PayoutMethodCountryEnum["default"];} });Object.defineProperty(exports, "PayoutMethodDetails", { enumerable: true, get: function get() {return _PayoutMethodDetails["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsBRLBank", { enumerable: true, get: function get() {return _PayoutMethodDetailsBRLBank["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsBTC", { enumerable: true, get: function get() {return _PayoutMethodDetailsBTC["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsBWPBank", { enumerable: true, get: function get() {return _PayoutMethodDetailsBWPBank["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsBalance", { enumerable: true, get: function get() {return _PayoutMethodDetailsBalance["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsCADBank", { enumerable: true, get: function get() {return _PayoutMethodDetailsCADBank["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsEGPBank", { enumerable: true, get: function get() {return _PayoutMethodDetailsEGPBank["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsGBPBank", { enumerable: true, get: function get() {return _PayoutMethodDetailsGBPBank["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsGHSBank", { enumerable: true, get: function get() {return _PayoutMethodDetailsGHSBank["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsGHSCash", { enumerable: true, get: function get() {return _PayoutMethodDetailsGHSCash["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsGNFMobile", { enumerable: true, get: function get() {return _PayoutMethodDetailsGNFMobile["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsIBAN", { enumerable: true, get: function get() {return _PayoutMethodDetailsIBAN["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsINRBank", { enumerable: true, get: function get() {return _PayoutMethodDetailsINRBank["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsKESBank", { enumerable: true, get: function get() {return _PayoutMethodDetailsKESBank["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsKESMobile", { enumerable: true, get: function get() {return _PayoutMethodDetailsKESMobile["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsMADCash", { enumerable: true, get: function get() {return _PayoutMethodDetailsMADCash["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsMobile", { enumerable: true, get: function get() {return _PayoutMethodDetailsMobile["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsNGNBank", { enumerable: true, get: function get() {return _PayoutMethodDetailsNGNBank["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsNZDBank", { enumerable: true, get: function get() {return _PayoutMethodDetailsNZDBank["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsUGXBank", { enumerable: true, get: function get() {return _PayoutMethodDetailsUGXBank["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsUSDBank", { enumerable: true, get: function get() {return _PayoutMethodDetailsUSDBank["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsUSDCash", { enumerable: true, get: function get() {return _PayoutMethodDetailsUSDCash["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsXAFBank", { enumerable: true, get: function get() {return _PayoutMethodDetailsXAFBank["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsXAFMobile", { enumerable: true, get: function get() {return _PayoutMethodDetailsXAFMobile["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsXOFBank", { enumerable: true, get: function get() {return _PayoutMethodDetailsXOFBank["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsXOFCash", { enumerable: true, get: function get() {return _PayoutMethodDetailsXOFCash["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsXOFMobile", { enumerable: true, get: function get() {return _PayoutMethodDetailsXOFMobile["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsZARBank", { enumerable: true, get: function get() {return _PayoutMethodDetailsZARBank["default"];} });Object.defineProperty(exports, "PayoutMethodDetailsZMWBank", { enumerable: true, get: function get() {return _PayoutMethodDetailsZMWBank["default"];} });Object.defineProperty(exports, "PayoutMethodGenderEnum", { enumerable: true, get: function get() {return _PayoutMethodGenderEnum["default"];} });Object.defineProperty(exports, "PayoutMethodIdentityCardTypeEnum", { enumerable: true, get: function get() {return _PayoutMethodIdentityCardTypeEnum["default"];} });Object.defineProperty(exports, "PayoutMethodLegalEntityTypeEnum", { enumerable: true, get: function get() {return _PayoutMethodLegalEntityTypeEnum["default"];} });Object.defineProperty(exports, "PayoutMethodListResponse", { enumerable: true, get: function get() {return _PayoutMethodListResponse["default"];} });Object.defineProperty(exports, "PayoutMethodMobileProviderEnum", { enumerable: true, get: function get() {return _PayoutMethodMobileProviderEnum["default"];} });Object.defineProperty(exports, "PayoutMethodNatureOfBusinessEnum", { enumerable: true, get: function get() {return _PayoutMethodNatureOfBusinessEnum["default"];} });Object.defineProperty(exports, "PayoutMethodPixKeyTypeEnum", { enumerable: true, get: function get() {return _PayoutMethodPixKeyTypeEnum["default"];} });Object.defineProperty(exports, "PayoutMethodRequest", { enumerable: true, get: function get() {return _PayoutMethodRequest["default"];} });Object.defineProperty(exports, "PayoutMethodResponse", { enumerable: true, get: function get() {return _PayoutMethodResponse["default"];} });Object.defineProperty(exports, "PayoutMethodTransferReasonEnum", { enumerable: true, get: function get() {return _PayoutMethodTransferReasonEnum["default"];} });Object.defineProperty(exports, "PayoutMethodWebhook", { enumerable: true, get: function get() {return _PayoutMethodWebhook["default"];} });Object.defineProperty(exports, "PayoutMethodWebhookAllOf", { enumerable: true, get: function get() {return _PayoutMethodWebhookAllOf["default"];} });Object.defineProperty(exports, "PayoutMethodsApi", { enumerable: true, get: function get() {return _PayoutMethodsApi["default"];} });Object.defineProperty(exports, "PoliticallyExposedPerson", { enumerable: true, get: function get() {return _PoliticallyExposedPerson["default"];} });Object.defineProperty(exports, "ProofOfPayment", { enumerable: true, get: function get() {return _ProofOfPayment["default"];} });Object.defineProperty(exports, "ProofOfPaymentListResponse", { enumerable: true, get: function get() {return _ProofOfPaymentListResponse["default"];} });Object.defineProperty(exports, "Recipient", { enumerable: true, get: function get() {return _Recipient["default"];} });Object.defineProperty(exports, "RecipientListResponse", { enumerable: true, get: function get() {return _RecipientListResponse["default"];} });Object.defineProperty(exports, "RecipientRequest", { enumerable: true, get: function get() {return _RecipientRequest["default"];} });Object.defineProperty(exports, "RecipientResponse", { enumerable: true, get: function get() {return _RecipientResponse["default"];} });Object.defineProperty(exports, "RecipientState", { enumerable: true, get: function get() {return _RecipientState["default"];} });Object.defineProperty(exports, "RecipientStateReasonDetails", { enumerable: true, get: function get() {return _RecipientStateReasonDetails["default"];} });Object.defineProperty(exports, "RecipientWebhook", { enumerable: true, get: function get() {return _RecipientWebhook["default"];} });Object.defineProperty(exports, "RecipientWebhookAllOf", { enumerable: true, get: function get() {return _RecipientWebhookAllOf["default"];} });Object.defineProperty(exports, "RecipientsApi", { enumerable: true, get: function get() {return _RecipientsApi["default"];} });Object.defineProperty(exports, "Sender", { enumerable: true, get: function get() {return _Sender["default"];} });Object.defineProperty(exports, "SenderListResponse", { enumerable: true, get: function get() {return _SenderListResponse["default"];} });Object.defineProperty(exports, "SenderRequest", { enumerable: true, get: function get() {return _SenderRequest["default"];} });Object.defineProperty(exports, "SenderResponse", { enumerable: true, get: function get() {return _SenderResponse["default"];} });Object.defineProperty(exports, "SenderResponseExisting", { enumerable: true, get: function get() {return _SenderResponseExisting["default"];} });Object.defineProperty(exports, "SenderResponseMeta", { enumerable: true, get: function get() {return _SenderResponseMeta["default"];} });Object.defineProperty(exports, "SenderState", { enumerable: true, get: function get() {return _SenderState["default"];} });Object.defineProperty(exports, "SenderWebhook", { enumerable: true, get: function get() {return _SenderWebhook["default"];} });Object.defineProperty(exports, "SenderWebhookAllOf", { enumerable: true, get: function get() {return _SenderWebhookAllOf["default"];} });Object.defineProperty(exports, "SendersApi", { enumerable: true, get: function get() {return _SendersApi["default"];} });Object.defineProperty(exports, "StateReasonDetails", { enumerable: true, get: function get() {return _StateReasonDetails["default"];} });Object.defineProperty(exports, "Transaction", { enumerable: true, get: function get() {return _Transaction["default"];} });Object.defineProperty(exports, "TransactionListResponse", { enumerable: true, get: function get() {return _TransactionListResponse["default"];} });Object.defineProperty(exports, "TransactionRequest", { enumerable: true, get: function get() {return _TransactionRequest["default"];} });Object.defineProperty(exports, "TransactionResponse", { enumerable: true, get: function get() {return _TransactionResponse["default"];} });Object.defineProperty(exports, "TransactionResponseExisting", { enumerable: true, get: function get() {return _TransactionResponseExisting["default"];} });Object.defineProperty(exports, "TransactionResponseMeta", { enumerable: true, get: function get() {return _TransactionResponseMeta["default"];} });Object.defineProperty(exports, "TransactionState", { enumerable: true, get: function get() {return _TransactionState["default"];} });Object.defineProperty(exports, "TransactionTraits", { enumerable: true, get: function get() {return _TransactionTraits["default"];} });Object.defineProperty(exports, "TransactionWebhook", { enumerable: true, get: function get() {return _TransactionWebhook["default"];} });Object.defineProperty(exports, "TransactionWebhookAllOf", { enumerable: true, get: function get() {return _TransactionWebhookAllOf["default"];} });Object.defineProperty(exports, "TransactionsApi", { enumerable: true, get: function get() {return _TransactionsApi["default"];} });Object.defineProperty(exports, "ValidationErrorDescription", { enumerable: true, get: function get() {return _ValidationErrorDescription["default"];} });Object.defineProperty(exports, "Webhook", { enumerable: true, get: function get() {return _Webhook["default"];} });Object.defineProperty(exports, "WebhookDefinition", { enumerable: true, get: function get() {return _WebhookDefinition["default"];} });Object.defineProperty(exports, "WebhookDefinitionEventListResponse", { enumerable: true, get: function get() {return _WebhookDefinitionEventListResponse["default"];} });Object.defineProperty(exports, "WebhookDefinitionListResponse", { enumerable: true, get: function get() {return _WebhookDefinitionListResponse["default"];} });Object.defineProperty(exports, "WebhookDefinitionRequest", { enumerable: true, get: function get() {return _WebhookDefinitionRequest["default"];} });Object.defineProperty(exports, "WebhookDefinitionResponse", { enumerable: true, get: function get() {return _WebhookDefinitionResponse["default"];} });Object.defineProperty(exports, "WebhookLog", { enumerable: true, get: function get() {return _WebhookLog["default"];} });Object.defineProperty(exports, "WebhookLogListResponse", { enumerable: true, get: function get() {return _WebhookLogListResponse["default"];} });Object.defineProperty(exports, "WebhookLogMetadata", { enumerable: true, get: function get() {return _WebhookLogMetadata["default"];} });Object.defineProperty(exports, "WebhookLogMetadataRequest", { enumerable: true, get: function get() {return _WebhookLogMetadataRequest["default"];} });Object.defineProperty(exports, "WebhookLogMetadataResponse", { enumerable: true, get: function get() {return _WebhookLogMetadataResponse["default"];} });Object.defineProperty(exports, "WebhookLogResponse", { enumerable: true, get: function get() {return _WebhookLogResponse["default"];} });Object.defineProperty(exports, "WebhooksApi", { enumerable: true, get: function get() {return _WebhooksApi["default"];} });exports["default"] = void 0;













var _ApiClient = _interopRequireDefault(require("./ApiClient"));
var _Account = _interopRequireDefault(require("./Model/Account"));
var _AccountListResponse = _interopRequireDefault(require("./Model/AccountListResponse"));
var _AccountMeta = _interopRequireDefault(require("./Model/AccountMeta"));
var _AccountResponse = _interopRequireDefault(require("./Model/AccountResponse"));
var _AccountValidationError = _interopRequireDefault(require("./Model/AccountValidationError"));
var _AccountValidationRequest = _interopRequireDefault(require("./Model/AccountValidationRequest"));
var _AccountValidationResponse = _interopRequireDefault(require("./Model/AccountValidationResponse"));
var _AccountValidationResult = _interopRequireDefault(require("./Model/AccountValidationResult"));
var _ApiLog = _interopRequireDefault(require("./Model/ApiLog"));
var _ApiLogListResponse = _interopRequireDefault(require("./Model/ApiLogListResponse"));
var _ApiLogResponse = _interopRequireDefault(require("./Model/ApiLogResponse"));
var _Currency = _interopRequireDefault(require("./Model/Currency"));
var _CurrencyExchange = _interopRequireDefault(require("./Model/CurrencyExchange"));
var _CurrencyExchangeAllOf = _interopRequireDefault(require("./Model/CurrencyExchangeAllOf"));
var _CurrencyExchangeListResponse = _interopRequireDefault(require("./Model/CurrencyExchangeListResponse"));
var _CurrencyListResponse = _interopRequireDefault(require("./Model/CurrencyListResponse"));
var _CurrencyOpposite = _interopRequireDefault(require("./Model/CurrencyOpposite"));
var _CurrencyOppositeAllOf = _interopRequireDefault(require("./Model/CurrencyOppositeAllOf"));
var _Debit = _interopRequireDefault(require("./Model/Debit"));
var _DebitListRequest = _interopRequireDefault(require("./Model/DebitListRequest"));
var _DebitListResponse = _interopRequireDefault(require("./Model/DebitListResponse"));
var _DebitRequest = _interopRequireDefault(require("./Model/DebitRequest"));
var _DebitRequestWrapper = _interopRequireDefault(require("./Model/DebitRequestWrapper"));
var _Document = _interopRequireDefault(require("./Model/Document"));
var _DocumentListResponse = _interopRequireDefault(require("./Model/DocumentListResponse"));
var _DocumentRequest = _interopRequireDefault(require("./Model/DocumentRequest"));
var _DocumentResponse = _interopRequireDefault(require("./Model/DocumentResponse"));
var _DocumentWebhook = _interopRequireDefault(require("./Model/DocumentWebhook"));
var _DocumentWebhookAllOf = _interopRequireDefault(require("./Model/DocumentWebhookAllOf"));
var _ErrorStatus = _interopRequireDefault(require("./Model/ErrorStatus"));
var _FieldDescription = _interopRequireDefault(require("./Model/FieldDescription"));
var _FieldSelectValidation = _interopRequireDefault(require("./Model/FieldSelectValidation"));
var _FieldValidation = _interopRequireDefault(require("./Model/FieldValidation"));
var _Pagination = _interopRequireDefault(require("./Model/Pagination"));
var _PaginationMeta = _interopRequireDefault(require("./Model/PaginationMeta"));
var _PayinMethod = _interopRequireDefault(require("./Model/PayinMethod"));
var _PayinMethodDetails = _interopRequireDefault(require("./Model/PayinMethodDetails"));
var _PayinMethodDetailsBTC = _interopRequireDefault(require("./Model/PayinMethodDetailsBTC"));
var _PayinMethodDetailsMobile = _interopRequireDefault(require("./Model/PayinMethodDetailsMobile"));
var _PayinMethodDetailsNGNBank = _interopRequireDefault(require("./Model/PayinMethodDetailsNGNBank"));
var _PayinMethodRequest = _interopRequireDefault(require("./Model/PayinMethodRequest"));
var _PayinMethodResponse = _interopRequireDefault(require("./Model/PayinMethodResponse"));
var _PayinMethodState = _interopRequireDefault(require("./Model/PayinMethodState"));
var _PayinMethodUxFlow = _interopRequireDefault(require("./Model/PayinMethodUxFlow"));
var _PaymentMethod = _interopRequireDefault(require("./Model/PaymentMethod"));
var _PaymentMethodListResponse = _interopRequireDefault(require("./Model/PaymentMethodListResponse"));
var _PaymentMethodOpposite = _interopRequireDefault(require("./Model/PaymentMethodOpposite"));
var _PaymentMethodOppositeAllOf = _interopRequireDefault(require("./Model/PaymentMethodOppositeAllOf"));
var _PayoutMethod = _interopRequireDefault(require("./Model/PayoutMethod"));
var _PayoutMethodBankAccountTypeEnum = _interopRequireDefault(require("./Model/PayoutMethodBankAccountTypeEnum"));
var _PayoutMethodCashProviderEnum = _interopRequireDefault(require("./Model/PayoutMethodCashProviderEnum"));
var _PayoutMethodCountryEnum = _interopRequireDefault(require("./Model/PayoutMethodCountryEnum"));
var _PayoutMethodDetails = _interopRequireDefault(require("./Model/PayoutMethodDetails"));
var _PayoutMethodDetailsBRLBank = _interopRequireDefault(require("./Model/PayoutMethodDetailsBRLBank"));
var _PayoutMethodDetailsBTC = _interopRequireDefault(require("./Model/PayoutMethodDetailsBTC"));
var _PayoutMethodDetailsBWPBank = _interopRequireDefault(require("./Model/PayoutMethodDetailsBWPBank"));
var _PayoutMethodDetailsBalance = _interopRequireDefault(require("./Model/PayoutMethodDetailsBalance"));
var _PayoutMethodDetailsCADBank = _interopRequireDefault(require("./Model/PayoutMethodDetailsCADBank"));
var _PayoutMethodDetailsEGPBank = _interopRequireDefault(require("./Model/PayoutMethodDetailsEGPBank"));
var _PayoutMethodDetailsGBPBank = _interopRequireDefault(require("./Model/PayoutMethodDetailsGBPBank"));
var _PayoutMethodDetailsGHSBank = _interopRequireDefault(require("./Model/PayoutMethodDetailsGHSBank"));
var _PayoutMethodDetailsGHSCash = _interopRequireDefault(require("./Model/PayoutMethodDetailsGHSCash"));
var _PayoutMethodDetailsGNFMobile = _interopRequireDefault(require("./Model/PayoutMethodDetailsGNFMobile"));
var _PayoutMethodDetailsIBAN = _interopRequireDefault(require("./Model/PayoutMethodDetailsIBAN"));
var _PayoutMethodDetailsINRBank = _interopRequireDefault(require("./Model/PayoutMethodDetailsINRBank"));
var _PayoutMethodDetailsKESBank = _interopRequireDefault(require("./Model/PayoutMethodDetailsKESBank"));
var _PayoutMethodDetailsKESMobile = _interopRequireDefault(require("./Model/PayoutMethodDetailsKESMobile"));
var _PayoutMethodDetailsMADCash = _interopRequireDefault(require("./Model/PayoutMethodDetailsMADCash"));
var _PayoutMethodDetailsMobile = _interopRequireDefault(require("./Model/PayoutMethodDetailsMobile"));
var _PayoutMethodDetailsNGNBank = _interopRequireDefault(require("./Model/PayoutMethodDetailsNGNBank"));
var _PayoutMethodDetailsNZDBank = _interopRequireDefault(require("./Model/PayoutMethodDetailsNZDBank"));
var _PayoutMethodDetailsUGXBank = _interopRequireDefault(require("./Model/PayoutMethodDetailsUGXBank"));
var _PayoutMethodDetailsUSDBank = _interopRequireDefault(require("./Model/PayoutMethodDetailsUSDBank"));
var _PayoutMethodDetailsUSDCash = _interopRequireDefault(require("./Model/PayoutMethodDetailsUSDCash"));
var _PayoutMethodDetailsXAFBank = _interopRequireDefault(require("./Model/PayoutMethodDetailsXAFBank"));
var _PayoutMethodDetailsXAFMobile = _interopRequireDefault(require("./Model/PayoutMethodDetailsXAFMobile"));
var _PayoutMethodDetailsXOFBank = _interopRequireDefault(require("./Model/PayoutMethodDetailsXOFBank"));
var _PayoutMethodDetailsXOFCash = _interopRequireDefault(require("./Model/PayoutMethodDetailsXOFCash"));
var _PayoutMethodDetailsXOFMobile = _interopRequireDefault(require("./Model/PayoutMethodDetailsXOFMobile"));
var _PayoutMethodDetailsZARBank = _interopRequireDefault(require("./Model/PayoutMethodDetailsZARBank"));
var _PayoutMethodDetailsZMWBank = _interopRequireDefault(require("./Model/PayoutMethodDetailsZMWBank"));
var _PayoutMethodGenderEnum = _interopRequireDefault(require("./Model/PayoutMethodGenderEnum"));
var _PayoutMethodIdentityCardTypeEnum = _interopRequireDefault(require("./Model/PayoutMethodIdentityCardTypeEnum"));
var _PayoutMethodLegalEntityTypeEnum = _interopRequireDefault(require("./Model/PayoutMethodLegalEntityTypeEnum"));
var _PayoutMethodListResponse = _interopRequireDefault(require("./Model/PayoutMethodListResponse"));
var _PayoutMethodMobileProviderEnum = _interopRequireDefault(require("./Model/PayoutMethodMobileProviderEnum"));
var _PayoutMethodNatureOfBusinessEnum = _interopRequireDefault(require("./Model/PayoutMethodNatureOfBusinessEnum"));
var _PayoutMethodPixKeyTypeEnum = _interopRequireDefault(require("./Model/PayoutMethodPixKeyTypeEnum"));
var _PayoutMethodRequest = _interopRequireDefault(require("./Model/PayoutMethodRequest"));
var _PayoutMethodResponse = _interopRequireDefault(require("./Model/PayoutMethodResponse"));
var _PayoutMethodTransferReasonEnum = _interopRequireDefault(require("./Model/PayoutMethodTransferReasonEnum"));
var _PayoutMethodWebhook = _interopRequireDefault(require("./Model/PayoutMethodWebhook"));
var _PayoutMethodWebhookAllOf = _interopRequireDefault(require("./Model/PayoutMethodWebhookAllOf"));
var _PoliticallyExposedPerson = _interopRequireDefault(require("./Model/PoliticallyExposedPerson"));
var _ProofOfPayment = _interopRequireDefault(require("./Model/ProofOfPayment"));
var _ProofOfPaymentListResponse = _interopRequireDefault(require("./Model/ProofOfPaymentListResponse"));
var _Recipient = _interopRequireDefault(require("./Model/Recipient"));
var _RecipientListResponse = _interopRequireDefault(require("./Model/RecipientListResponse"));
var _RecipientRequest = _interopRequireDefault(require("./Model/RecipientRequest"));
var _RecipientResponse = _interopRequireDefault(require("./Model/RecipientResponse"));
var _RecipientState = _interopRequireDefault(require("./Model/RecipientState"));
var _RecipientStateReasonDetails = _interopRequireDefault(require("./Model/RecipientStateReasonDetails"));
var _RecipientWebhook = _interopRequireDefault(require("./Model/RecipientWebhook"));
var _RecipientWebhookAllOf = _interopRequireDefault(require("./Model/RecipientWebhookAllOf"));
var _Sender = _interopRequireDefault(require("./Model/Sender"));
var _SenderListResponse = _interopRequireDefault(require("./Model/SenderListResponse"));
var _SenderRequest = _interopRequireDefault(require("./Model/SenderRequest"));
var _SenderResponse = _interopRequireDefault(require("./Model/SenderResponse"));
var _SenderResponseExisting = _interopRequireDefault(require("./Model/SenderResponseExisting"));
var _SenderResponseMeta = _interopRequireDefault(require("./Model/SenderResponseMeta"));
var _SenderState = _interopRequireDefault(require("./Model/SenderState"));
var _SenderWebhook = _interopRequireDefault(require("./Model/SenderWebhook"));
var _SenderWebhookAllOf = _interopRequireDefault(require("./Model/SenderWebhookAllOf"));
var _StateReasonDetails = _interopRequireDefault(require("./Model/StateReasonDetails"));
var _Transaction = _interopRequireDefault(require("./Model/Transaction"));
var _TransactionListResponse = _interopRequireDefault(require("./Model/TransactionListResponse"));
var _TransactionRequest = _interopRequireDefault(require("./Model/TransactionRequest"));
var _TransactionResponse = _interopRequireDefault(require("./Model/TransactionResponse"));
var _TransactionResponseExisting = _interopRequireDefault(require("./Model/TransactionResponseExisting"));
var _TransactionResponseMeta = _interopRequireDefault(require("./Model/TransactionResponseMeta"));
var _TransactionState = _interopRequireDefault(require("./Model/TransactionState"));
var _TransactionTraits = _interopRequireDefault(require("./Model/TransactionTraits"));
var _TransactionWebhook = _interopRequireDefault(require("./Model/TransactionWebhook"));
var _TransactionWebhookAllOf = _interopRequireDefault(require("./Model/TransactionWebhookAllOf"));
var _ValidationErrorDescription = _interopRequireDefault(require("./Model/ValidationErrorDescription"));
var _Webhook = _interopRequireDefault(require("./Model/Webhook"));
var _WebhookDefinition = _interopRequireDefault(require("./Model/WebhookDefinition"));
var _WebhookDefinitionEventListResponse = _interopRequireDefault(require("./Model/WebhookDefinitionEventListResponse"));
var _WebhookDefinitionListResponse = _interopRequireDefault(require("./Model/WebhookDefinitionListResponse"));
var _WebhookDefinitionRequest = _interopRequireDefault(require("./Model/WebhookDefinitionRequest"));
var _WebhookDefinitionResponse = _interopRequireDefault(require("./Model/WebhookDefinitionResponse"));
var _WebhookLog = _interopRequireDefault(require("./Model/WebhookLog"));
var _WebhookLogListResponse = _interopRequireDefault(require("./Model/WebhookLogListResponse"));
var _WebhookLogMetadata = _interopRequireDefault(require("./Model/WebhookLogMetadata"));
var _WebhookLogMetadataRequest = _interopRequireDefault(require("./Model/WebhookLogMetadataRequest"));
var _WebhookLogMetadataResponse = _interopRequireDefault(require("./Model/WebhookLogMetadataResponse"));
var _WebhookLogResponse = _interopRequireDefault(require("./Model/WebhookLogResponse"));
var _APILogsApi = _interopRequireDefault(require("./Api/APILogsApi"));
var _AccountDebitsApi = _interopRequireDefault(require("./Api/AccountDebitsApi"));
var _AccountValidationApi = _interopRequireDefault(require("./Api/AccountValidationApi"));
var _AccountsApi = _interopRequireDefault(require("./Api/AccountsApi"));
var _CurrencyInfoApi = _interopRequireDefault(require("./Api/CurrencyInfoApi"));
var _DocumentsApi = _interopRequireDefault(require("./Api/DocumentsApi"));
var _LogsApi = _interopRequireDefault(require("./Api/LogsApi"));
var _PayinMethodsApi = _interopRequireDefault(require("./Api/PayinMethodsApi"));
var _PaymentMethodsApi = _interopRequireDefault(require("./Api/PaymentMethodsApi"));
var _PayoutMethodsApi = _interopRequireDefault(require("./Api/PayoutMethodsApi"));
var _RecipientsApi = _interopRequireDefault(require("./Api/RecipientsApi"));
var _SendersApi = _interopRequireDefault(require("./Api/SendersApi"));
var _TransactionsApi = _interopRequireDefault(require("./Api/TransactionsApi"));
var _WebhooksApi = _interopRequireDefault(require("./Api/WebhooksApi"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
* Client library for TransferZero API..<br>
* The <code>index</code> module provides access to constructors for all the classes which comprise the public API.
* <p>
* An AMD (recommended!) or CommonJS application will generally do something equivalent to the following:
* <pre>
* var TransferzeroSdk = require('index'); // See note below*.
* var xxxSvc = new TransferzeroSdk.XxxApi(); // Allocate the API class we're going to use.
* var yyyModel = new TransferzeroSdk.Yyy(); // Construct a model instance.
* yyyModel.someProperty = 'someValue';
* ...
* var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
* ...
* </pre>
* <em>*NOTE: For a top-level AMD script, use require(['index'], function(){...})
* and put the application logic within the callback function.</em>
* </p>
* <p>
* A non-AMD browser application (discouraged) might do something like this:
* <pre>
* var xxxSvc = new TransferzeroSdk.XxxApi(); // Allocate the API class we're going to use.
* var yyy = new TransferzeroSdk.Yyy(); // Construct a model instance.
* yyyModel.someProperty = 'someValue';
* ...
* var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
* ...
* </pre>
* </p>
* @module index
* @version 1.35.1
*/var _default = exports["default"] = { /**
   * The ApiClient constructor.
   * @property {module:ApiClient}
   */ApiClient: _ApiClient["default"], /**
   * The Account model constructor.
   * @property {module:Model/Account}
   */Account: _Account["default"], /**
   * The AccountListResponse model constructor.
   * @property {module:Model/AccountListResponse}
   */
  AccountListResponse: _AccountListResponse["default"],

  /**
   * The AccountMeta model constructor.
   * @property {module:Model/AccountMeta}
   */
  AccountMeta: _AccountMeta["default"],

  /**
   * The AccountResponse model constructor.
   * @property {module:Model/AccountResponse}
   */
  AccountResponse: _AccountResponse["default"],

  /**
   * The AccountValidationError model constructor.
   * @property {module:Model/AccountValidationError}
   */
  AccountValidationError: _AccountValidationError["default"],

  /**
   * The AccountValidationRequest model constructor.
   * @property {module:Model/AccountValidationRequest}
   */
  AccountValidationRequest: _AccountValidationRequest["default"],

  /**
   * The AccountValidationResponse model constructor.
   * @property {module:Model/AccountValidationResponse}
   */
  AccountValidationResponse: _AccountValidationResponse["default"],

  /**
   * The AccountValidationResult model constructor.
   * @property {module:Model/AccountValidationResult}
   */
  AccountValidationResult: _AccountValidationResult["default"],

  /**
   * The ApiLog model constructor.
   * @property {module:Model/ApiLog}
   */
  ApiLog: _ApiLog["default"],

  /**
   * The ApiLogListResponse model constructor.
   * @property {module:Model/ApiLogListResponse}
   */
  ApiLogListResponse: _ApiLogListResponse["default"],

  /**
   * The ApiLogResponse model constructor.
   * @property {module:Model/ApiLogResponse}
   */
  ApiLogResponse: _ApiLogResponse["default"],

  /**
   * The Currency model constructor.
   * @property {module:Model/Currency}
   */
  Currency: _Currency["default"],

  /**
   * The CurrencyExchange model constructor.
   * @property {module:Model/CurrencyExchange}
   */
  CurrencyExchange: _CurrencyExchange["default"],

  /**
   * The CurrencyExchangeAllOf model constructor.
   * @property {module:Model/CurrencyExchangeAllOf}
   */
  CurrencyExchangeAllOf: _CurrencyExchangeAllOf["default"],

  /**
   * The CurrencyExchangeListResponse model constructor.
   * @property {module:Model/CurrencyExchangeListResponse}
   */
  CurrencyExchangeListResponse: _CurrencyExchangeListResponse["default"],

  /**
   * The CurrencyListResponse model constructor.
   * @property {module:Model/CurrencyListResponse}
   */
  CurrencyListResponse: _CurrencyListResponse["default"],

  /**
   * The CurrencyOpposite model constructor.
   * @property {module:Model/CurrencyOpposite}
   */
  CurrencyOpposite: _CurrencyOpposite["default"],

  /**
   * The CurrencyOppositeAllOf model constructor.
   * @property {module:Model/CurrencyOppositeAllOf}
   */
  CurrencyOppositeAllOf: _CurrencyOppositeAllOf["default"],

  /**
   * The Debit model constructor.
   * @property {module:Model/Debit}
   */
  Debit: _Debit["default"],

  /**
   * The DebitListRequest model constructor.
   * @property {module:Model/DebitListRequest}
   */
  DebitListRequest: _DebitListRequest["default"],

  /**
   * The DebitListResponse model constructor.
   * @property {module:Model/DebitListResponse}
   */
  DebitListResponse: _DebitListResponse["default"],

  /**
   * The DebitRequest model constructor.
   * @property {module:Model/DebitRequest}
   */
  DebitRequest: _DebitRequest["default"],

  /**
   * The DebitRequestWrapper model constructor.
   * @property {module:Model/DebitRequestWrapper}
   */
  DebitRequestWrapper: _DebitRequestWrapper["default"],

  /**
   * The Document model constructor.
   * @property {module:Model/Document}
   */
  Document: _Document["default"],

  /**
   * The DocumentListResponse model constructor.
   * @property {module:Model/DocumentListResponse}
   */
  DocumentListResponse: _DocumentListResponse["default"],

  /**
   * The DocumentRequest model constructor.
   * @property {module:Model/DocumentRequest}
   */
  DocumentRequest: _DocumentRequest["default"],

  /**
   * The DocumentResponse model constructor.
   * @property {module:Model/DocumentResponse}
   */
  DocumentResponse: _DocumentResponse["default"],

  /**
   * The DocumentWebhook model constructor.
   * @property {module:Model/DocumentWebhook}
   */
  DocumentWebhook: _DocumentWebhook["default"],

  /**
   * The DocumentWebhookAllOf model constructor.
   * @property {module:Model/DocumentWebhookAllOf}
   */
  DocumentWebhookAllOf: _DocumentWebhookAllOf["default"],

  /**
   * The ErrorStatus model constructor.
   * @property {module:Model/ErrorStatus}
   */
  ErrorStatus: _ErrorStatus["default"],

  /**
   * The FieldDescription model constructor.
   * @property {module:Model/FieldDescription}
   */
  FieldDescription: _FieldDescription["default"],

  /**
   * The FieldSelectValidation model constructor.
   * @property {module:Model/FieldSelectValidation}
   */
  FieldSelectValidation: _FieldSelectValidation["default"],

  /**
   * The FieldValidation model constructor.
   * @property {module:Model/FieldValidation}
   */
  FieldValidation: _FieldValidation["default"],

  /**
   * The Pagination model constructor.
   * @property {module:Model/Pagination}
   */
  Pagination: _Pagination["default"],

  /**
   * The PaginationMeta model constructor.
   * @property {module:Model/PaginationMeta}
   */
  PaginationMeta: _PaginationMeta["default"],

  /**
   * The PayinMethod model constructor.
   * @property {module:Model/PayinMethod}
   */
  PayinMethod: _PayinMethod["default"],

  /**
   * The PayinMethodDetails model constructor.
   * @property {module:Model/PayinMethodDetails}
   */
  PayinMethodDetails: _PayinMethodDetails["default"],

  /**
   * The PayinMethodDetailsBTC model constructor.
   * @property {module:Model/PayinMethodDetailsBTC}
   */
  PayinMethodDetailsBTC: _PayinMethodDetailsBTC["default"],

  /**
   * The PayinMethodDetailsMobile model constructor.
   * @property {module:Model/PayinMethodDetailsMobile}
   */
  PayinMethodDetailsMobile: _PayinMethodDetailsMobile["default"],

  /**
   * The PayinMethodDetailsNGNBank model constructor.
   * @property {module:Model/PayinMethodDetailsNGNBank}
   */
  PayinMethodDetailsNGNBank: _PayinMethodDetailsNGNBank["default"],

  /**
   * The PayinMethodRequest model constructor.
   * @property {module:Model/PayinMethodRequest}
   */
  PayinMethodRequest: _PayinMethodRequest["default"],

  /**
   * The PayinMethodResponse model constructor.
   * @property {module:Model/PayinMethodResponse}
   */
  PayinMethodResponse: _PayinMethodResponse["default"],

  /**
   * The PayinMethodState model constructor.
   * @property {module:Model/PayinMethodState}
   */
  PayinMethodState: _PayinMethodState["default"],

  /**
   * The PayinMethodUxFlow model constructor.
   * @property {module:Model/PayinMethodUxFlow}
   */
  PayinMethodUxFlow: _PayinMethodUxFlow["default"],

  /**
   * The PaymentMethod model constructor.
   * @property {module:Model/PaymentMethod}
   */
  PaymentMethod: _PaymentMethod["default"],

  /**
   * The PaymentMethodListResponse model constructor.
   * @property {module:Model/PaymentMethodListResponse}
   */
  PaymentMethodListResponse: _PaymentMethodListResponse["default"],

  /**
   * The PaymentMethodOpposite model constructor.
   * @property {module:Model/PaymentMethodOpposite}
   */
  PaymentMethodOpposite: _PaymentMethodOpposite["default"],

  /**
   * The PaymentMethodOppositeAllOf model constructor.
   * @property {module:Model/PaymentMethodOppositeAllOf}
   */
  PaymentMethodOppositeAllOf: _PaymentMethodOppositeAllOf["default"],

  /**
   * The PayoutMethod model constructor.
   * @property {module:Model/PayoutMethod}
   */
  PayoutMethod: _PayoutMethod["default"],

  /**
   * The PayoutMethodBankAccountTypeEnum model constructor.
   * @property {module:Model/PayoutMethodBankAccountTypeEnum}
   */
  PayoutMethodBankAccountTypeEnum: _PayoutMethodBankAccountTypeEnum["default"],

  /**
   * The PayoutMethodCashProviderEnum model constructor.
   * @property {module:Model/PayoutMethodCashProviderEnum}
   */
  PayoutMethodCashProviderEnum: _PayoutMethodCashProviderEnum["default"],

  /**
   * The PayoutMethodCountryEnum model constructor.
   * @property {module:Model/PayoutMethodCountryEnum}
   */
  PayoutMethodCountryEnum: _PayoutMethodCountryEnum["default"],

  /**
   * The PayoutMethodDetails model constructor.
   * @property {module:Model/PayoutMethodDetails}
   */
  PayoutMethodDetails: _PayoutMethodDetails["default"],

  /**
   * The PayoutMethodDetailsBRLBank model constructor.
   * @property {module:Model/PayoutMethodDetailsBRLBank}
   */
  PayoutMethodDetailsBRLBank: _PayoutMethodDetailsBRLBank["default"],

  /**
   * The PayoutMethodDetailsBTC model constructor.
   * @property {module:Model/PayoutMethodDetailsBTC}
   */
  PayoutMethodDetailsBTC: _PayoutMethodDetailsBTC["default"],

  /**
   * The PayoutMethodDetailsBWPBank model constructor.
   * @property {module:Model/PayoutMethodDetailsBWPBank}
   */
  PayoutMethodDetailsBWPBank: _PayoutMethodDetailsBWPBank["default"],

  /**
   * The PayoutMethodDetailsBalance model constructor.
   * @property {module:Model/PayoutMethodDetailsBalance}
   */
  PayoutMethodDetailsBalance: _PayoutMethodDetailsBalance["default"],

  /**
   * The PayoutMethodDetailsCADBank model constructor.
   * @property {module:Model/PayoutMethodDetailsCADBank}
   */
  PayoutMethodDetailsCADBank: _PayoutMethodDetailsCADBank["default"],

  /**
   * The PayoutMethodDetailsEGPBank model constructor.
   * @property {module:Model/PayoutMethodDetailsEGPBank}
   */
  PayoutMethodDetailsEGPBank: _PayoutMethodDetailsEGPBank["default"],

  /**
   * The PayoutMethodDetailsGBPBank model constructor.
   * @property {module:Model/PayoutMethodDetailsGBPBank}
   */
  PayoutMethodDetailsGBPBank: _PayoutMethodDetailsGBPBank["default"],

  /**
   * The PayoutMethodDetailsGHSBank model constructor.
   * @property {module:Model/PayoutMethodDetailsGHSBank}
   */
  PayoutMethodDetailsGHSBank: _PayoutMethodDetailsGHSBank["default"],

  /**
   * The PayoutMethodDetailsGHSCash model constructor.
   * @property {module:Model/PayoutMethodDetailsGHSCash}
   */
  PayoutMethodDetailsGHSCash: _PayoutMethodDetailsGHSCash["default"],

  /**
   * The PayoutMethodDetailsGNFMobile model constructor.
   * @property {module:Model/PayoutMethodDetailsGNFMobile}
   */
  PayoutMethodDetailsGNFMobile: _PayoutMethodDetailsGNFMobile["default"],

  /**
   * The PayoutMethodDetailsIBAN model constructor.
   * @property {module:Model/PayoutMethodDetailsIBAN}
   */
  PayoutMethodDetailsIBAN: _PayoutMethodDetailsIBAN["default"],

  /**
   * The PayoutMethodDetailsINRBank model constructor.
   * @property {module:Model/PayoutMethodDetailsINRBank}
   */
  PayoutMethodDetailsINRBank: _PayoutMethodDetailsINRBank["default"],

  /**
   * The PayoutMethodDetailsKESBank model constructor.
   * @property {module:Model/PayoutMethodDetailsKESBank}
   */
  PayoutMethodDetailsKESBank: _PayoutMethodDetailsKESBank["default"],

  /**
   * The PayoutMethodDetailsKESMobile model constructor.
   * @property {module:Model/PayoutMethodDetailsKESMobile}
   */
  PayoutMethodDetailsKESMobile: _PayoutMethodDetailsKESMobile["default"],

  /**
   * The PayoutMethodDetailsMADCash model constructor.
   * @property {module:Model/PayoutMethodDetailsMADCash}
   */
  PayoutMethodDetailsMADCash: _PayoutMethodDetailsMADCash["default"],

  /**
   * The PayoutMethodDetailsMobile model constructor.
   * @property {module:Model/PayoutMethodDetailsMobile}
   */
  PayoutMethodDetailsMobile: _PayoutMethodDetailsMobile["default"],

  /**
   * The PayoutMethodDetailsNGNBank model constructor.
   * @property {module:Model/PayoutMethodDetailsNGNBank}
   */
  PayoutMethodDetailsNGNBank: _PayoutMethodDetailsNGNBank["default"],

  /**
   * The PayoutMethodDetailsNZDBank model constructor.
   * @property {module:Model/PayoutMethodDetailsNZDBank}
   */
  PayoutMethodDetailsNZDBank: _PayoutMethodDetailsNZDBank["default"],

  /**
   * The PayoutMethodDetailsUGXBank model constructor.
   * @property {module:Model/PayoutMethodDetailsUGXBank}
   */
  PayoutMethodDetailsUGXBank: _PayoutMethodDetailsUGXBank["default"],

  /**
   * The PayoutMethodDetailsUSDBank model constructor.
   * @property {module:Model/PayoutMethodDetailsUSDBank}
   */
  PayoutMethodDetailsUSDBank: _PayoutMethodDetailsUSDBank["default"],

  /**
   * The PayoutMethodDetailsUSDCash model constructor.
   * @property {module:Model/PayoutMethodDetailsUSDCash}
   */
  PayoutMethodDetailsUSDCash: _PayoutMethodDetailsUSDCash["default"],

  /**
   * The PayoutMethodDetailsXAFBank model constructor.
   * @property {module:Model/PayoutMethodDetailsXAFBank}
   */
  PayoutMethodDetailsXAFBank: _PayoutMethodDetailsXAFBank["default"],

  /**
   * The PayoutMethodDetailsXAFMobile model constructor.
   * @property {module:Model/PayoutMethodDetailsXAFMobile}
   */
  PayoutMethodDetailsXAFMobile: _PayoutMethodDetailsXAFMobile["default"],

  /**
   * The PayoutMethodDetailsXOFBank model constructor.
   * @property {module:Model/PayoutMethodDetailsXOFBank}
   */
  PayoutMethodDetailsXOFBank: _PayoutMethodDetailsXOFBank["default"],

  /**
   * The PayoutMethodDetailsXOFCash model constructor.
   * @property {module:Model/PayoutMethodDetailsXOFCash}
   */
  PayoutMethodDetailsXOFCash: _PayoutMethodDetailsXOFCash["default"],

  /**
   * The PayoutMethodDetailsXOFMobile model constructor.
   * @property {module:Model/PayoutMethodDetailsXOFMobile}
   */
  PayoutMethodDetailsXOFMobile: _PayoutMethodDetailsXOFMobile["default"],

  /**
   * The PayoutMethodDetailsZARBank model constructor.
   * @property {module:Model/PayoutMethodDetailsZARBank}
   */
  PayoutMethodDetailsZARBank: _PayoutMethodDetailsZARBank["default"],

  /**
   * The PayoutMethodDetailsZMWBank model constructor.
   * @property {module:Model/PayoutMethodDetailsZMWBank}
   */
  PayoutMethodDetailsZMWBank: _PayoutMethodDetailsZMWBank["default"],

  /**
   * The PayoutMethodGenderEnum model constructor.
   * @property {module:Model/PayoutMethodGenderEnum}
   */
  PayoutMethodGenderEnum: _PayoutMethodGenderEnum["default"],

  /**
   * The PayoutMethodIdentityCardTypeEnum model constructor.
   * @property {module:Model/PayoutMethodIdentityCardTypeEnum}
   */
  PayoutMethodIdentityCardTypeEnum: _PayoutMethodIdentityCardTypeEnum["default"],

  /**
   * The PayoutMethodLegalEntityTypeEnum model constructor.
   * @property {module:Model/PayoutMethodLegalEntityTypeEnum}
   */
  PayoutMethodLegalEntityTypeEnum: _PayoutMethodLegalEntityTypeEnum["default"],

  /**
   * The PayoutMethodListResponse model constructor.
   * @property {module:Model/PayoutMethodListResponse}
   */
  PayoutMethodListResponse: _PayoutMethodListResponse["default"],

  /**
   * The PayoutMethodMobileProviderEnum model constructor.
   * @property {module:Model/PayoutMethodMobileProviderEnum}
   */
  PayoutMethodMobileProviderEnum: _PayoutMethodMobileProviderEnum["default"],

  /**
   * The PayoutMethodNatureOfBusinessEnum model constructor.
   * @property {module:Model/PayoutMethodNatureOfBusinessEnum}
   */
  PayoutMethodNatureOfBusinessEnum: _PayoutMethodNatureOfBusinessEnum["default"],

  /**
   * The PayoutMethodPixKeyTypeEnum model constructor.
   * @property {module:Model/PayoutMethodPixKeyTypeEnum}
   */
  PayoutMethodPixKeyTypeEnum: _PayoutMethodPixKeyTypeEnum["default"],

  /**
   * The PayoutMethodRequest model constructor.
   * @property {module:Model/PayoutMethodRequest}
   */
  PayoutMethodRequest: _PayoutMethodRequest["default"],

  /**
   * The PayoutMethodResponse model constructor.
   * @property {module:Model/PayoutMethodResponse}
   */
  PayoutMethodResponse: _PayoutMethodResponse["default"],

  /**
   * The PayoutMethodTransferReasonEnum model constructor.
   * @property {module:Model/PayoutMethodTransferReasonEnum}
   */
  PayoutMethodTransferReasonEnum: _PayoutMethodTransferReasonEnum["default"],

  /**
   * The PayoutMethodWebhook model constructor.
   * @property {module:Model/PayoutMethodWebhook}
   */
  PayoutMethodWebhook: _PayoutMethodWebhook["default"],

  /**
   * The PayoutMethodWebhookAllOf model constructor.
   * @property {module:Model/PayoutMethodWebhookAllOf}
   */
  PayoutMethodWebhookAllOf: _PayoutMethodWebhookAllOf["default"],

  /**
   * The PoliticallyExposedPerson model constructor.
   * @property {module:Model/PoliticallyExposedPerson}
   */
  PoliticallyExposedPerson: _PoliticallyExposedPerson["default"],

  /**
   * The ProofOfPayment model constructor.
   * @property {module:Model/ProofOfPayment}
   */
  ProofOfPayment: _ProofOfPayment["default"],

  /**
   * The ProofOfPaymentListResponse model constructor.
   * @property {module:Model/ProofOfPaymentListResponse}
   */
  ProofOfPaymentListResponse: _ProofOfPaymentListResponse["default"],

  /**
   * The Recipient model constructor.
   * @property {module:Model/Recipient}
   */
  Recipient: _Recipient["default"],

  /**
   * The RecipientListResponse model constructor.
   * @property {module:Model/RecipientListResponse}
   */
  RecipientListResponse: _RecipientListResponse["default"],

  /**
   * The RecipientRequest model constructor.
   * @property {module:Model/RecipientRequest}
   */
  RecipientRequest: _RecipientRequest["default"],

  /**
   * The RecipientResponse model constructor.
   * @property {module:Model/RecipientResponse}
   */
  RecipientResponse: _RecipientResponse["default"],

  /**
   * The RecipientState model constructor.
   * @property {module:Model/RecipientState}
   */
  RecipientState: _RecipientState["default"],

  /**
   * The RecipientStateReasonDetails model constructor.
   * @property {module:Model/RecipientStateReasonDetails}
   */
  RecipientStateReasonDetails: _RecipientStateReasonDetails["default"],

  /**
   * The RecipientWebhook model constructor.
   * @property {module:Model/RecipientWebhook}
   */
  RecipientWebhook: _RecipientWebhook["default"],

  /**
   * The RecipientWebhookAllOf model constructor.
   * @property {module:Model/RecipientWebhookAllOf}
   */
  RecipientWebhookAllOf: _RecipientWebhookAllOf["default"],

  /**
   * The Sender model constructor.
   * @property {module:Model/Sender}
   */
  Sender: _Sender["default"],

  /**
   * The SenderListResponse model constructor.
   * @property {module:Model/SenderListResponse}
   */
  SenderListResponse: _SenderListResponse["default"],

  /**
   * The SenderRequest model constructor.
   * @property {module:Model/SenderRequest}
   */
  SenderRequest: _SenderRequest["default"],

  /**
   * The SenderResponse model constructor.
   * @property {module:Model/SenderResponse}
   */
  SenderResponse: _SenderResponse["default"],

  /**
   * The SenderResponseExisting model constructor.
   * @property {module:Model/SenderResponseExisting}
   */
  SenderResponseExisting: _SenderResponseExisting["default"],

  /**
   * The SenderResponseMeta model constructor.
   * @property {module:Model/SenderResponseMeta}
   */
  SenderResponseMeta: _SenderResponseMeta["default"],

  /**
   * The SenderState model constructor.
   * @property {module:Model/SenderState}
   */
  SenderState: _SenderState["default"],

  /**
   * The SenderWebhook model constructor.
   * @property {module:Model/SenderWebhook}
   */
  SenderWebhook: _SenderWebhook["default"],

  /**
   * The SenderWebhookAllOf model constructor.
   * @property {module:Model/SenderWebhookAllOf}
   */
  SenderWebhookAllOf: _SenderWebhookAllOf["default"],

  /**
   * The StateReasonDetails model constructor.
   * @property {module:Model/StateReasonDetails}
   */
  StateReasonDetails: _StateReasonDetails["default"],

  /**
   * The Transaction model constructor.
   * @property {module:Model/Transaction}
   */
  Transaction: _Transaction["default"],

  /**
   * The TransactionListResponse model constructor.
   * @property {module:Model/TransactionListResponse}
   */
  TransactionListResponse: _TransactionListResponse["default"],

  /**
   * The TransactionRequest model constructor.
   * @property {module:Model/TransactionRequest}
   */
  TransactionRequest: _TransactionRequest["default"],

  /**
   * The TransactionResponse model constructor.
   * @property {module:Model/TransactionResponse}
   */
  TransactionResponse: _TransactionResponse["default"],

  /**
   * The TransactionResponseExisting model constructor.
   * @property {module:Model/TransactionResponseExisting}
   */
  TransactionResponseExisting: _TransactionResponseExisting["default"],

  /**
   * The TransactionResponseMeta model constructor.
   * @property {module:Model/TransactionResponseMeta}
   */
  TransactionResponseMeta: _TransactionResponseMeta["default"],

  /**
   * The TransactionState model constructor.
   * @property {module:Model/TransactionState}
   */
  TransactionState: _TransactionState["default"],

  /**
   * The TransactionTraits model constructor.
   * @property {module:Model/TransactionTraits}
   */
  TransactionTraits: _TransactionTraits["default"],

  /**
   * The TransactionWebhook model constructor.
   * @property {module:Model/TransactionWebhook}
   */
  TransactionWebhook: _TransactionWebhook["default"],

  /**
   * The TransactionWebhookAllOf model constructor.
   * @property {module:Model/TransactionWebhookAllOf}
   */
  TransactionWebhookAllOf: _TransactionWebhookAllOf["default"],

  /**
   * The ValidationErrorDescription model constructor.
   * @property {module:Model/ValidationErrorDescription}
   */
  ValidationErrorDescription: _ValidationErrorDescription["default"],

  /**
   * The Webhook model constructor.
   * @property {module:Model/Webhook}
   */
  Webhook: _Webhook["default"],

  /**
   * The WebhookDefinition model constructor.
   * @property {module:Model/WebhookDefinition}
   */
  WebhookDefinition: _WebhookDefinition["default"],

  /**
   * The WebhookDefinitionEventListResponse model constructor.
   * @property {module:Model/WebhookDefinitionEventListResponse}
   */
  WebhookDefinitionEventListResponse: _WebhookDefinitionEventListResponse["default"],

  /**
   * The WebhookDefinitionListResponse model constructor.
   * @property {module:Model/WebhookDefinitionListResponse}
   */
  WebhookDefinitionListResponse: _WebhookDefinitionListResponse["default"],

  /**
   * The WebhookDefinitionRequest model constructor.
   * @property {module:Model/WebhookDefinitionRequest}
   */
  WebhookDefinitionRequest: _WebhookDefinitionRequest["default"],

  /**
   * The WebhookDefinitionResponse model constructor.
   * @property {module:Model/WebhookDefinitionResponse}
   */
  WebhookDefinitionResponse: _WebhookDefinitionResponse["default"],

  /**
   * The WebhookLog model constructor.
   * @property {module:Model/WebhookLog}
   */
  WebhookLog: _WebhookLog["default"],

  /**
   * The WebhookLogListResponse model constructor.
   * @property {module:Model/WebhookLogListResponse}
   */
  WebhookLogListResponse: _WebhookLogListResponse["default"],

  /**
   * The WebhookLogMetadata model constructor.
   * @property {module:Model/WebhookLogMetadata}
   */
  WebhookLogMetadata: _WebhookLogMetadata["default"],

  /**
   * The WebhookLogMetadataRequest model constructor.
   * @property {module:Model/WebhookLogMetadataRequest}
   */
  WebhookLogMetadataRequest: _WebhookLogMetadataRequest["default"],

  /**
   * The WebhookLogMetadataResponse model constructor.
   * @property {module:Model/WebhookLogMetadataResponse}
   */
  WebhookLogMetadataResponse: _WebhookLogMetadataResponse["default"],

  /**
   * The WebhookLogResponse model constructor.
   * @property {module:Model/WebhookLogResponse}
   */
  WebhookLogResponse: _WebhookLogResponse["default"],

  /**
  * The APILogsApi service constructor.
  * @property {module:Api/APILogsApi}
  */
  APILogsApi: _APILogsApi["default"],

  /**
  * The AccountDebitsApi service constructor.
  * @property {module:Api/AccountDebitsApi}
  */
  AccountDebitsApi: _AccountDebitsApi["default"],

  /**
  * The AccountValidationApi service constructor.
  * @property {module:Api/AccountValidationApi}
  */
  AccountValidationApi: _AccountValidationApi["default"],

  /**
  * The AccountsApi service constructor.
  * @property {module:Api/AccountsApi}
  */
  AccountsApi: _AccountsApi["default"],

  /**
  * The CurrencyInfoApi service constructor.
  * @property {module:Api/CurrencyInfoApi}
  */
  CurrencyInfoApi: _CurrencyInfoApi["default"],

  /**
  * The DocumentsApi service constructor.
  * @property {module:Api/DocumentsApi}
  */
  DocumentsApi: _DocumentsApi["default"],

  /**
  * The LogsApi service constructor.
  * @property {module:Api/LogsApi}
  */
  LogsApi: _LogsApi["default"],

  /**
  * The PayinMethodsApi service constructor.
  * @property {module:Api/PayinMethodsApi}
  */
  PayinMethodsApi: _PayinMethodsApi["default"],

  /**
  * The PaymentMethodsApi service constructor.
  * @property {module:Api/PaymentMethodsApi}
  */
  PaymentMethodsApi: _PaymentMethodsApi["default"],

  /**
  * The PayoutMethodsApi service constructor.
  * @property {module:Api/PayoutMethodsApi}
  */
  PayoutMethodsApi: _PayoutMethodsApi["default"],

  /**
  * The RecipientsApi service constructor.
  * @property {module:Api/RecipientsApi}
  */
  RecipientsApi: _RecipientsApi["default"],

  /**
  * The SendersApi service constructor.
  * @property {module:Api/SendersApi}
  */
  SendersApi: _SendersApi["default"],

  /**
  * The TransactionsApi service constructor.
  * @property {module:Api/TransactionsApi}
  */
  TransactionsApi: _TransactionsApi["default"],

  /**
  * The WebhooksApi service constructor.
  * @property {module:Api/WebhooksApi}
  */
  WebhooksApi: _WebhooksApi["default"]
};
