// transferzero-sdk/dist/Model/AccountValidationRequest.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports["default"] = void 0;












var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _PayoutMethodMobileProviderEnum = _interopRequireDefault(require("./PayoutMethodMobileProviderEnum"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };}function _typeof(o) {"@babel/helpers - typeof";return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) {return typeof o;} : function (o) {return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o;}, _typeof(o);}function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError("Cannot call a class as a function");}}function _defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ("value" in descriptor) descriptor.writable = true;Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);}}function _createClass(Constructor, protoProps, staticProps) {if (protoProps) _defineProperties(Constructor.prototype, protoProps);if (staticProps) _defineProperties(Constructor, staticProps);Object.defineProperty(Constructor, "prototype", { writable: false });return Constructor;}function _toPropertyKey(t) {var i = _toPrimitive(t, "string");return "symbol" == _typeof(i) ? i : i + "";}function _toPrimitive(t, r) {if ("object" != _typeof(t) || !t) return t;var e = t[Symbol.toPrimitive];if (void 0 !== e) {var i = e.call(t, r || "default");if ("object" != _typeof(i)) return i;throw new TypeError("@@toPrimitive must return a primitive value.");}return ("string" === r ? String : Number)(t);} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
 * The AccountValidationRequest model module.
 * @module Model/AccountValidationRequest
 * @version 1.35.1
 */var AccountValidationRequest = /*#__PURE__*/function () {/**
   * Constructs a new <code>AccountValidationRequest</code>.
   * @alias module:Model/AccountValidationRequest
   * @param country {module:Model/AccountValidationRequest.CountryEnum} Country of account in 2-character alpha ISO 3166-2 country format
   * @param currency {module:Model/AccountValidationRequest.CurrencyEnum} The currency the bank account is in
   * @param method {module:Model/AccountValidationRequest.MethodEnum} The method of the payment. Currently bank and mobile are supported
   */function AccountValidationRequest(country, currency, method) {_classCallCheck(this, AccountValidationRequest);AccountValidationRequest.initialize(this, country, currency, method);} /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */return _createClass(AccountValidationRequest, null, [{ key: "initialize", value: function initialize(obj, country, currency, method) {
      obj['country'] = country;
      obj['currency'] = currency;
      obj['method'] = method;
    }

    /**
     * Constructs a <code>AccountValidationRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:Model/AccountValidationRequest} obj Optional instance to populate.
     * @return {module:Model/AccountValidationRequest} The populated <code>AccountValidationRequest</code> instance.
     */ }, { key: "constructFromObject", value:
    function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AccountValidationRequest();

        if (data.hasOwnProperty('bank_account')) {
          obj['bank_account'] = _ApiClient["default"].convertToType(data['bank_account'], 'String');
        }
        if (data.hasOwnProperty('bank_code')) {
          obj['bank_code'] = _ApiClient["default"].convertToType(data['bank_code'], 'String');
        }
        if (data.hasOwnProperty('iban')) {
          obj['iban'] = _ApiClient["default"].convertToType(data['iban'], 'String');
        }
        if (data.hasOwnProperty('phone_number')) {
          obj['phone_number'] = _ApiClient["default"].convertToType(data['phone_number'], 'String');
        }
        if (data.hasOwnProperty('mobile_provider')) {
          obj['mobile_provider'] = _PayoutMethodMobileProviderEnum["default"].constructFromObject(data['mobile_provider']);
        }
        if (data.hasOwnProperty('country')) {
          obj['country'] = _ApiClient["default"].convertToType(data['country'], 'String');
        }
        if (data.hasOwnProperty('currency')) {
          obj['currency'] = _ApiClient["default"].convertToType(data['currency'], 'String');
        }
        if (data.hasOwnProperty('method')) {
          obj['method'] = _ApiClient["default"].convertToType(data['method'], 'String');
        }
      }
      return obj;
    } }]);}();




/**
 * Account Number to query
 * @member {String} bank_account
 */
AccountValidationRequest.prototype['bank_account'] = undefined;

/**
 * Bank Code to query - same codes are used as for creating the transactions
 * @member {String} bank_code
 */
AccountValidationRequest.prototype['bank_code'] = undefined;

/**
 * IBAN to query - BBAN format for XOF bank accounts
 * @member {String} iban
 */
AccountValidationRequest.prototype['iban'] = undefined;

/**
 * Phone number to query in E.164 format
 * @member {String} phone_number
 */
AccountValidationRequest.prototype['phone_number'] = undefined;

/**
 * @member {module:Model/PayoutMethodMobileProviderEnum} mobile_provider
 */
AccountValidationRequest.prototype['mobile_provider'] = undefined;

/**
 * Country of account in 2-character alpha ISO 3166-2 country format
 * @member {module:Model/AccountValidationRequest.CountryEnum} country
 */
AccountValidationRequest.prototype['country'] = undefined;

/**
 * The currency the bank account is in
 * @member {module:Model/AccountValidationRequest.CurrencyEnum} currency
 */
AccountValidationRequest.prototype['currency'] = undefined;

/**
 * The method of the payment. Currently bank and mobile are supported
 * @member {module:Model/AccountValidationRequest.MethodEnum} method
 */
AccountValidationRequest.prototype['method'] = undefined;





/**
 * Allowed values for the <code>country</code> property.
 * @enum {String}
 * @readonly
 */
AccountValidationRequest['CountryEnum'] = {

  /**
   * value: "BF"
   * @const
   */
  "BF": "BF",

  /**
   * value: "BJ"
   * @const
   */
  "BJ": "BJ",

  /**
   * value: "CG"
   * @const
   */
  "CG": "CG",

  /**
   * value: "CI"
   * @const
   */
  "CI": "CI",

  /**
   * value: "CM"
   * @const
   */
  "CM": "CM",

  /**
   * value: "GA"
   * @const
   */
  "GA": "GA",

  /**
   * value: "GH"
   * @const
   */
  "GH": "GH",

  /**
   * value: "NG"
   * @const
   */
  "NG": "NG",

  /**
   * value: "SN"
   * @const
   */
  "SN": "SN",

  /**
   * value: "TD"
   * @const
   */
  "TD": "TD",

  /**
   * value: "TG"
   * @const
   */
  "TG": "TG",

  /**
   * value: "UG"
   * @const
   */
  "UG": "UG"
};


/**
 * Allowed values for the <code>currency</code> property.
 * @enum {String}
 * @readonly
 */
AccountValidationRequest['CurrencyEnum'] = {

  /**
   * value: "NGN"
   * @const
   */
  "NGN": "NGN",

  /**
   * value: "GHS"
   * @const
   */
  "GHS": "GHS",

  /**
   * value: "XOF"
   * @const
   */
  "XOF": "XOF",

  /**
   * value: "UGX"
   * @const
   */
  "UGX": "UGX",

  /**
   * value: "XAF"
   * @const
   */
  "XAF": "XAF"
};


/**
 * Allowed values for the <code>method</code> property.
 * @enum {String}
 * @readonly
 */
AccountValidationRequest['MethodEnum'] = {

  /**
   * value: "bank"
   * @const
   */
  "bank": "bank",

  /**
   * value: "mobile"
   * @const
   */
  "mobile": "mobile"
};var _default = exports["default"] =



AccountValidationRequest;
