// transferzero-sdk/dist/Model/ApiLog.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports["default"] = void 0;












var _ApiClient = _interopRequireDefault(require("../ApiClient"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };}function _typeof(o) {"@babel/helpers - typeof";return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) {return typeof o;} : function (o) {return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o;}, _typeof(o);}function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError("Cannot call a class as a function");}}function _defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ("value" in descriptor) descriptor.writable = true;Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);}}function _createClass(Constructor, protoProps, staticProps) {if (protoProps) _defineProperties(Constructor.prototype, protoProps);if (staticProps) _defineProperties(Constructor, staticProps);Object.defineProperty(Constructor, "prototype", { writable: false });return Constructor;}function _toPropertyKey(t) {var i = _toPrimitive(t, "string");return "symbol" == _typeof(i) ? i : i + "";}function _toPrimitive(t, r) {if ("object" != _typeof(t) || !t) return t;var e = t[Symbol.toPrimitive];if (void 0 !== e) {var i = e.call(t, r || "default");if ("object" != _typeof(i)) return i;throw new TypeError("@@toPrimitive must return a primitive value.");}return ("string" === r ? String : Number)(t);} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
 * The ApiLog model module.
 * @module Model/ApiLog
 * @version 1.35.1
 */var ApiLog = /*#__PURE__*/function () {/**
   * Constructs a new <code>ApiLog</code>.
   * The response object returned with requests to access individual API logs 
   * @alias module:Model/ApiLog
   */function ApiLog() {_classCallCheck(this, ApiLog);ApiLog.initialize(this);} /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */return _createClass(ApiLog, null, [{ key: "initialize", value: function initialize(obj) {
    }

    /**
     * Constructs a <code>ApiLog</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:Model/ApiLog} obj Optional instance to populate.
     * @return {module:Model/ApiLog} The populated <code>ApiLog</code> instance.
     */ }, { key: "constructFromObject", value:
    function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ApiLog();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('request_method')) {
          obj['request_method'] = _ApiClient["default"].convertToType(data['request_method'], 'String');
        }
        if (data.hasOwnProperty('request_url')) {
          obj['request_url'] = _ApiClient["default"].convertToType(data['request_url'], 'String');
        }
        if (data.hasOwnProperty('request_body')) {
          obj['request_body'] = _ApiClient["default"].convertToType(data['request_body'], 'String');
        }
        if (data.hasOwnProperty('request_headers')) {
          obj['request_headers'] = _ApiClient["default"].convertToType(data['request_headers'], { 'String': 'String' });
        }
        if (data.hasOwnProperty('response_status')) {
          obj['response_status'] = _ApiClient["default"].convertToType(data['response_status'], 'String');
        }
        if (data.hasOwnProperty('response_body')) {
          obj['response_body'] = _ApiClient["default"].convertToType(data['response_body'], 'String');
        }
        if (data.hasOwnProperty('response_headers')) {
          obj['response_headers'] = _ApiClient["default"].convertToType(data['response_headers'], { 'String': 'String' });
        }
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'String');
        }
      }
      return obj;
    } }]);}();




/**
 * ID of the API log
 * @member {String} id
 */
ApiLog.prototype['id'] = undefined;

/**
 * Initial HTTP request method
 * @member {String} request_method
 */
ApiLog.prototype['request_method'] = undefined;

/**
 * URL to which initial request was sent
 * @member {String} request_url
 */
ApiLog.prototype['request_url'] = undefined;

/**
 * Body sent with initial request returned as a string
 * @member {String} request_body
 */
ApiLog.prototype['request_body'] = undefined;

/**
 * Authorization and access information sent with initial request
 * @member {Object.<String, String>} request_headers
 */
ApiLog.prototype['request_headers'] = undefined;

/**
 * Status of response
 * @member {String} response_status
 */
ApiLog.prototype['response_status'] = undefined;

/**
 * Full response body returned as a string
 * @member {String} response_body
 */
ApiLog.prototype['response_body'] = undefined;

/**
 * Headers returned with response
 * @member {Object.<String, String>} response_headers
 */
ApiLog.prototype['response_headers'] = undefined;

/**
 * Date and time of response
 * @member {String} created_at
 */
ApiLog.prototype['created_at'] = undefined;var _default = exports["default"] =






ApiLog;
