// transferzero-sdk/dist/Model/CurrencyOpposite.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports["default"] = void 0;












var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Currency = _interopRequireDefault(require("./Currency"));
var _CurrencyOppositeAllOf = _interopRequireDefault(require("./CurrencyOppositeAllOf"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };}function _typeof(o) {"@babel/helpers - typeof";return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) {return typeof o;} : function (o) {return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o;}, _typeof(o);}function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError("Cannot call a class as a function");}}function _defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ("value" in descriptor) descriptor.writable = true;Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);}}function _createClass(Constructor, protoProps, staticProps) {if (protoProps) _defineProperties(Constructor.prototype, protoProps);if (staticProps) _defineProperties(Constructor, staticProps);Object.defineProperty(Constructor, "prototype", { writable: false });return Constructor;}function _toPropertyKey(t) {var i = _toPrimitive(t, "string");return "symbol" == _typeof(i) ? i : i + "";}function _toPrimitive(t, r) {if ("object" != _typeof(t) || !t) return t;var e = t[Symbol.toPrimitive];if (void 0 !== e) {var i = e.call(t, r || "default");if ("object" != _typeof(i)) return i;throw new TypeError("@@toPrimitive must return a primitive value.");}return ("string" === r ? String : Number)(t);} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
 * The CurrencyOpposite model module.
 * @module Model/CurrencyOpposite
 * @version 1.35.1
 */var CurrencyOpposite = /*#__PURE__*/function () {/**
   * Constructs a new <code>CurrencyOpposite</code>.
   * @alias module:Model/CurrencyOpposite
   * @implements module:Model/Currency
   * @implements module:Model/CurrencyOppositeAllOf
   */function CurrencyOpposite() {_classCallCheck(this, CurrencyOpposite);_Currency["default"].initialize(this);_CurrencyOppositeAllOf["default"].initialize(this);CurrencyOpposite.initialize(this);} /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */return _createClass(CurrencyOpposite, null, [{ key: "initialize", value: function initialize(obj) {
    }

    /**
     * Constructs a <code>CurrencyOpposite</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:Model/CurrencyOpposite} obj Optional instance to populate.
     * @return {module:Model/CurrencyOpposite} The populated <code>CurrencyOpposite</code> instance.
     */ }, { key: "constructFromObject", value:
    function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CurrencyOpposite();
        _Currency["default"].constructFromObject(data, obj);
        _CurrencyOppositeAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('code')) {
          obj['code'] = _ApiClient["default"].convertToType(data['code'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('symbol')) {
          obj['symbol'] = _ApiClient["default"].convertToType(data['symbol'], 'String');
        }
        if (data.hasOwnProperty('decimals')) {
          obj['decimals'] = _ApiClient["default"].convertToType(data['decimals'], 'Number');
        }
        if (data.hasOwnProperty('subunit_to_unit')) {
          obj['subunit_to_unit'] = _ApiClient["default"].convertToType(data['subunit_to_unit'], 'String');
        }
        if (data.hasOwnProperty('primary')) {
          obj['primary'] = _ApiClient["default"].convertToType(data['primary'], 'Boolean');
        }
        if (data.hasOwnProperty('min')) {
          obj['min'] = _ApiClient["default"].convertToType(data['min'], 'String');
        }
        if (data.hasOwnProperty('max')) {
          obj['max'] = _ApiClient["default"].convertToType(data['max'], 'String');
        }
        if (data.hasOwnProperty('usd_equivalent')) {
          obj['usd_equivalent'] = _ApiClient["default"].convertToType(data['usd_equivalent'], 'String');
        }
        if (data.hasOwnProperty('rate')) {
          obj['rate'] = _ApiClient["default"].convertToType(data['rate'], 'Number');
        }
        if (data.hasOwnProperty('mtm_rate')) {
          obj['mtm_rate'] = _ApiClient["default"].convertToType(data['mtm_rate'], 'Number');
        }
        if (data.hasOwnProperty('margin')) {
          obj['margin'] = _ApiClient["default"].convertToType(data['margin'], 'String');
        }
      }
      return obj;
    } }]);}();




/**
 * The currency code in 3-character alpha ISO 4217 currency format
 * @member {String} code
 */
CurrencyOpposite.prototype['code'] = undefined;

/**
 * Name of currency
 * @member {String} name
 */
CurrencyOpposite.prototype['name'] = undefined;

/**
 * Symbol of currency
 * @member {String} symbol
 */
CurrencyOpposite.prototype['symbol'] = undefined;

/**
 * Number of decimal points
 * @member {Number} decimals
 */
CurrencyOpposite.prototype['decimals'] = undefined;

/**
 * Subunits in Unit (eg. there are 100 cents in 1 US Dollar)
 * @member {String} subunit_to_unit
 */
CurrencyOpposite.prototype['subunit_to_unit'] = undefined;

/**
 * Is this a primary currency?
 * @member {Boolean} primary
 */
CurrencyOpposite.prototype['primary'] = undefined;

/**
 * The minimum amount allowed in a transaction
 * @member {String} min
 */
CurrencyOpposite.prototype['min'] = undefined;

/**
 * The maximum amount allowed in a transaction
 * @member {String} max
 */
CurrencyOpposite.prototype['max'] = undefined;

/**
 * The equivalent of the currency to 1 USD
 * @member {String} usd_equivalent
 */
CurrencyOpposite.prototype['usd_equivalent'] = undefined;

/**
 * The rate of this particular currency with the base one
 * @member {Number} rate
 */
CurrencyOpposite.prototype['rate'] = undefined;

/**
 * Mark to market rate of this particular currency against the base one with the margin factored in
 * @member {Number} mtm_rate
 */
CurrencyOpposite.prototype['mtm_rate'] = undefined;

/**
 * The margin set for transactions of this particular currency with the base one
 * @member {String} margin
 */
CurrencyOpposite.prototype['margin'] = undefined;


// Implement Currency interface:
/**
 * The currency code in 3-character alpha ISO 4217 currency format
 * @member {String} code
 */
_Currency["default"].prototype['code'] = undefined;
/**
 * Name of currency
 * @member {String} name
 */
_Currency["default"].prototype['name'] = undefined;
/**
 * Symbol of currency
 * @member {String} symbol
 */
_Currency["default"].prototype['symbol'] = undefined;
/**
 * Number of decimal points
 * @member {Number} decimals
 */
_Currency["default"].prototype['decimals'] = undefined;
/**
 * Subunits in Unit (eg. there are 100 cents in 1 US Dollar)
 * @member {String} subunit_to_unit
 */
_Currency["default"].prototype['subunit_to_unit'] = undefined;
/**
 * Is this a primary currency?
 * @member {Boolean} primary
 */
_Currency["default"].prototype['primary'] = undefined;
/**
 * The minimum amount allowed in a transaction
 * @member {String} min
 */
_Currency["default"].prototype['min'] = undefined;
/**
 * The maximum amount allowed in a transaction
 * @member {String} max
 */
_Currency["default"].prototype['max'] = undefined;
/**
 * The equivalent of the currency to 1 USD
 * @member {String} usd_equivalent
 */
_Currency["default"].prototype['usd_equivalent'] = undefined;
// Implement CurrencyOppositeAllOf interface:
/**
 * The rate of this particular currency with the base one
 * @member {Number} rate
 */
_CurrencyOppositeAllOf["default"].prototype['rate'] = undefined;
/**
 * Mark to market rate of this particular currency against the base one with the margin factored in
 * @member {Number} mtm_rate
 */
_CurrencyOppositeAllOf["default"].prototype['mtm_rate'] = undefined;
/**
 * The margin set for transactions of this particular currency with the base one
 * @member {String} margin
 */
_CurrencyOppositeAllOf["default"].prototype['margin'] = undefined;var _default = exports["default"] =




CurrencyOpposite;
