// transferzero-sdk/dist/Model/Document.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports["default"] = void 0;












var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ValidationErrorDescription = _interopRequireDefault(require("./ValidationErrorDescription"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };}function _typeof(o) {"@babel/helpers - typeof";return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) {return typeof o;} : function (o) {return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o;}, _typeof(o);}function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError("Cannot call a class as a function");}}function _defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ("value" in descriptor) descriptor.writable = true;Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);}}function _createClass(Constructor, protoProps, staticProps) {if (protoProps) _defineProperties(Constructor.prototype, protoProps);if (staticProps) _defineProperties(Constructor, staticProps);Object.defineProperty(Constructor, "prototype", { writable: false });return Constructor;}function _toPropertyKey(t) {var i = _toPrimitive(t, "string");return "symbol" == _typeof(i) ? i : i + "";}function _toPrimitive(t, r) {if ("object" != _typeof(t) || !t) return t;var e = t[Symbol.toPrimitive];if (void 0 !== e) {var i = e.call(t, r || "default");if ("object" != _typeof(i)) return i;throw new TypeError("@@toPrimitive must return a primitive value.");}return ("string" === r ? String : Number)(t);} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
 * The Document model module.
 * @module Model/Document
 * @version 1.35.1
 */var Document = /*#__PURE__*/function () {/**
   * Constructs a new <code>Document</code>.
   * @alias module:Model/Document
   * @param upload {String} Base64 encoded data uri of an image/pdf file or a fully qualified url
   * @param uploadFileName {String} Name of the upload
   */function Document(upload, uploadFileName) {_classCallCheck(this, Document);Document.initialize(this, upload, uploadFileName);} /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */return _createClass(Document, null, [{ key: "initialize", value: function initialize(obj, upload, uploadFileName) {
      obj['upload'] = upload;
      obj['upload_file_name'] = uploadFileName;
    }

    /**
     * Constructs a <code>Document</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:Model/Document} obj Optional instance to populate.
     * @return {module:Model/Document} The populated <code>Document</code> instance.
     */ }, { key: "constructFromObject", value:
    function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Document();

        if (data.hasOwnProperty('upload')) {
          obj['upload'] = _ApiClient["default"].convertToType(data['upload'], 'String');
        }
        if (data.hasOwnProperty('url')) {
          obj['url'] = _ApiClient["default"].convertToType(data['url'], 'String');
        }
        if (data.hasOwnProperty('upload_file_name')) {
          obj['upload_file_name'] = _ApiClient["default"].convertToType(data['upload_file_name'], 'String');
        }
        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], Object);
        }
        if (data.hasOwnProperty('upload_content_type')) {
          obj['upload_content_type'] = _ApiClient["default"].convertToType(data['upload_content_type'], 'String');
        }
        if (data.hasOwnProperty('upload_file_size')) {
          obj['upload_file_size'] = _ApiClient["default"].convertToType(data['upload_file_size'], 'Number');
        }
        if (data.hasOwnProperty('category')) {
          obj['category'] = _ApiClient["default"].convertToType(data['category'], 'String');
        }
        if (data.hasOwnProperty('side')) {
          obj['side'] = _ApiClient["default"].convertToType(data['side'], 'String');
        }
        if (data.hasOwnProperty('document_type')) {
          obj['document_type'] = _ApiClient["default"].convertToType(data['document_type'], 'String');
        }
        if (data.hasOwnProperty('issuing_country')) {
          obj['issuing_country'] = _ApiClient["default"].convertToType(data['issuing_country'], 'String');
        }
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('document_id')) {
          obj['document_id'] = _ApiClient["default"].convertToType(data['document_id'], 'String');
        }
        if (data.hasOwnProperty('expiry_date')) {
          obj['expiry_date'] = _ApiClient["default"].convertToType(data['expiry_date'], 'Date');
        }
        if (data.hasOwnProperty('source')) {
          obj['source'] = _ApiClient["default"].convertToType(data['source'], 'String');
        }
        if (data.hasOwnProperty('errors')) {
          obj['errors'] = _ApiClient["default"].convertToType(data['errors'], { 'String': [_ValidationErrorDescription["default"]] });
        }
      }
      return obj;
    } }]);}();




/**
 * Base64 encoded data uri of an image/pdf file or a fully qualified url
 * @member {String} upload
 */
Document.prototype['upload'] = undefined;

/**
 * URL of the document location
 * @member {String} url
 */
Document.prototype['url'] = undefined;

/**
 * Name of the upload
 * @member {String} upload_file_name
 */
Document.prototype['upload_file_name'] = undefined;

/**
 * Metadata of document
 * @member {Object} metadata
 */
Document.prototype['metadata'] = undefined;

/**
 * @member {String} upload_content_type
 */
Document.prototype['upload_content_type'] = undefined;

/**
 * @member {Number} upload_file_size
 */
Document.prototype['upload_file_size'] = undefined;

/**
 * uncategorised
 * @member {String} category
 */
Document.prototype['category'] = undefined;

/**
 * The side of the KYC ID. One of 'front' or 'back'
 * @member {module:Model/Document.SideEnum} side
 */
Document.prototype['side'] = undefined;

/**
 * This is a brief description of the document type
 * @member {String} document_type
 */
Document.prototype['document_type'] = undefined;

/**
 * Issuing country of ID in 2-character alpha ISO 3166-2 country format
 * @member {String} issuing_country
 */
Document.prototype['issuing_country'] = undefined;

/**
 * @member {String} id
 */
Document.prototype['id'] = undefined;

/**
 * Document ID issued by government
 * @member {String} document_id
 */
Document.prototype['document_id'] = undefined;

/**
 * Document expiry date issued by government
 * @member {Date} expiry_date
 */
Document.prototype['expiry_date'] = undefined;

/**
 * Determines the document's source. Default value \"Manual\"
 * @member {String} source
 */
Document.prototype['source'] = undefined;

/**
 * The fields that have some problems and don't pass validation
 * @member {Object.<String, Array.<module:Model/ValidationErrorDescription>>} errors
 */
Document.prototype['errors'] = undefined;





/**
 * Allowed values for the <code>side</code> property.
 * @enum {String}
 * @readonly
 */
Document['SideEnum'] = {

  /**
   * value: "front"
   * @const
   */
  "front": "front",

  /**
   * value: "back"
   * @const
   */
  "back": "back"
};var _default = exports["default"] =



Document;
