// transferzero-sdk/dist/Model/PayinMethod.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports["default"] = void 0;












var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _PayinMethodDetails = _interopRequireDefault(require("./PayinMethodDetails"));
var _PayinMethodState = _interopRequireDefault(require("./PayinMethodState"));
var _PayinMethodUxFlow = _interopRequireDefault(require("./PayinMethodUxFlow"));
var _StateReasonDetails = _interopRequireDefault(require("./StateReasonDetails"));
var _ValidationErrorDescription = _interopRequireDefault(require("./ValidationErrorDescription"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };}function _typeof(o) {"@babel/helpers - typeof";return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) {return typeof o;} : function (o) {return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o;}, _typeof(o);}function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError("Cannot call a class as a function");}}function _defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ("value" in descriptor) descriptor.writable = true;Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);}}function _createClass(Constructor, protoProps, staticProps) {if (protoProps) _defineProperties(Constructor.prototype, protoProps);if (staticProps) _defineProperties(Constructor, staticProps);Object.defineProperty(Constructor, "prototype", { writable: false });return Constructor;}function _toPropertyKey(t) {var i = _toPrimitive(t, "string");return "symbol" == _typeof(i) ? i : i + "";}function _toPrimitive(t, r) {if ("object" != _typeof(t) || !t) return t;var e = t[Symbol.toPrimitive];if (void 0 !== e) {var i = e.call(t, r || "default");if ("object" != _typeof(i)) return i;throw new TypeError("@@toPrimitive must return a primitive value.");}return ("string" === r ? String : Number)(t);} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
 * The PayinMethod model module.
 * @module Model/PayinMethod
 * @version 1.35.1
 */var PayinMethod = /*#__PURE__*/function () {/**
   * Constructs a new <code>PayinMethod</code>.
   * This describes the specific details on how the funds should be collected from the sender.
   * @alias module:Model/PayinMethod
   */function PayinMethod() {_classCallCheck(this, PayinMethod);PayinMethod.initialize(this);} /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */return _createClass(PayinMethod, null, [{ key: "initialize", value: function initialize(obj) {
    }

    /**
     * Constructs a <code>PayinMethod</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:Model/PayinMethod} obj Optional instance to populate.
     * @return {module:Model/PayinMethod} The populated <code>PayinMethod</code> instance.
     */ }, { key: "constructFromObject", value:
    function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PayinMethod();

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('ux_flow')) {
          obj['ux_flow'] = _PayinMethodUxFlow["default"].constructFromObject(data['ux_flow']);
        }
        if (data.hasOwnProperty('in_details')) {
          obj['in_details'] = _PayinMethodDetails["default"].constructFromObject(data['in_details']);
        }
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('state')) {
          obj['state'] = _PayinMethodState["default"].constructFromObject(data['state']);
        }
        if (data.hasOwnProperty('state_reason_details')) {
          obj['state_reason_details'] = _StateReasonDetails["default"].constructFromObject(data['state_reason_details']);
        }
        if (data.hasOwnProperty('out_details')) {
          obj['out_details'] = _ApiClient["default"].convertToType(data['out_details'], Object);
        }
        if (data.hasOwnProperty('instructions')) {
          obj['instructions'] = _ApiClient["default"].convertToType(data['instructions'], Object);
        }
        if (data.hasOwnProperty('errors')) {
          obj['errors'] = _ApiClient["default"].convertToType(data['errors'], { 'String': [_ValidationErrorDescription["default"]] });
        }
      }
      return obj;
    } }]);}();




/**
 * Describes how the payment should be requested from the sender.  Possible values: - `GHS::Mobile`: GHS mobile collections - `UGX::Mobile`: UGX mobile collections - `XOF::Mobile`: XOF mobile collections - `EUR::Bank`: EUR IBAN collections - `GBP::Bank`: GBP IBAN collections 
 * @member {String} type
 */
PayinMethod.prototype['type'] = undefined;

/**
 * @member {module:Model/PayinMethodUxFlow} ux_flow
 */
PayinMethod.prototype['ux_flow'] = undefined;

/**
 * @member {module:Model/PayinMethodDetails} in_details
 */
PayinMethod.prototype['in_details'] = undefined;

/**
 * @member {String} id
 */
PayinMethod.prototype['id'] = undefined;

/**
 * @member {module:Model/PayinMethodState} state
 */
PayinMethod.prototype['state'] = undefined;

/**
 * @member {module:Model/StateReasonDetails} state_reason_details
 */
PayinMethod.prototype['state_reason_details'] = undefined;

/**
 * This will contain the description on where to pay the funds. Please see the [Collections Details](https://docs.transferzero.com/docs/collection-details) in the API documentation on what to expect here.
 * @member {Object} out_details
 */
PayinMethod.prototype['out_details'] = undefined;

/**
 * This will contain the instructions on how to pay the funds. Please see the [Collections Details](https://docs.transferzero.com/docs/collection-details) in the API documentation on what to expect here.
 * @member {Object} instructions
 */
PayinMethod.prototype['instructions'] = undefined;

/**
 * The fields that have some problems and don't pass validation
 * @member {Object.<String, Array.<module:Model/ValidationErrorDescription>>} errors
 */
PayinMethod.prototype['errors'] = undefined;var _default = exports["default"] =






PayinMethod;
