// transferzero-sdk/dist/Model/PayinMethodDetailsNGNBank.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports["default"] = void 0;












var _ApiClient = _interopRequireDefault(require("../ApiClient"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };}function _typeof(o) {"@babel/helpers - typeof";return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) {return typeof o;} : function (o) {return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o;}, _typeof(o);}function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError("Cannot call a class as a function");}}function _defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ("value" in descriptor) descriptor.writable = true;Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);}}function _createClass(Constructor, protoProps, staticProps) {if (protoProps) _defineProperties(Constructor.prototype, protoProps);if (staticProps) _defineProperties(Constructor, staticProps);Object.defineProperty(Constructor, "prototype", { writable: false });return Constructor;}function _toPropertyKey(t) {var i = _toPrimitive(t, "string");return "symbol" == _typeof(i) ? i : i + "";}function _toPrimitive(t, r) {if ("object" != _typeof(t) || !t) return t;var e = t[Symbol.toPrimitive];if (void 0 !== e) {var i = e.call(t, r || "default");if ("object" != _typeof(i)) return i;throw new TypeError("@@toPrimitive must return a primitive value.");}return ("string" === r ? String : Number)(t);} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
 * The PayinMethodDetailsNGNBank model module.
 * @module Model/PayinMethodDetailsNGNBank
 * @version 1.35.1
 */var PayinMethodDetailsNGNBank = /*#__PURE__*/function () {/**
   * Constructs a new <code>PayinMethodDetailsNGNBank</code>.
   * &#x60;&#x60;&#x60;JSON   \&quot;details\&quot;: {     \&quot;redirect_url\&quot;: \&quot;http://redirect.back.to\&quot;   } &#x60;&#x60;&#x60;
   * @alias module:Model/PayinMethodDetailsNGNBank
   */function PayinMethodDetailsNGNBank() {_classCallCheck(this, PayinMethodDetailsNGNBank);PayinMethodDetailsNGNBank.initialize(this);} /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */return _createClass(PayinMethodDetailsNGNBank, null, [{ key: "initialize", value: function initialize(obj) {
    }

    /**
     * Constructs a <code>PayinMethodDetailsNGNBank</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:Model/PayinMethodDetailsNGNBank} obj Optional instance to populate.
     * @return {module:Model/PayinMethodDetailsNGNBank} The populated <code>PayinMethodDetailsNGNBank</code> instance.
     */ }, { key: "constructFromObject", value:
    function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PayinMethodDetailsNGNBank();

        if (data.hasOwnProperty('payment_method')) {
          obj['payment_method'] = _ApiClient["default"].convertToType(data['payment_method'], 'String');
        }
        if (data.hasOwnProperty('redirect_url')) {
          obj['redirect_url'] = _ApiClient["default"].convertToType(data['redirect_url'], 'String');
        }
      }
      return obj;
    } }]);}();




/**
 * The payment method which the sender will use to make the payments. Options are `bank`, `card` or you can leave empty to support both.
 * @member {String} payment_method
 */
PayinMethodDetailsNGNBank.prototype['payment_method'] = undefined;

/**
 * This is where the sender should be redirected back when the payment has been finished
 * @member {String} redirect_url
 */
PayinMethodDetailsNGNBank.prototype['redirect_url'] = undefined;var _default = exports["default"] =






PayinMethodDetailsNGNBank;
