// transferzero-sdk/dist/Model/PayoutMethodDetailsBRLBank.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports["default"] = void 0;












var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _PayoutMethodBankAccountTypeEnum = _interopRequireDefault(require("./PayoutMethodBankAccountTypeEnum"));
var _PayoutMethodPixKeyTypeEnum = _interopRequireDefault(require("./PayoutMethodPixKeyTypeEnum"));
var _PayoutMethodTransferReasonEnum = _interopRequireDefault(require("./PayoutMethodTransferReasonEnum"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };}function _typeof(o) {"@babel/helpers - typeof";return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) {return typeof o;} : function (o) {return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o;}, _typeof(o);}function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError("Cannot call a class as a function");}}function _defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ("value" in descriptor) descriptor.writable = true;Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);}}function _createClass(Constructor, protoProps, staticProps) {if (protoProps) _defineProperties(Constructor.prototype, protoProps);if (staticProps) _defineProperties(Constructor, staticProps);Object.defineProperty(Constructor, "prototype", { writable: false });return Constructor;}function _toPropertyKey(t) {var i = _toPrimitive(t, "string");return "symbol" == _typeof(i) ? i : i + "";}function _toPrimitive(t, r) {if ("object" != _typeof(t) || !t) return t;var e = t[Symbol.toPrimitive];if (void 0 !== e) {var i = e.call(t, r || "default");if ("object" != _typeof(i)) return i;throw new TypeError("@@toPrimitive must return a primitive value.");}return ("string" === r ? String : Number)(t);} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
 * The PayoutMethodDetailsBRLBank model module.
 * @module Model/PayoutMethodDetailsBRLBank
 * @version 1.35.1
 */var PayoutMethodDetailsBRLBank = /*#__PURE__*/function () {/**
   * Constructs a new <code>PayoutMethodDetailsBRLBank</code>.
   * PIX Payment: &#x60;&#x60;&#x60;JSON   \&quot;details\&quot;: {     \&quot;first_name\&quot;: \&quot;First\&quot;,     \&quot;last_name\&quot;: \&quot;Last\&quot;,     \&quot;city\&quot;: \&quot;Brasilia\&quot;,     \&quot;postal_code\&quot;: \&quot;70070\&quot;,     \&quot;phone_number\&quot;: \&quot;+552112345678\&quot;, // E.164 international format     \&quot;pix_key_type\&quot;: \&quot;email\&quot;,     \&quot;pix_key_value\&quot;: \&quot;person@example.com\&quot;,     \&quot;identity_card_id\&quot;: \&quot;01234567890\&quot;, // CPF or CNPJ     \&quot;transfer_reason\&quot;: \&quot;personal_account\&quot;   } &#x60;&#x60;&#x60;  TED Payment: &#x60;&#x60;&#x60;JSON   \&quot;details\&quot;: {     \&quot;first_name\&quot;: \&quot;First\&quot;,     \&quot;last_name\&quot;: \&quot;Last\&quot;,     \&quot;city\&quot;: \&quot;Brasilia\&quot;,     \&quot;postal_code\&quot;: \&quot;70070\&quot;,     \&quot;phone_number\&quot;: \&quot;+552112345678\&quot;, // E.164 international format     \&quot;bank_code\&quot;: \&quot;104\&quot;,     \&quot;branch_code\&quot;: \&quot;00001\&quot;,     \&quot;bank_account\&quot;: \&quot;0009795493\&quot;,     \&quot;bank_account_type\&quot;: \&quot;10\&quot;,     \&quot;identity_card_id\&quot;: \&quot;01234567890\&quot;, // CPF or CNPJ     \&quot;transfer_reason\&quot;: \&quot;personal_account\&quot;   } &#x60;&#x60;&#x60;  See [BRL Bank](https://docs.transferzero.com/docs/payout-details/#brlbank) documentation for the bank_code and transfer_reason lists
   * @alias module:Model/PayoutMethodDetailsBRLBank
   * @param firstName {String} 
   * @param lastName {String} 
   * @param city {String} 
   * @param postalCode {String} 
   * @param identityCardId {String} 
   * @param transferReason {module:Model/PayoutMethodTransferReasonEnum} 
   */function PayoutMethodDetailsBRLBank(firstName, lastName, city, postalCode, identityCardId, transferReason) {_classCallCheck(this, PayoutMethodDetailsBRLBank);PayoutMethodDetailsBRLBank.initialize(this, firstName, lastName, city, postalCode, identityCardId, transferReason);} /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */return _createClass(PayoutMethodDetailsBRLBank, null, [{ key: "initialize", value: function initialize(obj, firstName, lastName, city, postalCode, identityCardId, transferReason) {
      obj['first_name'] = firstName;
      obj['last_name'] = lastName;
      obj['city'] = city;
      obj['postal_code'] = postalCode;
      obj['identity_card_id'] = identityCardId;
      obj['transfer_reason'] = transferReason;
    }

    /**
     * Constructs a <code>PayoutMethodDetailsBRLBank</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:Model/PayoutMethodDetailsBRLBank} obj Optional instance to populate.
     * @return {module:Model/PayoutMethodDetailsBRLBank} The populated <code>PayoutMethodDetailsBRLBank</code> instance.
     */ }, { key: "constructFromObject", value:
    function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PayoutMethodDetailsBRLBank();

        if (data.hasOwnProperty('first_name')) {
          obj['first_name'] = _ApiClient["default"].convertToType(data['first_name'], 'String');
        }
        if (data.hasOwnProperty('last_name')) {
          obj['last_name'] = _ApiClient["default"].convertToType(data['last_name'], 'String');
        }
        if (data.hasOwnProperty('city')) {
          obj['city'] = _ApiClient["default"].convertToType(data['city'], 'String');
        }
        if (data.hasOwnProperty('postal_code')) {
          obj['postal_code'] = _ApiClient["default"].convertToType(data['postal_code'], 'String');
        }
        if (data.hasOwnProperty('phone_number')) {
          obj['phone_number'] = _ApiClient["default"].convertToType(data['phone_number'], 'String');
        }
        if (data.hasOwnProperty('bank_code')) {
          obj['bank_code'] = _ApiClient["default"].convertToType(data['bank_code'], 'String');
        }
        if (data.hasOwnProperty('branch_code')) {
          obj['branch_code'] = _ApiClient["default"].convertToType(data['branch_code'], 'String');
        }
        if (data.hasOwnProperty('bank_account')) {
          obj['bank_account'] = _ApiClient["default"].convertToType(data['bank_account'], 'String');
        }
        if (data.hasOwnProperty('bank_account_type')) {
          obj['bank_account_type'] = _PayoutMethodBankAccountTypeEnum["default"].constructFromObject(data['bank_account_type']);
        }
        if (data.hasOwnProperty('pix_key_type')) {
          obj['pix_key_type'] = _PayoutMethodPixKeyTypeEnum["default"].constructFromObject(data['pix_key_type']);
        }
        if (data.hasOwnProperty('pix_key_value')) {
          obj['pix_key_value'] = _ApiClient["default"].convertToType(data['pix_key_value'], 'String');
        }
        if (data.hasOwnProperty('identity_card_id')) {
          obj['identity_card_id'] = _ApiClient["default"].convertToType(data['identity_card_id'], 'String');
        }
        if (data.hasOwnProperty('transfer_reason')) {
          obj['transfer_reason'] = _PayoutMethodTransferReasonEnum["default"].constructFromObject(data['transfer_reason']);
        }
        if (data.hasOwnProperty('birth_date')) {
          obj['birth_date'] = _ApiClient["default"].convertToType(data['birth_date'], 'Date');
        }
      }
      return obj;
    } }]);}();




/**
 * @member {String} first_name
 */
PayoutMethodDetailsBRLBank.prototype['first_name'] = undefined;

/**
 * @member {String} last_name
 */
PayoutMethodDetailsBRLBank.prototype['last_name'] = undefined;

/**
 * @member {String} city
 */
PayoutMethodDetailsBRLBank.prototype['city'] = undefined;

/**
 * @member {String} postal_code
 */
PayoutMethodDetailsBRLBank.prototype['postal_code'] = undefined;

/**
 * @member {String} phone_number
 */
PayoutMethodDetailsBRLBank.prototype['phone_number'] = undefined;

/**
 * @member {String} bank_code
 */
PayoutMethodDetailsBRLBank.prototype['bank_code'] = undefined;

/**
 * @member {String} branch_code
 */
PayoutMethodDetailsBRLBank.prototype['branch_code'] = undefined;

/**
 * @member {String} bank_account
 */
PayoutMethodDetailsBRLBank.prototype['bank_account'] = undefined;

/**
 * @member {module:Model/PayoutMethodBankAccountTypeEnum} bank_account_type
 */
PayoutMethodDetailsBRLBank.prototype['bank_account_type'] = undefined;

/**
 * @member {module:Model/PayoutMethodPixKeyTypeEnum} pix_key_type
 */
PayoutMethodDetailsBRLBank.prototype['pix_key_type'] = undefined;

/**
 * @member {String} pix_key_value
 */
PayoutMethodDetailsBRLBank.prototype['pix_key_value'] = undefined;

/**
 * @member {String} identity_card_id
 */
PayoutMethodDetailsBRLBank.prototype['identity_card_id'] = undefined;

/**
 * @member {module:Model/PayoutMethodTransferReasonEnum} transfer_reason
 */
PayoutMethodDetailsBRLBank.prototype['transfer_reason'] = undefined;

/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
PayoutMethodDetailsBRLBank.prototype['birth_date'] = undefined;var _default = exports["default"] =






PayoutMethodDetailsBRLBank;
