// transferzero-sdk/dist/Model/PayoutMethodDetailsKESMobile.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports["default"] = void 0;












var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _PayoutMethodIdentityCardTypeEnum = _interopRequireDefault(require("./PayoutMethodIdentityCardTypeEnum"));
var _PayoutMethodMobileProviderEnum = _interopRequireDefault(require("./PayoutMethodMobileProviderEnum"));
var _PayoutMethodTransferReasonEnum = _interopRequireDefault(require("./PayoutMethodTransferReasonEnum"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };}function _typeof(o) {"@babel/helpers - typeof";return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) {return typeof o;} : function (o) {return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o;}, _typeof(o);}function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError("Cannot call a class as a function");}}function _defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ("value" in descriptor) descriptor.writable = true;Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);}}function _createClass(Constructor, protoProps, staticProps) {if (protoProps) _defineProperties(Constructor.prototype, protoProps);if (staticProps) _defineProperties(Constructor, staticProps);Object.defineProperty(Constructor, "prototype", { writable: false });return Constructor;}function _toPropertyKey(t) {var i = _toPrimitive(t, "string");return "symbol" == _typeof(i) ? i : i + "";}function _toPrimitive(t, r) {if ("object" != _typeof(t) || !t) return t;var e = t[Symbol.toPrimitive];if (void 0 !== e) {var i = e.call(t, r || "default");if ("object" != _typeof(i)) return i;throw new TypeError("@@toPrimitive must return a primitive value.");}return ("string" === r ? String : Number)(t);} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
 * The PayoutMethodDetailsKESMobile model module.
 * @module Model/PayoutMethodDetailsKESMobile
 * @version 1.35.1
 */var PayoutMethodDetailsKESMobile = /*#__PURE__*/function () {/**
   * Constructs a new <code>PayoutMethodDetailsKESMobile</code>.
   * &#x60;&#x60;&#x60;JSON   \&quot;details\&quot;: {     \&quot;first_name\&quot;: \&quot;First\&quot;,     \&quot;last_name\&quot;: \&quot;Last\&quot;,     \&quot;street\&quot;: \&quot;1 Linford Street\&quot;,     \&quot;city\&quot;: \&quot;Nairobi\&quot;,     \&quot;phone_number\&quot;: \&quot;+254123456789\&quot;, // E.164 international format     \&quot;identity_card_type\&quot;: \&quot;ID\&quot;, // refers to the recipient&#39;s ID details; Values: \&quot;PP\&quot;: Passport, \&quot;ID\&quot;: National ID or \&quot;O\&quot;: Other     \&quot;identity_card_id\&quot;: &#39;AB12345678&#39;, // refers to the recipient&#39;s ID details     \&quot;transfer_reason\&quot;: \&quot;personal_account\&quot;,     \&quot;mobile_provider\&quot;: \&quot;mpesa\&quot;,     \&quot;relationship_to_sender\&quot;: \&quot;Aunt\&quot; // Optional   } &#x60;&#x60;&#x60;  See [KES Mobile](https://docs.transferzero.com/docs/payout-details/#kesmobile) documentation for transfer_reason lists
   * @alias module:Model/PayoutMethodDetailsKESMobile
   * @param firstName {String} 
   * @param lastName {String} 
   * @param street {String} 
   * @param phoneNumber {String} 
   * @param mobileProvider {module:Model/PayoutMethodMobileProviderEnum} 
   * @param transferReason {module:Model/PayoutMethodTransferReasonEnum} 
   * @param identityCardType {module:Model/PayoutMethodIdentityCardTypeEnum} 
   * @param identityCardId {String} 
   */function PayoutMethodDetailsKESMobile(firstName, lastName, street, phoneNumber, mobileProvider, transferReason, identityCardType, identityCardId) {_classCallCheck(this, PayoutMethodDetailsKESMobile);PayoutMethodDetailsKESMobile.initialize(this, firstName, lastName, street, phoneNumber, mobileProvider, transferReason, identityCardType, identityCardId);} /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */return _createClass(PayoutMethodDetailsKESMobile, null, [{ key: "initialize", value: function initialize(obj, firstName, lastName, street, phoneNumber, mobileProvider, transferReason, identityCardType, identityCardId) {
      obj['first_name'] = firstName;
      obj['last_name'] = lastName;
      obj['street'] = street;
      obj['phone_number'] = phoneNumber;
      obj['mobile_provider'] = mobileProvider;
      obj['transfer_reason'] = transferReason;
      obj['identity_card_type'] = identityCardType;
      obj['identity_card_id'] = identityCardId;
    }

    /**
     * Constructs a <code>PayoutMethodDetailsKESMobile</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:Model/PayoutMethodDetailsKESMobile} obj Optional instance to populate.
     * @return {module:Model/PayoutMethodDetailsKESMobile} The populated <code>PayoutMethodDetailsKESMobile</code> instance.
     */ }, { key: "constructFromObject", value:
    function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PayoutMethodDetailsKESMobile();

        if (data.hasOwnProperty('first_name')) {
          obj['first_name'] = _ApiClient["default"].convertToType(data['first_name'], 'String');
        }
        if (data.hasOwnProperty('last_name')) {
          obj['last_name'] = _ApiClient["default"].convertToType(data['last_name'], 'String');
        }
        if (data.hasOwnProperty('street')) {
          obj['street'] = _ApiClient["default"].convertToType(data['street'], 'String');
        }
        if (data.hasOwnProperty('city')) {
          obj['city'] = _ApiClient["default"].convertToType(data['city'], 'String');
        }
        if (data.hasOwnProperty('phone_number')) {
          obj['phone_number'] = _ApiClient["default"].convertToType(data['phone_number'], 'String');
        }
        if (data.hasOwnProperty('mobile_provider')) {
          obj['mobile_provider'] = _PayoutMethodMobileProviderEnum["default"].constructFromObject(data['mobile_provider']);
        }
        if (data.hasOwnProperty('transfer_reason_code')) {
          obj['transfer_reason_code'] = _ApiClient["default"].convertToType(data['transfer_reason_code'], 'String');
        }
        if (data.hasOwnProperty('transfer_reason')) {
          obj['transfer_reason'] = _PayoutMethodTransferReasonEnum["default"].constructFromObject(data['transfer_reason']);
        }
        if (data.hasOwnProperty('identity_card_type')) {
          obj['identity_card_type'] = _PayoutMethodIdentityCardTypeEnum["default"].constructFromObject(data['identity_card_type']);
        }
        if (data.hasOwnProperty('identity_card_id')) {
          obj['identity_card_id'] = _ApiClient["default"].convertToType(data['identity_card_id'], 'String');
        }
        if (data.hasOwnProperty('relationship_to_sender')) {
          obj['relationship_to_sender'] = _ApiClient["default"].convertToType(data['relationship_to_sender'], 'String');
        }
        if (data.hasOwnProperty('birth_date')) {
          obj['birth_date'] = _ApiClient["default"].convertToType(data['birth_date'], 'Date');
        }
      }
      return obj;
    } }]);}();




/**
 * @member {String} first_name
 */
PayoutMethodDetailsKESMobile.prototype['first_name'] = undefined;

/**
 * @member {String} last_name
 */
PayoutMethodDetailsKESMobile.prototype['last_name'] = undefined;

/**
 * @member {String} street
 */
PayoutMethodDetailsKESMobile.prototype['street'] = undefined;

/**
 * @member {String} city
 */
PayoutMethodDetailsKESMobile.prototype['city'] = undefined;

/**
 * @member {String} phone_number
 */
PayoutMethodDetailsKESMobile.prototype['phone_number'] = undefined;

/**
 * @member {module:Model/PayoutMethodMobileProviderEnum} mobile_provider
 */
PayoutMethodDetailsKESMobile.prototype['mobile_provider'] = undefined;

/**
 * @member {String} transfer_reason_code
 */
PayoutMethodDetailsKESMobile.prototype['transfer_reason_code'] = undefined;

/**
 * @member {module:Model/PayoutMethodTransferReasonEnum} transfer_reason
 */
PayoutMethodDetailsKESMobile.prototype['transfer_reason'] = undefined;

/**
 * @member {module:Model/PayoutMethodIdentityCardTypeEnum} identity_card_type
 */
PayoutMethodDetailsKESMobile.prototype['identity_card_type'] = undefined;

/**
 * @member {String} identity_card_id
 */
PayoutMethodDetailsKESMobile.prototype['identity_card_id'] = undefined;

/**
 * @member {String} relationship_to_sender
 */
PayoutMethodDetailsKESMobile.prototype['relationship_to_sender'] = undefined;

/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
PayoutMethodDetailsKESMobile.prototype['birth_date'] = undefined;var _default = exports["default"] =






PayoutMethodDetailsKESMobile;
