// transferzero-sdk/dist/Model/PayoutMethodDetailsINRBank.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports["default"] = void 0;












var _ApiClient = _interopRequireDefault(require("../ApiClient"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };}function _typeof(o) {"@babel/helpers - typeof";return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) {return typeof o;} : function (o) {return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o;}, _typeof(o);}function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError("Cannot call a class as a function");}}function _defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ("value" in descriptor) descriptor.writable = true;Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);}}function _createClass(Constructor, protoProps, staticProps) {if (protoProps) _defineProperties(Constructor.prototype, protoProps);if (staticProps) _defineProperties(Constructor, staticProps);Object.defineProperty(Constructor, "prototype", { writable: false });return Constructor;}function _toPropertyKey(t) {var i = _toPrimitive(t, "string");return "symbol" == _typeof(i) ? i : i + "";}function _toPrimitive(t, r) {if ("object" != _typeof(t) || !t) return t;var e = t[Symbol.toPrimitive];if (void 0 !== e) {var i = e.call(t, r || "default");if ("object" != _typeof(i)) return i;throw new TypeError("@@toPrimitive must return a primitive value.");}return ("string" === r ? String : Number)(t);} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
 * The PayoutMethodDetailsINRBank model module.
 * @module Model/PayoutMethodDetailsINRBank
 * @version 1.35.1
 */var PayoutMethodDetailsINRBank = /*#__PURE__*/function () {/**
   * Constructs a new <code>PayoutMethodDetailsINRBank</code>.
   * &#x60;&#x60;&#x60;JSON \&quot;details\&quot;: {   \&quot;first_name\&quot;: \&quot;First\&quot;,   \&quot;last_name\&quot;: \&quot;Last\&quot;,   \&quot;city\&quot;: \&quot;New Delhi\&quot;,   \&quot;street\&quot;: \&quot;1, Akbar Road\&quot;,   \&quot;postal_code\&quot;: \&quot;110016\&quot;,   \&quot;phone_number\&quot;: \&quot;+919876543210\&quot;,   \&quot;bank_name\&quot;: \&quot;Bank of India\&quot;   \&quot;bank_account\&quot;: \&quot;1234567890\&quot;,   \&quot;ifsc_code\&quot;: \&quot;BKID0006005\&quot; } &#x60;&#x60;&#x60; See [INR Bank](https://docs.transferzero.com/docs/payout-details/#inrbank) documentation for the bank_name list
   * @alias module:Model/PayoutMethodDetailsINRBank
   * @param firstName {String} 
   * @param lastName {String} 
   * @param city {String} 
   * @param street {String} 
   * @param postalCode {String} 
   * @param phoneNumber {String} 
   * @param bankName {String} 
   * @param bankAccount {String} 
   * @param ifscCode {String} 
   */function PayoutMethodDetailsINRBank(firstName, lastName, city, street, postalCode, phoneNumber, bankName, bankAccount, ifscCode) {_classCallCheck(this, PayoutMethodDetailsINRBank);PayoutMethodDetailsINRBank.initialize(this, firstName, lastName, city, street, postalCode, phoneNumber, bankName, bankAccount, ifscCode);} /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */return _createClass(PayoutMethodDetailsINRBank, null, [{ key: "initialize", value: function initialize(obj, firstName, lastName, city, street, postalCode, phoneNumber, bankName, bankAccount, ifscCode) {
      obj['first_name'] = firstName;
      obj['last_name'] = lastName;
      obj['city'] = city;
      obj['street'] = street;
      obj['postal_code'] = postalCode;
      obj['phone_number'] = phoneNumber;
      obj['bank_name'] = bankName;
      obj['bank_account'] = bankAccount;
      obj['ifsc_code'] = ifscCode;
    }

    /**
     * Constructs a <code>PayoutMethodDetailsINRBank</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:Model/PayoutMethodDetailsINRBank} obj Optional instance to populate.
     * @return {module:Model/PayoutMethodDetailsINRBank} The populated <code>PayoutMethodDetailsINRBank</code> instance.
     */ }, { key: "constructFromObject", value:
    function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PayoutMethodDetailsINRBank();

        if (data.hasOwnProperty('first_name')) {
          obj['first_name'] = _ApiClient["default"].convertToType(data['first_name'], 'String');
        }
        if (data.hasOwnProperty('last_name')) {
          obj['last_name'] = _ApiClient["default"].convertToType(data['last_name'], 'String');
        }
        if (data.hasOwnProperty('city')) {
          obj['city'] = _ApiClient["default"].convertToType(data['city'], 'String');
        }
        if (data.hasOwnProperty('street')) {
          obj['street'] = _ApiClient["default"].convertToType(data['street'], 'String');
        }
        if (data.hasOwnProperty('postal_code')) {
          obj['postal_code'] = _ApiClient["default"].convertToType(data['postal_code'], 'String');
        }
        if (data.hasOwnProperty('phone_number')) {
          obj['phone_number'] = _ApiClient["default"].convertToType(data['phone_number'], 'String');
        }
        if (data.hasOwnProperty('bank_name')) {
          obj['bank_name'] = _ApiClient["default"].convertToType(data['bank_name'], 'String');
        }
        if (data.hasOwnProperty('bank_account')) {
          obj['bank_account'] = _ApiClient["default"].convertToType(data['bank_account'], 'String');
        }
        if (data.hasOwnProperty('ifsc_code')) {
          obj['ifsc_code'] = _ApiClient["default"].convertToType(data['ifsc_code'], 'String');
        }
        if (data.hasOwnProperty('birth_date')) {
          obj['birth_date'] = _ApiClient["default"].convertToType(data['birth_date'], 'Date');
        }
      }
      return obj;
    } }]);}();




/**
 * @member {String} first_name
 */
PayoutMethodDetailsINRBank.prototype['first_name'] = undefined;

/**
 * @member {String} last_name
 */
PayoutMethodDetailsINRBank.prototype['last_name'] = undefined;

/**
 * @member {String} city
 */
PayoutMethodDetailsINRBank.prototype['city'] = undefined;

/**
 * @member {String} street
 */
PayoutMethodDetailsINRBank.prototype['street'] = undefined;

/**
 * @member {String} postal_code
 */
PayoutMethodDetailsINRBank.prototype['postal_code'] = undefined;

/**
 * @member {String} phone_number
 */
PayoutMethodDetailsINRBank.prototype['phone_number'] = undefined;

/**
 * @member {String} bank_name
 */
PayoutMethodDetailsINRBank.prototype['bank_name'] = undefined;

/**
 * @member {String} bank_account
 */
PayoutMethodDetailsINRBank.prototype['bank_account'] = undefined;

/**
 * @member {String} ifsc_code
 */
PayoutMethodDetailsINRBank.prototype['ifsc_code'] = undefined;

/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
PayoutMethodDetailsINRBank.prototype['birth_date'] = undefined;var _default = exports["default"] =






PayoutMethodDetailsINRBank;
