// transferzero-sdk/dist/Model/PayoutMethodDetails.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports["default"] = void 0;












var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _PayoutMethodBankAccountTypeEnum = _interopRequireDefault(require("./PayoutMethodBankAccountTypeEnum"));
var _PayoutMethodCashProviderEnum = _interopRequireDefault(require("./PayoutMethodCashProviderEnum"));
var _PayoutMethodCountryEnum = _interopRequireDefault(require("./PayoutMethodCountryEnum"));
var _PayoutMethodDetailsBRLBank = _interopRequireDefault(require("./PayoutMethodDetailsBRLBank"));
var _PayoutMethodDetailsBTC = _interopRequireDefault(require("./PayoutMethodDetailsBTC"));
var _PayoutMethodDetailsBWPBank = _interopRequireDefault(require("./PayoutMethodDetailsBWPBank"));
var _PayoutMethodDetailsBalance = _interopRequireDefault(require("./PayoutMethodDetailsBalance"));
var _PayoutMethodDetailsCADBank = _interopRequireDefault(require("./PayoutMethodDetailsCADBank"));
var _PayoutMethodDetailsEGPBank = _interopRequireDefault(require("./PayoutMethodDetailsEGPBank"));
var _PayoutMethodDetailsGBPBank = _interopRequireDefault(require("./PayoutMethodDetailsGBPBank"));
var _PayoutMethodDetailsGHSBank = _interopRequireDefault(require("./PayoutMethodDetailsGHSBank"));
var _PayoutMethodDetailsGHSCash = _interopRequireDefault(require("./PayoutMethodDetailsGHSCash"));
var _PayoutMethodDetailsGNFMobile = _interopRequireDefault(require("./PayoutMethodDetailsGNFMobile"));
var _PayoutMethodDetailsIBAN = _interopRequireDefault(require("./PayoutMethodDetailsIBAN"));
var _PayoutMethodDetailsINRBank = _interopRequireDefault(require("./PayoutMethodDetailsINRBank"));
var _PayoutMethodDetailsKESBank = _interopRequireDefault(require("./PayoutMethodDetailsKESBank"));
var _PayoutMethodDetailsKESMobile = _interopRequireDefault(require("./PayoutMethodDetailsKESMobile"));
var _PayoutMethodDetailsMADCash = _interopRequireDefault(require("./PayoutMethodDetailsMADCash"));
var _PayoutMethodDetailsMobile = _interopRequireDefault(require("./PayoutMethodDetailsMobile"));
var _PayoutMethodDetailsNGNBank = _interopRequireDefault(require("./PayoutMethodDetailsNGNBank"));
var _PayoutMethodDetailsNZDBank = _interopRequireDefault(require("./PayoutMethodDetailsNZDBank"));
var _PayoutMethodDetailsUGXBank = _interopRequireDefault(require("./PayoutMethodDetailsUGXBank"));
var _PayoutMethodDetailsUSDBank = _interopRequireDefault(require("./PayoutMethodDetailsUSDBank"));
var _PayoutMethodDetailsUSDCash = _interopRequireDefault(require("./PayoutMethodDetailsUSDCash"));
var _PayoutMethodDetailsXAFBank = _interopRequireDefault(require("./PayoutMethodDetailsXAFBank"));
var _PayoutMethodDetailsXAFMobile = _interopRequireDefault(require("./PayoutMethodDetailsXAFMobile"));
var _PayoutMethodDetailsXOFBank = _interopRequireDefault(require("./PayoutMethodDetailsXOFBank"));
var _PayoutMethodDetailsXOFCash = _interopRequireDefault(require("./PayoutMethodDetailsXOFCash"));
var _PayoutMethodDetailsXOFMobile = _interopRequireDefault(require("./PayoutMethodDetailsXOFMobile"));
var _PayoutMethodDetailsZARBank = _interopRequireDefault(require("./PayoutMethodDetailsZARBank"));
var _PayoutMethodDetailsZMWBank = _interopRequireDefault(require("./PayoutMethodDetailsZMWBank"));
var _PayoutMethodGenderEnum = _interopRequireDefault(require("./PayoutMethodGenderEnum"));
var _PayoutMethodIdentityCardTypeEnum = _interopRequireDefault(require("./PayoutMethodIdentityCardTypeEnum"));
var _PayoutMethodLegalEntityTypeEnum = _interopRequireDefault(require("./PayoutMethodLegalEntityTypeEnum"));
var _PayoutMethodMobileProviderEnum = _interopRequireDefault(require("./PayoutMethodMobileProviderEnum"));
var _PayoutMethodNatureOfBusinessEnum = _interopRequireDefault(require("./PayoutMethodNatureOfBusinessEnum"));
var _PayoutMethodPixKeyTypeEnum = _interopRequireDefault(require("./PayoutMethodPixKeyTypeEnum"));
var _PayoutMethodTransferReasonEnum = _interopRequireDefault(require("./PayoutMethodTransferReasonEnum"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };}function _typeof(o) {"@babel/helpers - typeof";return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) {return typeof o;} : function (o) {return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o;}, _typeof(o);}function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError("Cannot call a class as a function");}}function _defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ("value" in descriptor) descriptor.writable = true;Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);}}function _createClass(Constructor, protoProps, staticProps) {if (protoProps) _defineProperties(Constructor.prototype, protoProps);if (staticProps) _defineProperties(Constructor, staticProps);Object.defineProperty(Constructor, "prototype", { writable: false });return Constructor;}function _toPropertyKey(t) {var i = _toPrimitive(t, "string");return "symbol" == _typeof(i) ? i : i + "";}function _toPrimitive(t, r) {if ("object" != _typeof(t) || !t) return t;var e = t[Symbol.toPrimitive];if (void 0 !== e) {var i = e.call(t, r || "default");if ("object" != _typeof(i)) return i;throw new TypeError("@@toPrimitive must return a primitive value.");}return ("string" === r ? String : Number)(t);} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
 * The PayoutMethodDetails model module.
 * @module Model/PayoutMethodDetails
 * @version 1.35.1
 */var PayoutMethodDetails = /*#__PURE__*/function () {/**
   * Constructs a new <code>PayoutMethodDetails</code>.
   * @alias module:Model/PayoutMethodDetails
   * @implements module:Model/PayoutMethodDetailsNGNBank
   * @implements module:Model/PayoutMethodDetailsMobile
   * @implements module:Model/PayoutMethodDetailsXOFMobile
   * @implements module:Model/PayoutMethodDetailsXOFBank
   * @implements module:Model/PayoutMethodDetailsXOFCash
   * @implements module:Model/PayoutMethodDetailsGHSBank
   * @implements module:Model/PayoutMethodDetailsGHSCash
   * @implements module:Model/PayoutMethodDetailsGBPBank
   * @implements module:Model/PayoutMethodDetailsIBAN
   * @implements module:Model/PayoutMethodDetailsMADCash
   * @implements module:Model/PayoutMethodDetailsBalance
   * @implements module:Model/PayoutMethodDetailsBTC
   * @implements module:Model/PayoutMethodDetailsZARBank
   * @implements module:Model/PayoutMethodDetailsUSDBank
   * @implements module:Model/PayoutMethodDetailsUSDCash
   * @implements module:Model/PayoutMethodDetailsKESMobile
   * @implements module:Model/PayoutMethodDetailsKESBank
   * @implements module:Model/PayoutMethodDetailsXAFMobile
   * @implements module:Model/PayoutMethodDetailsXAFBank
   * @implements module:Model/PayoutMethodDetailsGNFMobile
   * @implements module:Model/PayoutMethodDetailsBRLBank
   * @implements module:Model/PayoutMethodDetailsNZDBank
   * @implements module:Model/PayoutMethodDetailsINRBank
   * @implements module:Model/PayoutMethodDetailsBWPBank
   * @implements module:Model/PayoutMethodDetailsZMWBank
   * @implements module:Model/PayoutMethodDetailsCADBank
   * @implements module:Model/PayoutMethodDetailsUGXBank
   * @implements module:Model/PayoutMethodDetailsEGPBank
   * @param firstName {String} 
   * @param lastName {String} 
   * @param bankCode {String} 
   * @param bankAccount {String} 
   * @param phoneNumber {String} 
   * @param mobileProvider {module:Model/PayoutMethodMobileProviderEnum} 
   * @param country {module:Model/PayoutMethodCountryEnum} 
   * @param transferReason {module:Model/PayoutMethodTransferReasonEnum} 
   * @param iban {String} 
   * @param bankName {String} 
   * @param cashProvider {module:Model/PayoutMethodCashProviderEnum} 
   * @param senderIdentityCardType {module:Model/PayoutMethodIdentityCardTypeEnum} 
   * @param senderIdentityCardId {String} 
   * @param identityCardType {module:Model/PayoutMethodIdentityCardTypeEnum} 
   * @param identityCardId {String} 
   * @param name {String} 
   * @param address {String} 
   * @param street {String} 
   * @param postalCode {String} 
   * @param city {String} 
   * @param branchCode {String} 
   * @param ifscCode {String} 
   */function PayoutMethodDetails(firstName, lastName, bankCode, bankAccount, phoneNumber, mobileProvider, country, transferReason, iban, bankName, cashProvider, senderIdentityCardType, senderIdentityCardId, identityCardType, identityCardId, name, address, street, postalCode, city, branchCode, ifscCode) {_classCallCheck(this, PayoutMethodDetails);_PayoutMethodDetailsNGNBank["default"].initialize(this, firstName, lastName, bankCode, bankAccount);_PayoutMethodDetailsMobile["default"].initialize(this, firstName, lastName, phoneNumber);_PayoutMethodDetailsXOFMobile["default"].initialize(this, firstName, lastName, phoneNumber, mobileProvider);_PayoutMethodDetailsXOFBank["default"].initialize(this, firstName, lastName, iban);_PayoutMethodDetailsXOFCash["default"].initialize(this, firstName, lastName, phoneNumber, cashProvider);_PayoutMethodDetailsGHSBank["default"].initialize(this, firstName, lastName, bankCode, bankAccount);_PayoutMethodDetailsGHSCash["default"].initialize(this, firstName, lastName, phoneNumber);_PayoutMethodDetailsGBPBank["default"].initialize(this, firstName, lastName);_PayoutMethodDetailsIBAN["default"].initialize(this, firstName, lastName, iban);_PayoutMethodDetailsMADCash["default"].initialize(this, firstName, lastName, phoneNumber, senderIdentityCardType, senderIdentityCardId);_PayoutMethodDetailsBalance["default"].initialize(this);_PayoutMethodDetailsBTC["default"].initialize(this, firstName, lastName, name, address);_PayoutMethodDetailsZARBank["default"].initialize(this, firstName, lastName, bankAccount, phoneNumber);_PayoutMethodDetailsUSDBank["default"].initialize(this, firstName, lastName, country);_PayoutMethodDetailsUSDCash["default"].initialize(this, firstName, lastName, phoneNumber, country);_PayoutMethodDetailsKESMobile["default"].initialize(this, firstName, lastName, street, phoneNumber, mobileProvider, transferReason, identityCardType, identityCardId);_PayoutMethodDetailsKESBank["default"].initialize(this, firstName, lastName, street, bankCode, bankAccount, transferReason, identityCardType, identityCardId);_PayoutMethodDetailsXAFMobile["default"].initialize(this, firstName, lastName, phoneNumber, mobileProvider);_PayoutMethodDetailsXAFBank["default"].initialize(this, firstName, lastName, iban);_PayoutMethodDetailsGNFMobile["default"].initialize(this, firstName, lastName, phoneNumber, mobileProvider);_PayoutMethodDetailsBRLBank["default"].initialize(this, firstName, lastName, city, postalCode, identityCardId, transferReason);_PayoutMethodDetailsNZDBank["default"].initialize(this, firstName, lastName, bankAccount, bankName);_PayoutMethodDetailsINRBank["default"].initialize(this, firstName, lastName, city, street, postalCode, phoneNumber, bankName, bankAccount, ifscCode);_PayoutMethodDetailsBWPBank["default"].initialize(this, firstName, lastName, bankName, bankAccount, branchCode);_PayoutMethodDetailsZMWBank["default"].initialize(this, firstName, lastName, bankAccount, branchCode);_PayoutMethodDetailsCADBank["default"].initialize(this, firstName, lastName, bankAccount, bankCode, branchCode);_PayoutMethodDetailsUGXBank["default"].initialize(this, firstName, lastName, street, city, postalCode, identityCardId, bankAccount, branchCode, transferReason);_PayoutMethodDetailsEGPBank["default"].initialize(this, firstName, lastName, street, bankAccount, bankCode, transferReason);PayoutMethodDetails.initialize(this, firstName, lastName, bankCode, bankAccount, phoneNumber, mobileProvider, country, transferReason, iban, bankName, cashProvider, senderIdentityCardType, senderIdentityCardId, identityCardType, identityCardId, name, address, street, postalCode, city, branchCode, ifscCode);} /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */return _createClass(PayoutMethodDetails, null, [{ key: "initialize", value: function initialize(obj, firstName, lastName, bankCode, bankAccount, phoneNumber, mobileProvider, country, transferReason, iban, bankName, cashProvider, senderIdentityCardType, senderIdentityCardId, identityCardType, identityCardId, name, address, street, postalCode, city, branchCode, ifscCode) {
      obj['first_name'] = firstName;
      obj['last_name'] = lastName;
      obj['bank_code'] = bankCode;
      obj['bank_account'] = bankAccount;
      obj['phone_number'] = phoneNumber;
      obj['mobile_provider'] = mobileProvider;
      obj['country'] = country;
      obj['transfer_reason'] = transferReason;
      obj['iban'] = iban;
      obj['bank_name'] = bankName;
      obj['cash_provider'] = cashProvider;
      obj['sender_identity_card_type'] = senderIdentityCardType;
      obj['sender_identity_card_id'] = senderIdentityCardId;
      obj['identity_card_type'] = identityCardType;
      obj['identity_card_id'] = identityCardId;
      obj['name'] = name;
      obj['address'] = address;
      obj['street'] = street;
      obj['postal_code'] = postalCode;
      obj['city'] = city;
      obj['branch_code'] = branchCode;
      obj['ifsc_code'] = ifscCode;
    }

    /**
     * Constructs a <code>PayoutMethodDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:Model/PayoutMethodDetails} obj Optional instance to populate.
     * @return {module:Model/PayoutMethodDetails} The populated <code>PayoutMethodDetails</code> instance.
     */ }, { key: "constructFromObject", value:
    function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PayoutMethodDetails();
        _PayoutMethodDetailsNGNBank["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsMobile["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsXOFMobile["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsXOFBank["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsXOFCash["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsGHSBank["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsGHSCash["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsGBPBank["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsIBAN["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsMADCash["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsBalance["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsBTC["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsZARBank["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsUSDBank["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsUSDCash["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsKESMobile["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsKESBank["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsXAFMobile["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsXAFBank["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsGNFMobile["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsBRLBank["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsNZDBank["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsINRBank["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsBWPBank["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsZMWBank["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsCADBank["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsUGXBank["default"].constructFromObject(data, obj);
        _PayoutMethodDetailsEGPBank["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('first_name')) {
          obj['first_name'] = _ApiClient["default"].convertToType(data['first_name'], 'String');
        }
        if (data.hasOwnProperty('last_name')) {
          obj['last_name'] = _ApiClient["default"].convertToType(data['last_name'], 'String');
        }
        if (data.hasOwnProperty('bank_code')) {
          obj['bank_code'] = _ApiClient["default"].convertToType(data['bank_code'], 'String');
        }
        if (data.hasOwnProperty('bank_account')) {
          obj['bank_account'] = _ApiClient["default"].convertToType(data['bank_account'], 'String');
        }
        if (data.hasOwnProperty('bank_account_type')) {
          obj['bank_account_type'] = _PayoutMethodBankAccountTypeEnum["default"].constructFromObject(data['bank_account_type']);
        }
        if (data.hasOwnProperty('birth_date')) {
          obj['birth_date'] = _ApiClient["default"].convertToType(data['birth_date'], 'Date');
        }
        if (data.hasOwnProperty('phone_number')) {
          obj['phone_number'] = _ApiClient["default"].convertToType(data['phone_number'], 'String');
        }
        if (data.hasOwnProperty('mobile_provider')) {
          obj['mobile_provider'] = _PayoutMethodMobileProviderEnum["default"].constructFromObject(data['mobile_provider']);
        }
        if (data.hasOwnProperty('country')) {
          obj['country'] = _PayoutMethodCountryEnum["default"].constructFromObject(data['country']);
        }
        if (data.hasOwnProperty('transfer_reason')) {
          obj['transfer_reason'] = _PayoutMethodTransferReasonEnum["default"].constructFromObject(data['transfer_reason']);
        }
        if (data.hasOwnProperty('iban')) {
          obj['iban'] = _ApiClient["default"].convertToType(data['iban'], 'String');
        }
        if (data.hasOwnProperty('bank_name')) {
          obj['bank_name'] = _ApiClient["default"].convertToType(data['bank_name'], 'String');
        }
        if (data.hasOwnProperty('bank_country')) {
          obj['bank_country'] = _ApiClient["default"].convertToType(data['bank_country'], 'String');
        }
        if (data.hasOwnProperty('cash_provider')) {
          obj['cash_provider'] = _PayoutMethodCashProviderEnum["default"].constructFromObject(data['cash_provider']);
        }
        if (data.hasOwnProperty('sort_code')) {
          obj['sort_code'] = _ApiClient["default"].convertToType(data['sort_code'], 'String');
        }
        if (data.hasOwnProperty('bic')) {
          obj['bic'] = _ApiClient["default"].convertToType(data['bic'], 'String');
        }
        if (data.hasOwnProperty('narration')) {
          obj['narration'] = _ApiClient["default"].convertToType(data['narration'], 'String');
        }
        if (data.hasOwnProperty('sender_identity_card_type')) {
          obj['sender_identity_card_type'] = _PayoutMethodIdentityCardTypeEnum["default"].constructFromObject(data['sender_identity_card_type']);
        }
        if (data.hasOwnProperty('sender_identity_card_id')) {
          obj['sender_identity_card_id'] = _ApiClient["default"].convertToType(data['sender_identity_card_id'], 'String');
        }
        if (data.hasOwnProperty('sender_city_of_birth')) {
          obj['sender_city_of_birth'] = _ApiClient["default"].convertToType(data['sender_city_of_birth'], 'String');
        }
        if (data.hasOwnProperty('sender_country_of_birth')) {
          obj['sender_country_of_birth'] = _ApiClient["default"].convertToType(data['sender_country_of_birth'], 'String');
        }
        if (data.hasOwnProperty('sender_gender')) {
          obj['sender_gender'] = _PayoutMethodGenderEnum["default"].constructFromObject(data['sender_gender']);
        }
        if (data.hasOwnProperty('reason')) {
          obj['reason'] = _ApiClient["default"].convertToType(data['reason'], 'String');
        }
        if (data.hasOwnProperty('identity_card_type')) {
          obj['identity_card_type'] = _PayoutMethodIdentityCardTypeEnum["default"].constructFromObject(data['identity_card_type']);
        }
        if (data.hasOwnProperty('identity_card_id')) {
          obj['identity_card_id'] = _ApiClient["default"].convertToType(data['identity_card_id'], 'String');
        }
        if (data.hasOwnProperty('reference')) {
          obj['reference'] = _ApiClient["default"].convertToType(data['reference'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('address')) {
          obj['address'] = _ApiClient["default"].convertToType(data['address'], 'String');
        }
        if (data.hasOwnProperty('street')) {
          obj['street'] = _ApiClient["default"].convertToType(data['street'], 'String');
        }
        if (data.hasOwnProperty('postal_code')) {
          obj['postal_code'] = _ApiClient["default"].convertToType(data['postal_code'], 'String');
        }
        if (data.hasOwnProperty('city')) {
          obj['city'] = _ApiClient["default"].convertToType(data['city'], 'String');
        }
        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], 'String');
        }
        if (data.hasOwnProperty('branch_code')) {
          obj['branch_code'] = _ApiClient["default"].convertToType(data['branch_code'], 'String');
        }
        if (data.hasOwnProperty('transfer_reason_code')) {
          obj['transfer_reason_code'] = _ApiClient["default"].convertToType(data['transfer_reason_code'], 'String');
        }
        if (data.hasOwnProperty('contact_first_name')) {
          obj['contact_first_name'] = _ApiClient["default"].convertToType(data['contact_first_name'], 'String');
        }
        if (data.hasOwnProperty('contact_last_name')) {
          obj['contact_last_name'] = _ApiClient["default"].convertToType(data['contact_last_name'], 'String');
        }
        if (data.hasOwnProperty('registration_number')) {
          obj['registration_number'] = _ApiClient["default"].convertToType(data['registration_number'], 'String');
        }
        if (data.hasOwnProperty('nature_of_business')) {
          obj['nature_of_business'] = _PayoutMethodNatureOfBusinessEnum["default"].constructFromObject(data['nature_of_business']);
        }
        if (data.hasOwnProperty('legal_entity_type')) {
          obj['legal_entity_type'] = _PayoutMethodLegalEntityTypeEnum["default"].constructFromObject(data['legal_entity_type']);
        }
        if (data.hasOwnProperty('middle_name')) {
          obj['middle_name'] = _ApiClient["default"].convertToType(data['middle_name'], 'String');
        }
        if (data.hasOwnProperty('routing_number')) {
          obj['routing_number'] = _ApiClient["default"].convertToType(data['routing_number'], 'String');
        }
        if (data.hasOwnProperty('swift_code')) {
          obj['swift_code'] = _ApiClient["default"].convertToType(data['swift_code'], 'String');
        }
        if (data.hasOwnProperty('relationship_to_sender')) {
          obj['relationship_to_sender'] = _ApiClient["default"].convertToType(data['relationship_to_sender'], 'String');
        }
        if (data.hasOwnProperty('pix_key_type')) {
          obj['pix_key_type'] = _PayoutMethodPixKeyTypeEnum["default"].constructFromObject(data['pix_key_type']);
        }
        if (data.hasOwnProperty('pix_key_value')) {
          obj['pix_key_value'] = _ApiClient["default"].convertToType(data['pix_key_value'], 'String');
        }
        if (data.hasOwnProperty('ifsc_code')) {
          obj['ifsc_code'] = _ApiClient["default"].convertToType(data['ifsc_code'], 'String');
        }
      }
      return obj;
    } }]);}();




/**
 * @member {String} first_name
 */
PayoutMethodDetails.prototype['first_name'] = undefined;

/**
 * @member {String} last_name
 */
PayoutMethodDetails.prototype['last_name'] = undefined;

/**
 * @member {String} bank_code
 */
PayoutMethodDetails.prototype['bank_code'] = undefined;

/**
 * @member {String} bank_account
 */
PayoutMethodDetails.prototype['bank_account'] = undefined;

/**
 * @member {module:Model/PayoutMethodBankAccountTypeEnum} bank_account_type
 */
PayoutMethodDetails.prototype['bank_account_type'] = undefined;

/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
PayoutMethodDetails.prototype['birth_date'] = undefined;

/**
 * @member {String} phone_number
 */
PayoutMethodDetails.prototype['phone_number'] = undefined;

/**
 * @member {module:Model/PayoutMethodMobileProviderEnum} mobile_provider
 */
PayoutMethodDetails.prototype['mobile_provider'] = undefined;

/**
 * @member {module:Model/PayoutMethodCountryEnum} country
 */
PayoutMethodDetails.prototype['country'] = undefined;

/**
 * @member {module:Model/PayoutMethodTransferReasonEnum} transfer_reason
 */
PayoutMethodDetails.prototype['transfer_reason'] = undefined;

/**
 * @member {String} iban
 */
PayoutMethodDetails.prototype['iban'] = undefined;

/**
 * @member {String} bank_name
 */
PayoutMethodDetails.prototype['bank_name'] = undefined;

/**
 * @member {String} bank_country
 */
PayoutMethodDetails.prototype['bank_country'] = undefined;

/**
 * @member {module:Model/PayoutMethodCashProviderEnum} cash_provider
 */
PayoutMethodDetails.prototype['cash_provider'] = undefined;

/**
 * @member {String} sort_code
 */
PayoutMethodDetails.prototype['sort_code'] = undefined;

/**
 * @member {String} bic
 */
PayoutMethodDetails.prototype['bic'] = undefined;

/**
 * @member {String} narration
 */
PayoutMethodDetails.prototype['narration'] = undefined;

/**
 * @member {module:Model/PayoutMethodIdentityCardTypeEnum} sender_identity_card_type
 */
PayoutMethodDetails.prototype['sender_identity_card_type'] = undefined;

/**
 * @member {String} sender_identity_card_id
 */
PayoutMethodDetails.prototype['sender_identity_card_id'] = undefined;

/**
 * @member {String} sender_city_of_birth
 */
PayoutMethodDetails.prototype['sender_city_of_birth'] = undefined;

/**
 * @member {String} sender_country_of_birth
 */
PayoutMethodDetails.prototype['sender_country_of_birth'] = undefined;

/**
 * @member {module:Model/PayoutMethodGenderEnum} sender_gender
 */
PayoutMethodDetails.prototype['sender_gender'] = undefined;

/**
 * @member {String} reason
 */
PayoutMethodDetails.prototype['reason'] = undefined;

/**
 * @member {module:Model/PayoutMethodIdentityCardTypeEnum} identity_card_type
 */
PayoutMethodDetails.prototype['identity_card_type'] = undefined;

/**
 * @member {String} identity_card_id
 */
PayoutMethodDetails.prototype['identity_card_id'] = undefined;

/**
 * @member {String} reference
 */
PayoutMethodDetails.prototype['reference'] = undefined;

/**
 * @member {String} name
 */
PayoutMethodDetails.prototype['name'] = undefined;

/**
 * @member {String} address
 */
PayoutMethodDetails.prototype['address'] = undefined;

/**
 * @member {String} street
 */
PayoutMethodDetails.prototype['street'] = undefined;

/**
 * @member {String} postal_code
 */
PayoutMethodDetails.prototype['postal_code'] = undefined;

/**
 * @member {String} city
 */
PayoutMethodDetails.prototype['city'] = undefined;

/**
 * @member {String} email
 */
PayoutMethodDetails.prototype['email'] = undefined;

/**
 * @member {String} branch_code
 */
PayoutMethodDetails.prototype['branch_code'] = undefined;

/**
 * @member {String} transfer_reason_code
 */
PayoutMethodDetails.prototype['transfer_reason_code'] = undefined;

/**
 * @member {String} contact_first_name
 */
PayoutMethodDetails.prototype['contact_first_name'] = undefined;

/**
 * @member {String} contact_last_name
 */
PayoutMethodDetails.prototype['contact_last_name'] = undefined;

/**
 * @member {String} registration_number
 */
PayoutMethodDetails.prototype['registration_number'] = undefined;

/**
 * @member {module:Model/PayoutMethodNatureOfBusinessEnum} nature_of_business
 */
PayoutMethodDetails.prototype['nature_of_business'] = undefined;

/**
 * @member {module:Model/PayoutMethodLegalEntityTypeEnum} legal_entity_type
 */
PayoutMethodDetails.prototype['legal_entity_type'] = undefined;

/**
 * @member {String} middle_name
 */
PayoutMethodDetails.prototype['middle_name'] = undefined;

/**
 * @member {String} routing_number
 */
PayoutMethodDetails.prototype['routing_number'] = undefined;

/**
 * @member {String} swift_code
 */
PayoutMethodDetails.prototype['swift_code'] = undefined;

/**
 * @member {String} relationship_to_sender
 */
PayoutMethodDetails.prototype['relationship_to_sender'] = undefined;

/**
 * @member {module:Model/PayoutMethodPixKeyTypeEnum} pix_key_type
 */
PayoutMethodDetails.prototype['pix_key_type'] = undefined;

/**
 * @member {String} pix_key_value
 */
PayoutMethodDetails.prototype['pix_key_value'] = undefined;

/**
 * @member {String} ifsc_code
 */
PayoutMethodDetails.prototype['ifsc_code'] = undefined;


// Implement PayoutMethodDetailsNGNBank interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsNGNBank["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsNGNBank["default"].prototype['last_name'] = undefined;
/**
 * @member {String} bank_code
 */
_PayoutMethodDetailsNGNBank["default"].prototype['bank_code'] = undefined;
/**
 * @member {String} bank_account
 */
_PayoutMethodDetailsNGNBank["default"].prototype['bank_account'] = undefined;
/**
 * @member {module:Model/PayoutMethodBankAccountTypeEnum} bank_account_type
 */
_PayoutMethodDetailsNGNBank["default"].prototype['bank_account_type'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsNGNBank["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsMobile interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsMobile["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsMobile["default"].prototype['last_name'] = undefined;
/**
 * @member {String} phone_number
 */
_PayoutMethodDetailsMobile["default"].prototype['phone_number'] = undefined;
/**
 * @member {module:Model/PayoutMethodMobileProviderEnum} mobile_provider
 */
_PayoutMethodDetailsMobile["default"].prototype['mobile_provider'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsMobile["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsXOFMobile interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsXOFMobile["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsXOFMobile["default"].prototype['last_name'] = undefined;
/**
 * @member {String} phone_number
 */
_PayoutMethodDetailsXOFMobile["default"].prototype['phone_number'] = undefined;
/**
 * @member {module:Model/PayoutMethodMobileProviderEnum} mobile_provider
 */
_PayoutMethodDetailsXOFMobile["default"].prototype['mobile_provider'] = undefined;
/**
 * @member {module:Model/PayoutMethodCountryEnum} country
 */
_PayoutMethodDetailsXOFMobile["default"].prototype['country'] = undefined;
/**
 * @member {module:Model/PayoutMethodTransferReasonEnum} transfer_reason
 */
_PayoutMethodDetailsXOFMobile["default"].prototype['transfer_reason'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsXOFMobile["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsXOFBank interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsXOFBank["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsXOFBank["default"].prototype['last_name'] = undefined;
/**
 * @member {String} iban
 */
_PayoutMethodDetailsXOFBank["default"].prototype['iban'] = undefined;
/**
 * @member {String} bank_name
 */
_PayoutMethodDetailsXOFBank["default"].prototype['bank_name'] = undefined;
/**
 * @member {String} bank_country
 */
_PayoutMethodDetailsXOFBank["default"].prototype['bank_country'] = undefined;
/**
 * @member {String} bank_code
 */
_PayoutMethodDetailsXOFBank["default"].prototype['bank_code'] = undefined;
/**
 * @member {module:Model/PayoutMethodTransferReasonEnum} transfer_reason
 */
_PayoutMethodDetailsXOFBank["default"].prototype['transfer_reason'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsXOFBank["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsXOFCash interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsXOFCash["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsXOFCash["default"].prototype['last_name'] = undefined;
/**
 * @member {String} phone_number
 */
_PayoutMethodDetailsXOFCash["default"].prototype['phone_number'] = undefined;
/**
 * @member {module:Model/PayoutMethodCashProviderEnum} cash_provider
 */
_PayoutMethodDetailsXOFCash["default"].prototype['cash_provider'] = undefined;
/**
 * @member {module:Model/PayoutMethodCountryEnum} country
 */
_PayoutMethodDetailsXOFCash["default"].prototype['country'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsXOFCash["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsGHSBank interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsGHSBank["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsGHSBank["default"].prototype['last_name'] = undefined;
/**
 * @member {String} bank_code
 */
_PayoutMethodDetailsGHSBank["default"].prototype['bank_code'] = undefined;
/**
 * @member {String} bank_account
 */
_PayoutMethodDetailsGHSBank["default"].prototype['bank_account'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsGHSBank["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsGHSCash interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsGHSCash["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsGHSCash["default"].prototype['last_name'] = undefined;
/**
 * @member {String} phone_number
 */
_PayoutMethodDetailsGHSCash["default"].prototype['phone_number'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsGHSCash["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsGBPBank interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsGBPBank["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsGBPBank["default"].prototype['last_name'] = undefined;
/**
 * @member {String} iban
 */
_PayoutMethodDetailsGBPBank["default"].prototype['iban'] = undefined;
/**
 * @member {String} bank_account
 */
_PayoutMethodDetailsGBPBank["default"].prototype['bank_account'] = undefined;
/**
 * @member {String} sort_code
 */
_PayoutMethodDetailsGBPBank["default"].prototype['sort_code'] = undefined;
/**
 * @member {String} bic
 */
_PayoutMethodDetailsGBPBank["default"].prototype['bic'] = undefined;
/**
 * @member {String} bank_name
 */
_PayoutMethodDetailsGBPBank["default"].prototype['bank_name'] = undefined;
/**
 * @member {String} narration
 */
_PayoutMethodDetailsGBPBank["default"].prototype['narration'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsGBPBank["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsIBAN interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsIBAN["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsIBAN["default"].prototype['last_name'] = undefined;
/**
 * @member {String} iban
 */
_PayoutMethodDetailsIBAN["default"].prototype['iban'] = undefined;
/**
 * @member {String} bic
 */
_PayoutMethodDetailsIBAN["default"].prototype['bic'] = undefined;
/**
 * @member {String} bank_name
 */
_PayoutMethodDetailsIBAN["default"].prototype['bank_name'] = undefined;
/**
 * @member {String} narration
 */
_PayoutMethodDetailsIBAN["default"].prototype['narration'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsIBAN["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsMADCash interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsMADCash["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsMADCash["default"].prototype['last_name'] = undefined;
/**
 * @member {String} phone_number
 */
_PayoutMethodDetailsMADCash["default"].prototype['phone_number'] = undefined;
/**
 * @member {module:Model/PayoutMethodIdentityCardTypeEnum} sender_identity_card_type
 */
_PayoutMethodDetailsMADCash["default"].prototype['sender_identity_card_type'] = undefined;
/**
 * @member {String} sender_identity_card_id
 */
_PayoutMethodDetailsMADCash["default"].prototype['sender_identity_card_id'] = undefined;
/**
 * @member {String} sender_city_of_birth
 */
_PayoutMethodDetailsMADCash["default"].prototype['sender_city_of_birth'] = undefined;
/**
 * @member {String} sender_country_of_birth
 */
_PayoutMethodDetailsMADCash["default"].prototype['sender_country_of_birth'] = undefined;
/**
 * @member {module:Model/PayoutMethodGenderEnum} sender_gender
 */
_PayoutMethodDetailsMADCash["default"].prototype['sender_gender'] = undefined;
/**
 * @member {String} reason
 */
_PayoutMethodDetailsMADCash["default"].prototype['reason'] = undefined;
/**
 * @member {module:Model/PayoutMethodIdentityCardTypeEnum} identity_card_type
 */
_PayoutMethodDetailsMADCash["default"].prototype['identity_card_type'] = undefined;
/**
 * @member {String} identity_card_id
 */
_PayoutMethodDetailsMADCash["default"].prototype['identity_card_id'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsMADCash["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsBalance interface:
/**
 * @member {String} reference
 */
_PayoutMethodDetailsBalance["default"].prototype['reference'] = undefined;
// Implement PayoutMethodDetailsBTC interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsBTC["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsBTC["default"].prototype['last_name'] = undefined;
/**
 * @member {String} name
 */
_PayoutMethodDetailsBTC["default"].prototype['name'] = undefined;
/**
 * @member {String} address
 */
_PayoutMethodDetailsBTC["default"].prototype['address'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsBTC["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsZARBank interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsZARBank["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsZARBank["default"].prototype['last_name'] = undefined;
/**
 * @member {String} street
 */
_PayoutMethodDetailsZARBank["default"].prototype['street'] = undefined;
/**
 * @member {String} postal_code
 */
_PayoutMethodDetailsZARBank["default"].prototype['postal_code'] = undefined;
/**
 * @member {String} city
 */
_PayoutMethodDetailsZARBank["default"].prototype['city'] = undefined;
/**
 * @member {String} email
 */
_PayoutMethodDetailsZARBank["default"].prototype['email'] = undefined;
/**
 * @member {String} bank_code
 */
_PayoutMethodDetailsZARBank["default"].prototype['bank_code'] = undefined;
/**
 * @member {String} branch_code
 */
_PayoutMethodDetailsZARBank["default"].prototype['branch_code'] = undefined;
/**
 * @member {String} bank_account
 */
_PayoutMethodDetailsZARBank["default"].prototype['bank_account'] = undefined;
/**
 * @member {String} phone_number
 */
_PayoutMethodDetailsZARBank["default"].prototype['phone_number'] = undefined;
/**
 * @member {String} transfer_reason_code
 */
_PayoutMethodDetailsZARBank["default"].prototype['transfer_reason_code'] = undefined;
/**
 * @member {module:Model/PayoutMethodTransferReasonEnum} transfer_reason
 */
_PayoutMethodDetailsZARBank["default"].prototype['transfer_reason'] = undefined;
/**
 * @member {String} narration
 */
_PayoutMethodDetailsZARBank["default"].prototype['narration'] = undefined;
/**
 * @member {String} name
 */
_PayoutMethodDetailsZARBank["default"].prototype['name'] = undefined;
/**
 * @member {String} contact_first_name
 */
_PayoutMethodDetailsZARBank["default"].prototype['contact_first_name'] = undefined;
/**
 * @member {String} contact_last_name
 */
_PayoutMethodDetailsZARBank["default"].prototype['contact_last_name'] = undefined;
/**
 * @member {String} registration_number
 */
_PayoutMethodDetailsZARBank["default"].prototype['registration_number'] = undefined;
/**
 * @member {module:Model/PayoutMethodNatureOfBusinessEnum} nature_of_business
 */
_PayoutMethodDetailsZARBank["default"].prototype['nature_of_business'] = undefined;
/**
 * @member {module:Model/PayoutMethodLegalEntityTypeEnum} legal_entity_type
 */
_PayoutMethodDetailsZARBank["default"].prototype['legal_entity_type'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsZARBank["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsUSDBank interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsUSDBank["default"].prototype['first_name'] = undefined;
/**
 * @member {String} middle_name
 */
_PayoutMethodDetailsUSDBank["default"].prototype['middle_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsUSDBank["default"].prototype['last_name'] = undefined;
/**
 * @member {String} phone_number
 */
_PayoutMethodDetailsUSDBank["default"].prototype['phone_number'] = undefined;
/**
 * @member {String} bank_code
 */
_PayoutMethodDetailsUSDBank["default"].prototype['bank_code'] = undefined;
/**
 * @member {String} bank_account
 */
_PayoutMethodDetailsUSDBank["default"].prototype['bank_account'] = undefined;
/**
 * @member {module:Model/PayoutMethodBankAccountTypeEnum} bank_account_type
 */
_PayoutMethodDetailsUSDBank["default"].prototype['bank_account_type'] = undefined;
/**
 * @member {String} bank_name
 */
_PayoutMethodDetailsUSDBank["default"].prototype['bank_name'] = undefined;
/**
 * @member {String} routing_number
 */
_PayoutMethodDetailsUSDBank["default"].prototype['routing_number'] = undefined;
/**
 * @member {String} swift_code
 */
_PayoutMethodDetailsUSDBank["default"].prototype['swift_code'] = undefined;
/**
 * @member {String} iban
 */
_PayoutMethodDetailsUSDBank["default"].prototype['iban'] = undefined;
/**
 * @member {String} street
 */
_PayoutMethodDetailsUSDBank["default"].prototype['street'] = undefined;
/**
 * @member {module:Model/PayoutMethodTransferReasonEnum} transfer_reason
 */
_PayoutMethodDetailsUSDBank["default"].prototype['transfer_reason'] = undefined;
/**
 * @member {module:Model/PayoutMethodCountryEnum} country
 */
_PayoutMethodDetailsUSDBank["default"].prototype['country'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsUSDBank["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsUSDCash interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsUSDCash["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsUSDCash["default"].prototype['last_name'] = undefined;
/**
 * @member {String} phone_number
 */
_PayoutMethodDetailsUSDCash["default"].prototype['phone_number'] = undefined;
/**
 * @member {module:Model/PayoutMethodCountryEnum} country
 */
_PayoutMethodDetailsUSDCash["default"].prototype['country'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsUSDCash["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsKESMobile interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsKESMobile["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsKESMobile["default"].prototype['last_name'] = undefined;
/**
 * @member {String} street
 */
_PayoutMethodDetailsKESMobile["default"].prototype['street'] = undefined;
/**
 * @member {String} city
 */
_PayoutMethodDetailsKESMobile["default"].prototype['city'] = undefined;
/**
 * @member {String} phone_number
 */
_PayoutMethodDetailsKESMobile["default"].prototype['phone_number'] = undefined;
/**
 * @member {module:Model/PayoutMethodMobileProviderEnum} mobile_provider
 */
_PayoutMethodDetailsKESMobile["default"].prototype['mobile_provider'] = undefined;
/**
 * @member {String} transfer_reason_code
 */
_PayoutMethodDetailsKESMobile["default"].prototype['transfer_reason_code'] = undefined;
/**
 * @member {module:Model/PayoutMethodTransferReasonEnum} transfer_reason
 */
_PayoutMethodDetailsKESMobile["default"].prototype['transfer_reason'] = undefined;
/**
 * @member {module:Model/PayoutMethodIdentityCardTypeEnum} identity_card_type
 */
_PayoutMethodDetailsKESMobile["default"].prototype['identity_card_type'] = undefined;
/**
 * @member {String} identity_card_id
 */
_PayoutMethodDetailsKESMobile["default"].prototype['identity_card_id'] = undefined;
/**
 * @member {String} relationship_to_sender
 */
_PayoutMethodDetailsKESMobile["default"].prototype['relationship_to_sender'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsKESMobile["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsKESBank interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsKESBank["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsKESBank["default"].prototype['last_name'] = undefined;
/**
 * @member {String} street
 */
_PayoutMethodDetailsKESBank["default"].prototype['street'] = undefined;
/**
 * @member {String} city
 */
_PayoutMethodDetailsKESBank["default"].prototype['city'] = undefined;
/**
 * @member {String} bank_code
 */
_PayoutMethodDetailsKESBank["default"].prototype['bank_code'] = undefined;
/**
 * @member {String} bank_account
 */
_PayoutMethodDetailsKESBank["default"].prototype['bank_account'] = undefined;
/**
 * @member {module:Model/PayoutMethodTransferReasonEnum} transfer_reason
 */
_PayoutMethodDetailsKESBank["default"].prototype['transfer_reason'] = undefined;
/**
 * @member {module:Model/PayoutMethodIdentityCardTypeEnum} identity_card_type
 */
_PayoutMethodDetailsKESBank["default"].prototype['identity_card_type'] = undefined;
/**
 * @member {String} identity_card_id
 */
_PayoutMethodDetailsKESBank["default"].prototype['identity_card_id'] = undefined;
/**
 * @member {String} relationship_to_sender
 */
_PayoutMethodDetailsKESBank["default"].prototype['relationship_to_sender'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsKESBank["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsXAFMobile interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsXAFMobile["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsXAFMobile["default"].prototype['last_name'] = undefined;
/**
 * @member {String} phone_number
 */
_PayoutMethodDetailsXAFMobile["default"].prototype['phone_number'] = undefined;
/**
 * @member {module:Model/PayoutMethodMobileProviderEnum} mobile_provider
 */
_PayoutMethodDetailsXAFMobile["default"].prototype['mobile_provider'] = undefined;
/**
 * @member {module:Model/PayoutMethodCountryEnum} country
 */
_PayoutMethodDetailsXAFMobile["default"].prototype['country'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsXAFMobile["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsXAFBank interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsXAFBank["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsXAFBank["default"].prototype['last_name'] = undefined;
/**
 * @member {String} iban
 */
_PayoutMethodDetailsXAFBank["default"].prototype['iban'] = undefined;
/**
 * @member {String} bank_account
 */
_PayoutMethodDetailsXAFBank["default"].prototype['bank_account'] = undefined;
/**
 * @member {String} bank_name
 */
_PayoutMethodDetailsXAFBank["default"].prototype['bank_name'] = undefined;
/**
 * @member {String} bank_country
 */
_PayoutMethodDetailsXAFBank["default"].prototype['bank_country'] = undefined;
/**
 * @member {String} bank_code
 */
_PayoutMethodDetailsXAFBank["default"].prototype['bank_code'] = undefined;
/**
 * @member {String} sort_code
 */
_PayoutMethodDetailsXAFBank["default"].prototype['sort_code'] = undefined;
/**
 * @member {module:Model/PayoutMethodTransferReasonEnum} transfer_reason
 */
_PayoutMethodDetailsXAFBank["default"].prototype['transfer_reason'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsXAFBank["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsGNFMobile interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsGNFMobile["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsGNFMobile["default"].prototype['last_name'] = undefined;
/**
 * @member {String} phone_number
 */
_PayoutMethodDetailsGNFMobile["default"].prototype['phone_number'] = undefined;
/**
 * @member {module:Model/PayoutMethodMobileProviderEnum} mobile_provider
 */
_PayoutMethodDetailsGNFMobile["default"].prototype['mobile_provider'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsGNFMobile["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsBRLBank interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsBRLBank["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsBRLBank["default"].prototype['last_name'] = undefined;
/**
 * @member {String} city
 */
_PayoutMethodDetailsBRLBank["default"].prototype['city'] = undefined;
/**
 * @member {String} postal_code
 */
_PayoutMethodDetailsBRLBank["default"].prototype['postal_code'] = undefined;
/**
 * @member {String} phone_number
 */
_PayoutMethodDetailsBRLBank["default"].prototype['phone_number'] = undefined;
/**
 * @member {String} bank_code
 */
_PayoutMethodDetailsBRLBank["default"].prototype['bank_code'] = undefined;
/**
 * @member {String} branch_code
 */
_PayoutMethodDetailsBRLBank["default"].prototype['branch_code'] = undefined;
/**
 * @member {String} bank_account
 */
_PayoutMethodDetailsBRLBank["default"].prototype['bank_account'] = undefined;
/**
 * @member {module:Model/PayoutMethodBankAccountTypeEnum} bank_account_type
 */
_PayoutMethodDetailsBRLBank["default"].prototype['bank_account_type'] = undefined;
/**
 * @member {module:Model/PayoutMethodPixKeyTypeEnum} pix_key_type
 */
_PayoutMethodDetailsBRLBank["default"].prototype['pix_key_type'] = undefined;
/**
 * @member {String} pix_key_value
 */
_PayoutMethodDetailsBRLBank["default"].prototype['pix_key_value'] = undefined;
/**
 * @member {String} identity_card_id
 */
_PayoutMethodDetailsBRLBank["default"].prototype['identity_card_id'] = undefined;
/**
 * @member {module:Model/PayoutMethodTransferReasonEnum} transfer_reason
 */
_PayoutMethodDetailsBRLBank["default"].prototype['transfer_reason'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsBRLBank["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsNZDBank interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsNZDBank["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsNZDBank["default"].prototype['last_name'] = undefined;
/**
 * @member {String} bank_account
 */
_PayoutMethodDetailsNZDBank["default"].prototype['bank_account'] = undefined;
/**
 * @member {String} bank_name
 */
_PayoutMethodDetailsNZDBank["default"].prototype['bank_name'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsNZDBank["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsINRBank interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsINRBank["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsINRBank["default"].prototype['last_name'] = undefined;
/**
 * @member {String} city
 */
_PayoutMethodDetailsINRBank["default"].prototype['city'] = undefined;
/**
 * @member {String} street
 */
_PayoutMethodDetailsINRBank["default"].prototype['street'] = undefined;
/**
 * @member {String} postal_code
 */
_PayoutMethodDetailsINRBank["default"].prototype['postal_code'] = undefined;
/**
 * @member {String} phone_number
 */
_PayoutMethodDetailsINRBank["default"].prototype['phone_number'] = undefined;
/**
 * @member {String} bank_name
 */
_PayoutMethodDetailsINRBank["default"].prototype['bank_name'] = undefined;
/**
 * @member {String} bank_account
 */
_PayoutMethodDetailsINRBank["default"].prototype['bank_account'] = undefined;
/**
 * @member {String} ifsc_code
 */
_PayoutMethodDetailsINRBank["default"].prototype['ifsc_code'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsINRBank["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsBWPBank interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsBWPBank["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsBWPBank["default"].prototype['last_name'] = undefined;
/**
 * @member {String} bank_name
 */
_PayoutMethodDetailsBWPBank["default"].prototype['bank_name'] = undefined;
/**
 * @member {String} bank_account
 */
_PayoutMethodDetailsBWPBank["default"].prototype['bank_account'] = undefined;
/**
 * @member {String} branch_code
 */
_PayoutMethodDetailsBWPBank["default"].prototype['branch_code'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsBWPBank["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsZMWBank interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsZMWBank["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsZMWBank["default"].prototype['last_name'] = undefined;
/**
 * @member {String} bank_account
 */
_PayoutMethodDetailsZMWBank["default"].prototype['bank_account'] = undefined;
/**
 * @member {String} branch_code
 */
_PayoutMethodDetailsZMWBank["default"].prototype['branch_code'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsZMWBank["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsCADBank interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsCADBank["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsCADBank["default"].prototype['last_name'] = undefined;
/**
 * @member {String} bank_account
 */
_PayoutMethodDetailsCADBank["default"].prototype['bank_account'] = undefined;
/**
 * @member {String} bank_code
 */
_PayoutMethodDetailsCADBank["default"].prototype['bank_code'] = undefined;
/**
 * @member {String} branch_code
 */
_PayoutMethodDetailsCADBank["default"].prototype['branch_code'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsCADBank["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsUGXBank interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsUGXBank["default"].prototype['first_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsUGXBank["default"].prototype['last_name'] = undefined;
/**
 * @member {String} street
 */
_PayoutMethodDetailsUGXBank["default"].prototype['street'] = undefined;
/**
 * @member {String} city
 */
_PayoutMethodDetailsUGXBank["default"].prototype['city'] = undefined;
/**
 * @member {String} postal_code
 */
_PayoutMethodDetailsUGXBank["default"].prototype['postal_code'] = undefined;
/**
 * @member {String} identity_card_id
 */
_PayoutMethodDetailsUGXBank["default"].prototype['identity_card_id'] = undefined;
/**
 * @member {String} bank_account
 */
_PayoutMethodDetailsUGXBank["default"].prototype['bank_account'] = undefined;
/**
 * @member {String} branch_code
 */
_PayoutMethodDetailsUGXBank["default"].prototype['branch_code'] = undefined;
/**
 * @member {module:Model/PayoutMethodTransferReasonEnum} transfer_reason
 */
_PayoutMethodDetailsUGXBank["default"].prototype['transfer_reason'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsUGXBank["default"].prototype['birth_date'] = undefined;
// Implement PayoutMethodDetailsEGPBank interface:
/**
 * @member {String} first_name
 */
_PayoutMethodDetailsEGPBank["default"].prototype['first_name'] = undefined;
/**
 * @member {String} middle_name
 */
_PayoutMethodDetailsEGPBank["default"].prototype['middle_name'] = undefined;
/**
 * @member {String} last_name
 */
_PayoutMethodDetailsEGPBank["default"].prototype['last_name'] = undefined;
/**
 * @member {String} street
 */
_PayoutMethodDetailsEGPBank["default"].prototype['street'] = undefined;
/**
 * @member {String} phone_number
 */
_PayoutMethodDetailsEGPBank["default"].prototype['phone_number'] = undefined;
/**
 * @member {String} bank_account
 */
_PayoutMethodDetailsEGPBank["default"].prototype['bank_account'] = undefined;
/**
 * @member {String} bank_code
 */
_PayoutMethodDetailsEGPBank["default"].prototype['bank_code'] = undefined;
/**
 * @member {module:Model/PayoutMethodTransferReasonEnum} transfer_reason
 */
_PayoutMethodDetailsEGPBank["default"].prototype['transfer_reason'] = undefined;
/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
_PayoutMethodDetailsEGPBank["default"].prototype['birth_date'] = undefined;var _default = exports["default"] =




PayoutMethodDetails;
