// transferzero-sdk/dist/Model/PayoutMethodDetailsXAFBank.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports["default"] = void 0;












var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _PayoutMethodTransferReasonEnum = _interopRequireDefault(require("./PayoutMethodTransferReasonEnum"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };}function _typeof(o) {"@babel/helpers - typeof";return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) {return typeof o;} : function (o) {return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o;}, _typeof(o);}function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError("Cannot call a class as a function");}}function _defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ("value" in descriptor) descriptor.writable = true;Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);}}function _createClass(Constructor, protoProps, staticProps) {if (protoProps) _defineProperties(Constructor.prototype, protoProps);if (staticProps) _defineProperties(Constructor, staticProps);Object.defineProperty(Constructor, "prototype", { writable: false });return Constructor;}function _toPropertyKey(t) {var i = _toPrimitive(t, "string");return "symbol" == _typeof(i) ? i : i + "";}function _toPrimitive(t, r) {if ("object" != _typeof(t) || !t) return t;var e = t[Symbol.toPrimitive];if (void 0 !== e) {var i = e.call(t, r || "default");if ("object" != _typeof(i)) return i;throw new TypeError("@@toPrimitive must return a primitive value.");}return ("string" === r ? String : Number)(t);} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
 * The PayoutMethodDetailsXAFBank model module.
 * @module Model/PayoutMethodDetailsXAFBank
 * @version 1.35.1
 */var PayoutMethodDetailsXAFBank = /*#__PURE__*/function () {/**
   * Constructs a new <code>PayoutMethodDetailsXAFBank</code>.
   * &#x60;&#x60;&#x60;JSON \&quot;details\&quot;: {   \&quot;first_name\&quot;: \&quot;First\&quot;,   \&quot;last_name\&quot;: \&quot;Last\&quot;,   \&quot;iban\&quot;: \&quot;CM2110033052090901100045910\&quot;, # IBAN format   \&quot;bank_account\&quot;: \&quot;09011000459\&quot;,   \&quot;bank_name\&quot;: \&quot;United Bank for Africa - Cameroon\&quot;,   \&quot;bank_country\&quot;: \&quot;CM\&quot;, # ISO country code for Cameroon   \&quot;bank_code\&quot;: \&quot;10033\&quot;,   \&quot;sort_code\&quot;: \&quot;10\&quot;,   \&quot;transfer_reason\&quot;: \&quot;personal_account\&quot; } &#x60;&#x60;&#x60;  See [XAF Bank](https://docs.transferzero.com/docs/payout-details/#xafbank) documentation for the bank_code and transfer_reason lists
   * @alias module:Model/PayoutMethodDetailsXAFBank
   * @param firstName {String} 
   * @param lastName {String} 
   * @param iban {String} 
   */function PayoutMethodDetailsXAFBank(firstName, lastName, iban) {_classCallCheck(this, PayoutMethodDetailsXAFBank);PayoutMethodDetailsXAFBank.initialize(this, firstName, lastName, iban);} /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */return _createClass(PayoutMethodDetailsXAFBank, null, [{ key: "initialize", value: function initialize(obj, firstName, lastName, iban) {
      obj['first_name'] = firstName;
      obj['last_name'] = lastName;
      obj['iban'] = iban;
    }

    /**
     * Constructs a <code>PayoutMethodDetailsXAFBank</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:Model/PayoutMethodDetailsXAFBank} obj Optional instance to populate.
     * @return {module:Model/PayoutMethodDetailsXAFBank} The populated <code>PayoutMethodDetailsXAFBank</code> instance.
     */ }, { key: "constructFromObject", value:
    function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PayoutMethodDetailsXAFBank();

        if (data.hasOwnProperty('first_name')) {
          obj['first_name'] = _ApiClient["default"].convertToType(data['first_name'], 'String');
        }
        if (data.hasOwnProperty('last_name')) {
          obj['last_name'] = _ApiClient["default"].convertToType(data['last_name'], 'String');
        }
        if (data.hasOwnProperty('iban')) {
          obj['iban'] = _ApiClient["default"].convertToType(data['iban'], 'String');
        }
        if (data.hasOwnProperty('bank_account')) {
          obj['bank_account'] = _ApiClient["default"].convertToType(data['bank_account'], 'String');
        }
        if (data.hasOwnProperty('bank_name')) {
          obj['bank_name'] = _ApiClient["default"].convertToType(data['bank_name'], 'String');
        }
        if (data.hasOwnProperty('bank_country')) {
          obj['bank_country'] = _ApiClient["default"].convertToType(data['bank_country'], 'String');
        }
        if (data.hasOwnProperty('bank_code')) {
          obj['bank_code'] = _ApiClient["default"].convertToType(data['bank_code'], 'String');
        }
        if (data.hasOwnProperty('sort_code')) {
          obj['sort_code'] = _ApiClient["default"].convertToType(data['sort_code'], 'String');
        }
        if (data.hasOwnProperty('transfer_reason')) {
          obj['transfer_reason'] = _PayoutMethodTransferReasonEnum["default"].constructFromObject(data['transfer_reason']);
        }
        if (data.hasOwnProperty('birth_date')) {
          obj['birth_date'] = _ApiClient["default"].convertToType(data['birth_date'], 'Date');
        }
      }
      return obj;
    } }]);}();




/**
 * @member {String} first_name
 */
PayoutMethodDetailsXAFBank.prototype['first_name'] = undefined;

/**
 * @member {String} last_name
 */
PayoutMethodDetailsXAFBank.prototype['last_name'] = undefined;

/**
 * @member {String} iban
 */
PayoutMethodDetailsXAFBank.prototype['iban'] = undefined;

/**
 * @member {String} bank_account
 */
PayoutMethodDetailsXAFBank.prototype['bank_account'] = undefined;

/**
 * @member {String} bank_name
 */
PayoutMethodDetailsXAFBank.prototype['bank_name'] = undefined;

/**
 * @member {String} bank_country
 */
PayoutMethodDetailsXAFBank.prototype['bank_country'] = undefined;

/**
 * @member {String} bank_code
 */
PayoutMethodDetailsXAFBank.prototype['bank_code'] = undefined;

/**
 * @member {String} sort_code
 */
PayoutMethodDetailsXAFBank.prototype['sort_code'] = undefined;

/**
 * @member {module:Model/PayoutMethodTransferReasonEnum} transfer_reason
 */
PayoutMethodDetailsXAFBank.prototype['transfer_reason'] = undefined;

/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
PayoutMethodDetailsXAFBank.prototype['birth_date'] = undefined;var _default = exports["default"] =






PayoutMethodDetailsXAFBank;
