// transferzero-sdk/dist/Model/PayoutMethodDetailsXOFMobile.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports["default"] = void 0;












var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _PayoutMethodCountryEnum = _interopRequireDefault(require("./PayoutMethodCountryEnum"));
var _PayoutMethodMobileProviderEnum = _interopRequireDefault(require("./PayoutMethodMobileProviderEnum"));
var _PayoutMethodTransferReasonEnum = _interopRequireDefault(require("./PayoutMethodTransferReasonEnum"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };}function _typeof(o) {"@babel/helpers - typeof";return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) {return typeof o;} : function (o) {return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o;}, _typeof(o);}function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError("Cannot call a class as a function");}}function _defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ("value" in descriptor) descriptor.writable = true;Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);}}function _createClass(Constructor, protoProps, staticProps) {if (protoProps) _defineProperties(Constructor.prototype, protoProps);if (staticProps) _defineProperties(Constructor, staticProps);Object.defineProperty(Constructor, "prototype", { writable: false });return Constructor;}function _toPropertyKey(t) {var i = _toPrimitive(t, "string");return "symbol" == _typeof(i) ? i : i + "";}function _toPrimitive(t, r) {if ("object" != _typeof(t) || !t) return t;var e = t[Symbol.toPrimitive];if (void 0 !== e) {var i = e.call(t, r || "default");if ("object" != _typeof(i)) return i;throw new TypeError("@@toPrimitive must return a primitive value.");}return ("string" === r ? String : Number)(t);} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
 * The PayoutMethodDetailsXOFMobile model module.
 * @module Model/PayoutMethodDetailsXOFMobile
 * @version 1.35.1
 */var PayoutMethodDetailsXOFMobile = /*#__PURE__*/function () {/**
   * Constructs a new <code>PayoutMethodDetailsXOFMobile</code>.
   * &#x60;&#x60;&#x60;JSON \&quot;details\&quot;: {   \&quot;first_name\&quot;: \&quot;First\&quot;,   \&quot;last_name\&quot;: \&quot;Last\&quot;,   \&quot;phone_number\&quot;: \&quot;+221708766123\&quot;, // E.164 international format   \&quot;mobile_provider\&quot;: \&quot;orange\&quot;, // \&quot;orange\&quot;, \&quot;tigo\&quot;, \&quot;emoney\&quot; or \&quot;free\&quot; for Senegal; \&quot;orange\&quot;, \&quot;mtn\&quot; or \&quot;moov\&quot; for Ivory Coast; \&quot;orange\&quot; or \&quot;mobicash\&quot; for Burkina Faso and Mali   \&quot;country\&quot; // Optional; Values: \&quot;SN\&quot; for Senegal; \&quot;CI\&quot; for Ivory Coast; \&quot;ML\&quot; for Mali; \&quot;BF\&quot; for Burkina Faso; Default value is \&quot;SN\&quot;   \&quot;transfer_reason\&quot;: \&quot;personal_account\&quot; // mandatory for Mali payouts, optional otherwise } &#x60;&#x60;&#x60;
   * @alias module:Model/PayoutMethodDetailsXOFMobile
   * @param firstName {String} 
   * @param lastName {String} 
   * @param phoneNumber {String} 
   * @param mobileProvider {module:Model/PayoutMethodMobileProviderEnum} 
   */function PayoutMethodDetailsXOFMobile(firstName, lastName, phoneNumber, mobileProvider) {_classCallCheck(this, PayoutMethodDetailsXOFMobile);PayoutMethodDetailsXOFMobile.initialize(this, firstName, lastName, phoneNumber, mobileProvider);} /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */return _createClass(PayoutMethodDetailsXOFMobile, null, [{ key: "initialize", value: function initialize(obj, firstName, lastName, phoneNumber, mobileProvider) {
      obj['first_name'] = firstName;
      obj['last_name'] = lastName;
      obj['phone_number'] = phoneNumber;
      obj['mobile_provider'] = mobileProvider;
    }

    /**
     * Constructs a <code>PayoutMethodDetailsXOFMobile</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:Model/PayoutMethodDetailsXOFMobile} obj Optional instance to populate.
     * @return {module:Model/PayoutMethodDetailsXOFMobile} The populated <code>PayoutMethodDetailsXOFMobile</code> instance.
     */ }, { key: "constructFromObject", value:
    function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PayoutMethodDetailsXOFMobile();

        if (data.hasOwnProperty('first_name')) {
          obj['first_name'] = _ApiClient["default"].convertToType(data['first_name'], 'String');
        }
        if (data.hasOwnProperty('last_name')) {
          obj['last_name'] = _ApiClient["default"].convertToType(data['last_name'], 'String');
        }
        if (data.hasOwnProperty('phone_number')) {
          obj['phone_number'] = _ApiClient["default"].convertToType(data['phone_number'], 'String');
        }
        if (data.hasOwnProperty('mobile_provider')) {
          obj['mobile_provider'] = _PayoutMethodMobileProviderEnum["default"].constructFromObject(data['mobile_provider']);
        }
        if (data.hasOwnProperty('country')) {
          obj['country'] = _PayoutMethodCountryEnum["default"].constructFromObject(data['country']);
        }
        if (data.hasOwnProperty('transfer_reason')) {
          obj['transfer_reason'] = _PayoutMethodTransferReasonEnum["default"].constructFromObject(data['transfer_reason']);
        }
        if (data.hasOwnProperty('birth_date')) {
          obj['birth_date'] = _ApiClient["default"].convertToType(data['birth_date'], 'Date');
        }
      }
      return obj;
    } }]);}();




/**
 * @member {String} first_name
 */
PayoutMethodDetailsXOFMobile.prototype['first_name'] = undefined;

/**
 * @member {String} last_name
 */
PayoutMethodDetailsXOFMobile.prototype['last_name'] = undefined;

/**
 * @member {String} phone_number
 */
PayoutMethodDetailsXOFMobile.prototype['phone_number'] = undefined;

/**
 * @member {module:Model/PayoutMethodMobileProviderEnum} mobile_provider
 */
PayoutMethodDetailsXOFMobile.prototype['mobile_provider'] = undefined;

/**
 * @member {module:Model/PayoutMethodCountryEnum} country
 */
PayoutMethodDetailsXOFMobile.prototype['country'] = undefined;

/**
 * @member {module:Model/PayoutMethodTransferReasonEnum} transfer_reason
 */
PayoutMethodDetailsXOFMobile.prototype['transfer_reason'] = undefined;

/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
PayoutMethodDetailsXOFMobile.prototype['birth_date'] = undefined;var _default = exports["default"] =






PayoutMethodDetailsXOFMobile;
