// transferzero-sdk/dist/Model/PayoutMethodDetailsBTC.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports["default"] = void 0;












var _ApiClient = _interopRequireDefault(require("../ApiClient"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };}function _typeof(o) {"@babel/helpers - typeof";return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) {return typeof o;} : function (o) {return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o;}, _typeof(o);}function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError("Cannot call a class as a function");}}function _defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ("value" in descriptor) descriptor.writable = true;Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);}}function _createClass(Constructor, protoProps, staticProps) {if (protoProps) _defineProperties(Constructor.prototype, protoProps);if (staticProps) _defineProperties(Constructor, staticProps);Object.defineProperty(Constructor, "prototype", { writable: false });return Constructor;}function _toPropertyKey(t) {var i = _toPrimitive(t, "string");return "symbol" == _typeof(i) ? i : i + "";}function _toPrimitive(t, r) {if ("object" != _typeof(t) || !t) return t;var e = t[Symbol.toPrimitive];if (void 0 !== e) {var i = e.call(t, r || "default");if ("object" != _typeof(i)) return i;throw new TypeError("@@toPrimitive must return a primitive value.");}return ("string" === r ? String : Number)(t);} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
 * The PayoutMethodDetailsBTC model module.
 * @module Model/PayoutMethodDetailsBTC
 * @version 1.35.1
 */var PayoutMethodDetailsBTC = /*#__PURE__*/function () {/**
   * Constructs a new <code>PayoutMethodDetailsBTC</code>.
   * &#x60;&#x60;&#x60;JSON \&quot;details\&quot;: {   \&quot;first_name\&quot;: \&quot;First\&quot;,   \&quot;last_name\&quot;: \&quot;Last\&quot;,   \&quot;name\&quot;: \&quot;Full Name\&quot;   \&quot;address\&quot;: \&quot;n4VQ5YdHf7hLQ2gWQYYrcxoE5B7nWuDFNF\&quot; }
   * @alias module:Model/PayoutMethodDetailsBTC
   * @param firstName {String} 
   * @param lastName {String} 
   * @param name {String} 
   * @param address {String} 
   */function PayoutMethodDetailsBTC(firstName, lastName, name, address) {_classCallCheck(this, PayoutMethodDetailsBTC);PayoutMethodDetailsBTC.initialize(this, firstName, lastName, name, address);} /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */return _createClass(PayoutMethodDetailsBTC, null, [{ key: "initialize", value: function initialize(obj, firstName, lastName, name, address) {
      obj['first_name'] = firstName;
      obj['last_name'] = lastName;
      obj['name'] = name;
      obj['address'] = address;
    }

    /**
     * Constructs a <code>PayoutMethodDetailsBTC</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:Model/PayoutMethodDetailsBTC} obj Optional instance to populate.
     * @return {module:Model/PayoutMethodDetailsBTC} The populated <code>PayoutMethodDetailsBTC</code> instance.
     */ }, { key: "constructFromObject", value:
    function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PayoutMethodDetailsBTC();

        if (data.hasOwnProperty('first_name')) {
          obj['first_name'] = _ApiClient["default"].convertToType(data['first_name'], 'String');
        }
        if (data.hasOwnProperty('last_name')) {
          obj['last_name'] = _ApiClient["default"].convertToType(data['last_name'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('address')) {
          obj['address'] = _ApiClient["default"].convertToType(data['address'], 'String');
        }
        if (data.hasOwnProperty('birth_date')) {
          obj['birth_date'] = _ApiClient["default"].convertToType(data['birth_date'], 'Date');
        }
      }
      return obj;
    } }]);}();




/**
 * @member {String} first_name
 */
PayoutMethodDetailsBTC.prototype['first_name'] = undefined;

/**
 * @member {String} last_name
 */
PayoutMethodDetailsBTC.prototype['last_name'] = undefined;

/**
 * @member {String} name
 */
PayoutMethodDetailsBTC.prototype['name'] = undefined;

/**
 * @member {String} address
 */
PayoutMethodDetailsBTC.prototype['address'] = undefined;

/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
PayoutMethodDetailsBTC.prototype['birth_date'] = undefined;var _default = exports["default"] =






PayoutMethodDetailsBTC;
