// transferzero-sdk/dist/Model/PayoutMethodDetailsZARBank.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports["default"] = void 0;












var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _PayoutMethodLegalEntityTypeEnum = _interopRequireDefault(require("./PayoutMethodLegalEntityTypeEnum"));
var _PayoutMethodNatureOfBusinessEnum = _interopRequireDefault(require("./PayoutMethodNatureOfBusinessEnum"));
var _PayoutMethodTransferReasonEnum = _interopRequireDefault(require("./PayoutMethodTransferReasonEnum"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };}function _typeof(o) {"@babel/helpers - typeof";return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) {return typeof o;} : function (o) {return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o;}, _typeof(o);}function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError("Cannot call a class as a function");}}function _defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ("value" in descriptor) descriptor.writable = true;Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);}}function _createClass(Constructor, protoProps, staticProps) {if (protoProps) _defineProperties(Constructor.prototype, protoProps);if (staticProps) _defineProperties(Constructor, staticProps);Object.defineProperty(Constructor, "prototype", { writable: false });return Constructor;}function _toPropertyKey(t) {var i = _toPrimitive(t, "string");return "symbol" == _typeof(i) ? i : i + "";}function _toPrimitive(t, r) {if ("object" != _typeof(t) || !t) return t;var e = t[Symbol.toPrimitive];if (void 0 !== e) {var i = e.call(t, r || "default");if ("object" != _typeof(i)) return i;throw new TypeError("@@toPrimitive must return a primitive value.");}return ("string" === r ? String : Number)(t);} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
 * The PayoutMethodDetailsZARBank model module.
 * @module Model/PayoutMethodDetailsZARBank
 * @version 1.35.1
 */var PayoutMethodDetailsZARBank = /*#__PURE__*/function () {/**
   * Constructs a new <code>PayoutMethodDetailsZARBank</code>.
   * &#x60;&#x60;&#x60;JSON   \&quot;details\&quot;: {     \&quot;first_name\&quot;: \&quot;First\&quot;, //  Mandatory for personal payouts;     \&quot;last_name\&quot;: \&quot;Last\&quot;, //  Mandatory for personal payouts;     \&quot;name\&quot; \&quot;First Ltd\&quot;, // Mandatory for business payouts;     \&quot;contact_first_name\&quot; \&quot;Business\&quot;,     \&quot;contact_last_name\&quot; \&quot;Contact\&quot;,     \&quot;street\&quot;: \&quot;Main Street\&quot;,     \&quot;postal_code\&quot;: \&quot;AB0001\&quot;,     \&quot;city\&quot;: \&quot;Cape Town\&quot;,     \&quot;email\&quot;: \&quot;recipient@email.com\&quot;,     \&quot;bank_name\&quot; &#39;Bank Zero&#39;, // Optional     \&quot;bank_code\&quot;: \&quot;334810\&quot;,  // Optional; Required if branch_code is empty     \&quot;branch_code\&quot;: \&quot;630067\&quot;, // Optional; Required if bank_code is empty     \&quot;bank_account\&quot;: \&quot;12345678\&quot;,     \&quot;phone_number\&quot;: \&quot;+27119785313\&quot;, // E.164 international format     \&quot;transfer_reason\&quot;: \&quot;personal_account\&quot;, // New transfer reason field     \&quot;narration\&quot;: \&quot;Birthday Gift\&quot;, // Optional     \&quot;legal_entity_type\&quot;: \&quot;sole_proprietorship\&quot;, // Optional; Default value is \&quot;person\&quot;;     \&quot;nature_of_business\&quot;: \&quot;mining\&quot;, // Optional for business payouts;     \&quot;registration_number\&quot;: \&quot;17364BGC\&quot; // Optional for business payouts;   } &#x60;&#x60;&#x60;  See [ZAR Bank](https://docs.transferzero.com/docs/payout-details/#zarbank) documentation for the bank_code and transfer_reason lists
   * @alias module:Model/PayoutMethodDetailsZARBank
   * @param firstName {String} 
   * @param lastName {String} 
   * @param bankAccount {String} 
   * @param phoneNumber {String} 
   */function PayoutMethodDetailsZARBank(firstName, lastName, bankAccount, phoneNumber) {_classCallCheck(this, PayoutMethodDetailsZARBank);PayoutMethodDetailsZARBank.initialize(this, firstName, lastName, bankAccount, phoneNumber);} /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */return _createClass(PayoutMethodDetailsZARBank, null, [{ key: "initialize", value: function initialize(obj, firstName, lastName, bankAccount, phoneNumber) {
      obj['first_name'] = firstName;
      obj['last_name'] = lastName;
      obj['bank_account'] = bankAccount;
      obj['phone_number'] = phoneNumber;
    }

    /**
     * Constructs a <code>PayoutMethodDetailsZARBank</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:Model/PayoutMethodDetailsZARBank} obj Optional instance to populate.
     * @return {module:Model/PayoutMethodDetailsZARBank} The populated <code>PayoutMethodDetailsZARBank</code> instance.
     */ }, { key: "constructFromObject", value:
    function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PayoutMethodDetailsZARBank();

        if (data.hasOwnProperty('first_name')) {
          obj['first_name'] = _ApiClient["default"].convertToType(data['first_name'], 'String');
        }
        if (data.hasOwnProperty('last_name')) {
          obj['last_name'] = _ApiClient["default"].convertToType(data['last_name'], 'String');
        }
        if (data.hasOwnProperty('street')) {
          obj['street'] = _ApiClient["default"].convertToType(data['street'], 'String');
        }
        if (data.hasOwnProperty('postal_code')) {
          obj['postal_code'] = _ApiClient["default"].convertToType(data['postal_code'], 'String');
        }
        if (data.hasOwnProperty('city')) {
          obj['city'] = _ApiClient["default"].convertToType(data['city'], 'String');
        }
        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], 'String');
        }
        if (data.hasOwnProperty('bank_code')) {
          obj['bank_code'] = _ApiClient["default"].convertToType(data['bank_code'], 'String');
        }
        if (data.hasOwnProperty('branch_code')) {
          obj['branch_code'] = _ApiClient["default"].convertToType(data['branch_code'], 'String');
        }
        if (data.hasOwnProperty('bank_account')) {
          obj['bank_account'] = _ApiClient["default"].convertToType(data['bank_account'], 'String');
        }
        if (data.hasOwnProperty('phone_number')) {
          obj['phone_number'] = _ApiClient["default"].convertToType(data['phone_number'], 'String');
        }
        if (data.hasOwnProperty('transfer_reason_code')) {
          obj['transfer_reason_code'] = _ApiClient["default"].convertToType(data['transfer_reason_code'], 'String');
        }
        if (data.hasOwnProperty('transfer_reason')) {
          obj['transfer_reason'] = _PayoutMethodTransferReasonEnum["default"].constructFromObject(data['transfer_reason']);
        }
        if (data.hasOwnProperty('narration')) {
          obj['narration'] = _ApiClient["default"].convertToType(data['narration'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('contact_first_name')) {
          obj['contact_first_name'] = _ApiClient["default"].convertToType(data['contact_first_name'], 'String');
        }
        if (data.hasOwnProperty('contact_last_name')) {
          obj['contact_last_name'] = _ApiClient["default"].convertToType(data['contact_last_name'], 'String');
        }
        if (data.hasOwnProperty('registration_number')) {
          obj['registration_number'] = _ApiClient["default"].convertToType(data['registration_number'], 'String');
        }
        if (data.hasOwnProperty('nature_of_business')) {
          obj['nature_of_business'] = _PayoutMethodNatureOfBusinessEnum["default"].constructFromObject(data['nature_of_business']);
        }
        if (data.hasOwnProperty('legal_entity_type')) {
          obj['legal_entity_type'] = _PayoutMethodLegalEntityTypeEnum["default"].constructFromObject(data['legal_entity_type']);
        }
        if (data.hasOwnProperty('birth_date')) {
          obj['birth_date'] = _ApiClient["default"].convertToType(data['birth_date'], 'Date');
        }
      }
      return obj;
    } }]);}();




/**
 * @member {String} first_name
 */
PayoutMethodDetailsZARBank.prototype['first_name'] = undefined;

/**
 * @member {String} last_name
 */
PayoutMethodDetailsZARBank.prototype['last_name'] = undefined;

/**
 * @member {String} street
 */
PayoutMethodDetailsZARBank.prototype['street'] = undefined;

/**
 * @member {String} postal_code
 */
PayoutMethodDetailsZARBank.prototype['postal_code'] = undefined;

/**
 * @member {String} city
 */
PayoutMethodDetailsZARBank.prototype['city'] = undefined;

/**
 * @member {String} email
 */
PayoutMethodDetailsZARBank.prototype['email'] = undefined;

/**
 * @member {String} bank_code
 */
PayoutMethodDetailsZARBank.prototype['bank_code'] = undefined;

/**
 * @member {String} branch_code
 */
PayoutMethodDetailsZARBank.prototype['branch_code'] = undefined;

/**
 * @member {String} bank_account
 */
PayoutMethodDetailsZARBank.prototype['bank_account'] = undefined;

/**
 * @member {String} phone_number
 */
PayoutMethodDetailsZARBank.prototype['phone_number'] = undefined;

/**
 * @member {String} transfer_reason_code
 */
PayoutMethodDetailsZARBank.prototype['transfer_reason_code'] = undefined;

/**
 * @member {module:Model/PayoutMethodTransferReasonEnum} transfer_reason
 */
PayoutMethodDetailsZARBank.prototype['transfer_reason'] = undefined;

/**
 * @member {String} narration
 */
PayoutMethodDetailsZARBank.prototype['narration'] = undefined;

/**
 * @member {String} name
 */
PayoutMethodDetailsZARBank.prototype['name'] = undefined;

/**
 * @member {String} contact_first_name
 */
PayoutMethodDetailsZARBank.prototype['contact_first_name'] = undefined;

/**
 * @member {String} contact_last_name
 */
PayoutMethodDetailsZARBank.prototype['contact_last_name'] = undefined;

/**
 * @member {String} registration_number
 */
PayoutMethodDetailsZARBank.prototype['registration_number'] = undefined;

/**
 * @member {module:Model/PayoutMethodNatureOfBusinessEnum} nature_of_business
 */
PayoutMethodDetailsZARBank.prototype['nature_of_business'] = undefined;

/**
 * @member {module:Model/PayoutMethodLegalEntityTypeEnum} legal_entity_type
 */
PayoutMethodDetailsZARBank.prototype['legal_entity_type'] = undefined;

/**
 * Date of birth of recipient
 * @member {Date} birth_date
 */
PayoutMethodDetailsZARBank.prototype['birth_date'] = undefined;var _default = exports["default"] =






PayoutMethodDetailsZARBank;
