// transferzero-sdk/dist/Model/PoliticallyExposedPerson.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports["default"] = void 0;












var _ApiClient = _interopRequireDefault(require("../ApiClient"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };}function _typeof(o) {"@babel/helpers - typeof";return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) {return typeof o;} : function (o) {return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o;}, _typeof(o);}function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError("Cannot call a class as a function");}}function _defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ("value" in descriptor) descriptor.writable = true;Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);}}function _createClass(Constructor, protoProps, staticProps) {if (protoProps) _defineProperties(Constructor.prototype, protoProps);if (staticProps) _defineProperties(Constructor, staticProps);Object.defineProperty(Constructor, "prototype", { writable: false });return Constructor;}function _toPropertyKey(t) {var i = _toPrimitive(t, "string");return "symbol" == _typeof(i) ? i : i + "";}function _toPrimitive(t, r) {if ("object" != _typeof(t) || !t) return t;var e = t[Symbol.toPrimitive];if (void 0 !== e) {var i = e.call(t, r || "default");if ("object" != _typeof(i)) return i;throw new TypeError("@@toPrimitive must return a primitive value.");}return ("string" === r ? String : Number)(t);} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
 * The PoliticallyExposedPerson model module.
 * @module Model/PoliticallyExposedPerson
 * @version 1.35.1
 */var PoliticallyExposedPerson = /*#__PURE__*/function () {/**
   * Constructs a new <code>PoliticallyExposedPerson</code>.
   * @alias module:Model/PoliticallyExposedPerson
   */function PoliticallyExposedPerson() {_classCallCheck(this, PoliticallyExposedPerson);PoliticallyExposedPerson.initialize(this);} /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */return _createClass(PoliticallyExposedPerson, null, [{ key: "initialize", value: function initialize(obj) {
    }

    /**
     * Constructs a <code>PoliticallyExposedPerson</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:Model/PoliticallyExposedPerson} obj Optional instance to populate.
     * @return {module:Model/PoliticallyExposedPerson} The populated <code>PoliticallyExposedPerson</code> instance.
     */ }, { key: "constructFromObject", value:
    function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PoliticallyExposedPerson();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('position')) {
          obj['position'] = _ApiClient["default"].convertToType(data['position'], 'String');
        }
        if (data.hasOwnProperty('started_date')) {
          obj['started_date'] = _ApiClient["default"].convertToType(data['started_date'], 'Date');
        }
        if (data.hasOwnProperty('ended_date')) {
          obj['ended_date'] = _ApiClient["default"].convertToType(data['ended_date'], 'Date');
        }
        if (data.hasOwnProperty('sender_id')) {
          obj['sender_id'] = _ApiClient["default"].convertToType(data['sender_id'], 'String');
        }
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }
      }
      return obj;
    } }]);}();




/**
 * @member {Number} id
 */
PoliticallyExposedPerson.prototype['id'] = undefined;

/**
 * Full name of the politically exposed person 
 * @member {String} name
 */
PoliticallyExposedPerson.prototype['name'] = undefined;

/**
 * The office held by the politically exposed person 
 * @member {String} position
 */
PoliticallyExposedPerson.prototype['position'] = undefined;

/**
 * The date on which the person started holding the office 
 * @member {Date} started_date
 */
PoliticallyExposedPerson.prototype['started_date'] = undefined;

/**
 * The date on which the person ended holding the office 
 * @member {Date} ended_date
 */
PoliticallyExposedPerson.prototype['ended_date'] = undefined;

/**
 * The ID of the Sender to whom the person is attached to 
 * @member {String} sender_id
 */
PoliticallyExposedPerson.prototype['sender_id'] = undefined;

/**
 * Date and time the person was created
 * @member {Date} created_at
 */
PoliticallyExposedPerson.prototype['created_at'] = undefined;

/**
 * Date and time the person was updated
 * @member {Date} updated_at
 */
PoliticallyExposedPerson.prototype['updated_at'] = undefined;var _default = exports["default"] =






PoliticallyExposedPerson;
