// transferzero-sdk/dist/Model/RecipientStateReasonDetails.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports["default"] = void 0;












var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _StateReasonDetails = _interopRequireDefault(require("./StateReasonDetails"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };}function _typeof(o) {"@babel/helpers - typeof";return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) {return typeof o;} : function (o) {return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o;}, _typeof(o);}function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError("Cannot call a class as a function");}}function _defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ("value" in descriptor) descriptor.writable = true;Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);}}function _createClass(Constructor, protoProps, staticProps) {if (protoProps) _defineProperties(Constructor.prototype, protoProps);if (staticProps) _defineProperties(Constructor, staticProps);Object.defineProperty(Constructor, "prototype", { writable: false });return Constructor;}function _toPropertyKey(t) {var i = _toPrimitive(t, "string");return "symbol" == _typeof(i) ? i : i + "";}function _toPrimitive(t, r) {if ("object" != _typeof(t) || !t) return t;var e = t[Symbol.toPrimitive];if (void 0 !== e) {var i = e.call(t, r || "default");if ("object" != _typeof(i)) return i;throw new TypeError("@@toPrimitive must return a primitive value.");}return ("string" === r ? String : Number)(t);} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
 * The RecipientStateReasonDetails model module.
 * @module Model/RecipientStateReasonDetails
 * @version 1.35.1
 */var RecipientStateReasonDetails = /*#__PURE__*/function () {/**
   * Constructs a new <code>RecipientStateReasonDetails</code>.
   * @alias module:Model/RecipientStateReasonDetails
   * @implements module:Model/StateReasonDetails
   */function RecipientStateReasonDetails() {_classCallCheck(this, RecipientStateReasonDetails);_StateReasonDetails["default"].initialize(this);RecipientStateReasonDetails.initialize(this);} /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */return _createClass(RecipientStateReasonDetails, null, [{ key: "initialize", value: function initialize(obj) {
    }

    /**
     * Constructs a <code>RecipientStateReasonDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:Model/RecipientStateReasonDetails} obj Optional instance to populate.
     * @return {module:Model/RecipientStateReasonDetails} The populated <code>RecipientStateReasonDetails</code> instance.
     */ }, { key: "constructFromObject", value:
    function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RecipientStateReasonDetails();
        _StateReasonDetails["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('code')) {
          obj['code'] = _ApiClient["default"].convertToType(data['code'], 'String');
        }
        if (data.hasOwnProperty('category')) {
          obj['category'] = _ApiClient["default"].convertToType(data['category'], 'String');
        }
        if (data.hasOwnProperty('messages')) {
          obj['messages'] = _ApiClient["default"].convertToType(data['messages'], ['String']);
        }
        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }
      }
      return obj;
    } }]);}();




/**
 * Status code of transaction
 * @member {String} code
 */
RecipientStateReasonDetails.prototype['code'] = undefined;

/**
 * Main category of status code, it could be paid, unknown, pickupable, temporary_error, recipient_error, sender_error, sender_action_required
 * @member {String} category
 */
RecipientStateReasonDetails.prototype['category'] = undefined;

/**
 * Tiered messages
 * @member {Array.<String>} messages
 */
RecipientStateReasonDetails.prototype['messages'] = undefined;

/**
 * Public, human readable, detailed error message
 * @member {String} description
 */
RecipientStateReasonDetails.prototype['description'] = undefined;


// Implement StateReasonDetails interface:
/**
 * Status code of transaction
 * @member {String} code
 */
_StateReasonDetails["default"].prototype['code'] = undefined;
/**
 * Main category of status code, it could be paid, unknown, pickupable, temporary_error, recipient_error, sender_error, sender_action_required
 * @member {String} category
 */
_StateReasonDetails["default"].prototype['category'] = undefined;
/**
 * Tiered messages
 * @member {Array.<String>} messages
 */
_StateReasonDetails["default"].prototype['messages'] = undefined;
/**
 * Public, human readable, detailed error message
 * @member {String} description
 */
_StateReasonDetails["default"].prototype['description'] = undefined;var _default = exports["default"] =




RecipientStateReasonDetails;
