// transferzero-sdk/dist/Model/Recipient.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports["default"] = void 0;












var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _PayoutMethod = _interopRequireDefault(require("./PayoutMethod"));
var _RecipientState = _interopRequireDefault(require("./RecipientState"));
var _RecipientStateReasonDetails = _interopRequireDefault(require("./RecipientStateReasonDetails"));
var _TransactionState = _interopRequireDefault(require("./TransactionState"));
var _ValidationErrorDescription = _interopRequireDefault(require("./ValidationErrorDescription"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };}function _typeof(o) {"@babel/helpers - typeof";return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) {return typeof o;} : function (o) {return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o;}, _typeof(o);}function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError("Cannot call a class as a function");}}function _defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ("value" in descriptor) descriptor.writable = true;Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);}}function _createClass(Constructor, protoProps, staticProps) {if (protoProps) _defineProperties(Constructor.prototype, protoProps);if (staticProps) _defineProperties(Constructor, staticProps);Object.defineProperty(Constructor, "prototype", { writable: false });return Constructor;}function _toPropertyKey(t) {var i = _toPrimitive(t, "string");return "symbol" == _typeof(i) ? i : i + "";}function _toPrimitive(t, r) {if ("object" != _typeof(t) || !t) return t;var e = t[Symbol.toPrimitive];if (void 0 !== e) {var i = e.call(t, r || "default");if ("object" != _typeof(i)) return i;throw new TypeError("@@toPrimitive must return a primitive value.");}return ("string" === r ? String : Number)(t);} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
 * The Recipient model module.
 * @module Model/Recipient
 * @version 1.35.1
 */var Recipient = /*#__PURE__*/function () {/**
   * Constructs a new <code>Recipient</code>.
   * The recipient describes the amount, the currency and the destination where the money should be sent.
   * @alias module:Model/Recipient
   * @param requestedAmount {Number} the amount that should be paid to the recipient. This can be in any currency, usually either the input or the output currency. If the value here is not the output currency we will calculate how much the recipient is going to get using the exchange rates active when the transaction was created.
   * @param requestedCurrency {String} the currency of the amount in 3-character alpha ISO 4217 currency format
   * @param payoutMethod {module:Model/PayoutMethod} 
   */function Recipient(requestedAmount, requestedCurrency, payoutMethod) {_classCallCheck(this, Recipient);Recipient.initialize(this, requestedAmount, requestedCurrency, payoutMethod);} /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */return _createClass(Recipient, null, [{ key: "initialize", value: function initialize(obj, requestedAmount, requestedCurrency, payoutMethod) {
      obj['requested_amount'] = requestedAmount;
      obj['requested_currency'] = requestedCurrency;
      obj['payout_method'] = payoutMethod;
    }

    /**
     * Constructs a <code>Recipient</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:Model/Recipient} obj Optional instance to populate.
     * @return {module:Model/Recipient} The populated <code>Recipient</code> instance.
     */ }, { key: "constructFromObject", value:
    function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Recipient();

        if (data.hasOwnProperty('requested_amount')) {
          obj['requested_amount'] = _ApiClient["default"].convertToType(data['requested_amount'], 'Number');
        }
        if (data.hasOwnProperty('requested_currency')) {
          obj['requested_currency'] = _ApiClient["default"].convertToType(data['requested_currency'], 'String');
        }
        if (data.hasOwnProperty('payout_method')) {
          obj['payout_method'] = _PayoutMethod["default"].constructFromObject(data['payout_method']);
        }
        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], Object);
        }
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }
        if (data.hasOwnProperty('editable')) {
          obj['editable'] = _ApiClient["default"].convertToType(data['editable'], 'Boolean');
        }
        if (data.hasOwnProperty('retriable')) {
          obj['retriable'] = _ApiClient["default"].convertToType(data['retriable'], 'Boolean');
        }
        if (data.hasOwnProperty('input_usd_amount')) {
          obj['input_usd_amount'] = _ApiClient["default"].convertToType(data['input_usd_amount'], 'Number');
        }
        if (data.hasOwnProperty('may_cancel')) {
          obj['may_cancel'] = _ApiClient["default"].convertToType(data['may_cancel'], 'Boolean');
        }
        if (data.hasOwnProperty('state_reason')) {
          obj['state_reason'] = _ApiClient["default"].convertToType(data['state_reason'], 'String');
        }
        if (data.hasOwnProperty('state_reason_details')) {
          obj['state_reason_details'] = _RecipientStateReasonDetails["default"].constructFromObject(data['state_reason_details']);
        }
        if (data.hasOwnProperty('state')) {
          obj['state'] = _RecipientState["default"].constructFromObject(data['state']);
        }
        if (data.hasOwnProperty('transaction_id')) {
          obj['transaction_id'] = _ApiClient["default"].convertToType(data['transaction_id'], 'String');
        }
        if (data.hasOwnProperty('transaction_external_id')) {
          obj['transaction_external_id'] = _ApiClient["default"].convertToType(data['transaction_external_id'], 'String');
        }
        if (data.hasOwnProperty('transaction_state')) {
          obj['transaction_state'] = _TransactionState["default"].constructFromObject(data['transaction_state']);
        }
        if (data.hasOwnProperty('exchange_rate')) {
          obj['exchange_rate'] = _ApiClient["default"].convertToType(data['exchange_rate'], 'Number');
        }
        if (data.hasOwnProperty('fee_fractional')) {
          obj['fee_fractional'] = _ApiClient["default"].convertToType(data['fee_fractional'], 'Number');
        }
        if (data.hasOwnProperty('input_amount')) {
          obj['input_amount'] = _ApiClient["default"].convertToType(data['input_amount'], 'Number');
        }
        if (data.hasOwnProperty('input_currency')) {
          obj['input_currency'] = _ApiClient["default"].convertToType(data['input_currency'], 'String');
        }
        if (data.hasOwnProperty('output_amount')) {
          obj['output_amount'] = _ApiClient["default"].convertToType(data['output_amount'], 'Number');
        }
        if (data.hasOwnProperty('output_currency')) {
          obj['output_currency'] = _ApiClient["default"].convertToType(data['output_currency'], 'String');
        }
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('errors')) {
          obj['errors'] = _ApiClient["default"].convertToType(data['errors'], { 'String': [_ValidationErrorDescription["default"]] });
        }
      }
      return obj;
    } }]);}();




/**
 * the amount that should be paid to the recipient. This can be in any currency, usually either the input or the output currency. If the value here is not the output currency we will calculate how much the recipient is going to get using the exchange rates active when the transaction was created.
 * @member {Number} requested_amount
 */
Recipient.prototype['requested_amount'] = undefined;

/**
 * the currency of the amount in 3-character alpha ISO 4217 currency format
 * @member {String} requested_currency
 */
Recipient.prototype['requested_currency'] = undefined;

/**
 * @member {module:Model/PayoutMethod} payout_method
 */
Recipient.prototype['payout_method'] = undefined;

/**
 * Additional metadata that can be added to a recipient. These values will be returned on request
 * @member {Object} metadata
 */
Recipient.prototype['metadata'] = undefined;

/**
 * Date and time that the recipient was created.
 * @member {Date} created_at
 */
Recipient.prototype['created_at'] = undefined;

/**
 * Shows whether the recipient can be edited using the PATCH /v1/recipients/{id} endpoint or not
 * @member {Boolean} editable
 */
Recipient.prototype['editable'] = undefined;

/**
 * Shows whether the transaction made to the recipient can be retried or not
 * @member {Boolean} retriable
 */
Recipient.prototype['retriable'] = undefined;

/**
 * Shows how much this payment is worth in USD
 * @member {Number} input_usd_amount
 */
Recipient.prototype['input_usd_amount'] = undefined;

/**
 * If true it shows that the payment can be cancelled immediately using the DELETE /v1/recipients/{id} endpoint. If false you can still try to cancel it, however it will only gets cancelled once we have confirmation from our partner that the payment has failed.
 * @member {Boolean} may_cancel
 */
Recipient.prototype['may_cancel'] = undefined;

/**
 * In case the payment is unsuccessful it holds the error message associated with the last unsuccessful payout.
 * @member {String} state_reason
 */
Recipient.prototype['state_reason'] = undefined;

/**
 * @member {module:Model/RecipientStateReasonDetails} state_reason_details
 */
Recipient.prototype['state_reason_details'] = undefined;

/**
 * @member {module:Model/RecipientState} state
 */
Recipient.prototype['state'] = undefined;

/**
 * The ID of the transaction that is related to this recipient
 * @member {String} transaction_id
 */
Recipient.prototype['transaction_id'] = undefined;

/**
 * Optional ID that is supplied by partner linking it to the partner's own Transaction ID.
 * @member {String} transaction_external_id
 */
Recipient.prototype['transaction_external_id'] = undefined;

/**
 * @member {module:Model/TransactionState} transaction_state
 */
Recipient.prototype['transaction_state'] = undefined;

/**
 * The exchange rate used in this payment
 * @member {Number} exchange_rate
 */
Recipient.prototype['exchange_rate'] = undefined;

/**
 * The fee for this payment in fractional units (for example cents for USD transactions)
 * @member {Number} fee_fractional
 */
Recipient.prototype['fee_fractional'] = undefined;

/**
 * The amount that had to be paid in for this payment to proceed
 * @member {Number} input_amount
 */
Recipient.prototype['input_amount'] = undefined;

/**
 * The currency this payment was paid in
 * @member {String} input_currency
 */
Recipient.prototype['input_currency'] = undefined;

/**
 * The amount that will be paid to the recipient
 * @member {Number} output_amount
 */
Recipient.prototype['output_amount'] = undefined;

/**
 * The currency the payment will be delivered in
 * @member {String} output_currency
 */
Recipient.prototype['output_currency'] = undefined;

/**
 * @member {String} id
 */
Recipient.prototype['id'] = undefined;

/**
 * Type of recipient to create - either person or business (defaults to person) 
 * @member {module:Model/Recipient.TypeEnum} type
 */
Recipient.prototype['type'] = undefined;

/**
 * The fields that have some problems and don't pass validation
 * @member {Object.<String, Array.<module:Model/ValidationErrorDescription>>} errors
 */
Recipient.prototype['errors'] = undefined;





/**
 * Allowed values for the <code>type</code> property.
 * @enum {String}
 * @readonly
 */
Recipient['TypeEnum'] = {

  /**
   * value: "person"
   * @const
   */
  "person": "person",

  /**
   * value: "business"
   * @const
   */
  "business": "business"
};var _default = exports["default"] =



Recipient;
