// transferzero-sdk/dist/Model/TransactionWebhook.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports["default"] = void 0;












var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Transaction = _interopRequireDefault(require("./Transaction"));
var _TransactionWebhookAllOf = _interopRequireDefault(require("./TransactionWebhookAllOf"));
var _Webhook = _interopRequireDefault(require("./Webhook"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };}function _typeof(o) {"@babel/helpers - typeof";return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) {return typeof o;} : function (o) {return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o;}, _typeof(o);}function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError("Cannot call a class as a function");}}function _defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ("value" in descriptor) descriptor.writable = true;Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);}}function _createClass(Constructor, protoProps, staticProps) {if (protoProps) _defineProperties(Constructor.prototype, protoProps);if (staticProps) _defineProperties(Constructor, staticProps);Object.defineProperty(Constructor, "prototype", { writable: false });return Constructor;}function _toPropertyKey(t) {var i = _toPrimitive(t, "string");return "symbol" == _typeof(i) ? i : i + "";}function _toPrimitive(t, r) {if ("object" != _typeof(t) || !t) return t;var e = t[Symbol.toPrimitive];if (void 0 !== e) {var i = e.call(t, r || "default");if ("object" != _typeof(i)) return i;throw new TypeError("@@toPrimitive must return a primitive value.");}return ("string" === r ? String : Number)(t);} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
 * The TransactionWebhook model module.
 * @module Model/TransactionWebhook
 * @version 1.35.1
 */var TransactionWebhook = /*#__PURE__*/function () {/**
   * Constructs a new <code>TransactionWebhook</code>.
   * @alias module:Model/TransactionWebhook
   * @implements module:Model/Webhook
   * @implements module:Model/TransactionWebhookAllOf
   * @param webhook {String} The ID of the webhook that was used to send out this callback
   * @param event {String} The event that triggered this webhook
   * @param object {module:Model/Transaction} 
   */function TransactionWebhook(webhook, event, object) {_classCallCheck(this, TransactionWebhook);_Webhook["default"].initialize(this, webhook, event, object);_TransactionWebhookAllOf["default"].initialize(this);TransactionWebhook.initialize(this, webhook, event, object);} /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */return _createClass(TransactionWebhook, null, [{ key: "initialize", value: function initialize(obj, webhook, event, object) {
      obj['webhook'] = webhook;
      obj['event'] = event;
      obj['object'] = object;
    }

    /**
     * Constructs a <code>TransactionWebhook</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:Model/TransactionWebhook} obj Optional instance to populate.
     * @return {module:Model/TransactionWebhook} The populated <code>TransactionWebhook</code> instance.
     */ }, { key: "constructFromObject", value:
    function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TransactionWebhook();
        _Webhook["default"].constructFromObject(data, obj);
        _TransactionWebhookAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('webhook')) {
          obj['webhook'] = _ApiClient["default"].convertToType(data['webhook'], 'String');
        }
        if (data.hasOwnProperty('event')) {
          obj['event'] = _ApiClient["default"].convertToType(data['event'], 'String');
        }
        if (data.hasOwnProperty('object')) {
          obj['object'] = _Transaction["default"].constructFromObject(data['object']);
        }
      }
      return obj;
    } }]);}();




/**
 * The ID of the webhook that was used to send out this callback
 * @member {String} webhook
 */
TransactionWebhook.prototype['webhook'] = undefined;

/**
 * The event that triggered this webhook
 * @member {String} event
 */
TransactionWebhook.prototype['event'] = undefined;

/**
 * @member {module:Model/Transaction} object
 */
TransactionWebhook.prototype['object'] = undefined;


// Implement Webhook interface:
/**
 * The ID of the webhook that was used to send out this callback
 * @member {String} webhook
 */
_Webhook["default"].prototype['webhook'] = undefined;
/**
 * The event that triggered this webhook
 * @member {String} event
 */
_Webhook["default"].prototype['event'] = undefined;
/**
 * The returned object
 * @member {Object} object
 */
_Webhook["default"].prototype['object'] = undefined;
// Implement TransactionWebhookAllOf interface:
/**
 * @member {module:Model/Transaction} object
 */
_TransactionWebhookAllOf["default"].prototype['object'] = undefined;var _default = exports["default"] =




TransactionWebhook;
