// transferzero-sdk/dist/Model/WebhookLogMetadata.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports["default"] = void 0;












var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _WebhookDefinition = _interopRequireDefault(require("./WebhookDefinition"));
var _WebhookLogMetadataRequest = _interopRequireDefault(require("./WebhookLogMetadataRequest"));
var _WebhookLogMetadataResponse = _interopRequireDefault(require("./WebhookLogMetadataResponse"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };}function _typeof(o) {"@babel/helpers - typeof";return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) {return typeof o;} : function (o) {return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o;}, _typeof(o);}function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError("Cannot call a class as a function");}}function _defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ("value" in descriptor) descriptor.writable = true;Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);}}function _createClass(Constructor, protoProps, staticProps) {if (protoProps) _defineProperties(Constructor.prototype, protoProps);if (staticProps) _defineProperties(Constructor, staticProps);Object.defineProperty(Constructor, "prototype", { writable: false });return Constructor;}function _toPropertyKey(t) {var i = _toPrimitive(t, "string");return "symbol" == _typeof(i) ? i : i + "";}function _toPrimitive(t, r) {if ("object" != _typeof(t) || !t) return t;var e = t[Symbol.toPrimitive];if (void 0 !== e) {var i = e.call(t, r || "default");if ("object" != _typeof(i)) return i;throw new TypeError("@@toPrimitive must return a primitive value.");}return ("string" === r ? String : Number)(t);} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
 * The WebhookLogMetadata model module.
 * @module Model/WebhookLogMetadata
 * @version 1.35.1
 */var WebhookLogMetadata = /*#__PURE__*/function () {/**
   * Constructs a new <code>WebhookLogMetadata</code>.
   * @alias module:Model/WebhookLogMetadata
   */function WebhookLogMetadata() {_classCallCheck(this, WebhookLogMetadata);WebhookLogMetadata.initialize(this);} /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */return _createClass(WebhookLogMetadata, null, [{ key: "initialize", value: function initialize(obj) {
    }

    /**
     * Constructs a <code>WebhookLogMetadata</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:Model/WebhookLogMetadata} obj Optional instance to populate.
     * @return {module:Model/WebhookLogMetadata} The populated <code>WebhookLogMetadata</code> instance.
     */ }, { key: "constructFromObject", value:
    function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WebhookLogMetadata();

        if (data.hasOwnProperty('params')) {
          obj['params'] = _WebhookDefinition["default"].constructFromObject(data['params']);
        }
        if (data.hasOwnProperty('event')) {
          obj['event'] = _ApiClient["default"].convertToType(data['event'], 'String');
        }
        if (data.hasOwnProperty('instance_id')) {
          obj['instance_id'] = _ApiClient["default"].convertToType(data['instance_id'], 'String');
        }
        if (data.hasOwnProperty('instance_type')) {
          obj['instance_type'] = _ApiClient["default"].convertToType(data['instance_type'], 'String');
        }
        if (data.hasOwnProperty('attempt_id')) {
          obj['attempt_id'] = _ApiClient["default"].convertToType(data['attempt_id'], 'String');
        }
        if (data.hasOwnProperty('tries')) {
          obj['tries'] = _ApiClient["default"].convertToType(data['tries'], 'Number');
        }
        if (data.hasOwnProperty('request')) {
          obj['request'] = _WebhookLogMetadataRequest["default"].constructFromObject(data['request']);
        }
        if (data.hasOwnProperty('response')) {
          obj['response'] = _WebhookLogMetadataResponse["default"].constructFromObject(data['response']);
        }
        if (data.hasOwnProperty('app_id')) {
          obj['app_id'] = _ApiClient["default"].convertToType(data['app_id'], 'String');
        }
        if (data.hasOwnProperty('api_key_id')) {
          obj['api_key_id'] = _ApiClient["default"].convertToType(data['api_key_id'], 'String');
        }
      }
      return obj;
    } }]);}();




/**
 * @member {module:Model/WebhookDefinition} params
 */
WebhookLogMetadata.prototype['params'] = undefined;

/**
 * Specific event that triggered the webhook
 * @member {String} event
 */
WebhookLogMetadata.prototype['event'] = undefined;

/**
 * Instance ID of the webhook event
 * @member {String} instance_id
 */
WebhookLogMetadata.prototype['instance_id'] = undefined;

/**
 * Type of instance event
 * @member {String} instance_type
 */
WebhookLogMetadata.prototype['instance_type'] = undefined;

/**
 * Attempt ID
 * @member {String} attempt_id
 */
WebhookLogMetadata.prototype['attempt_id'] = undefined;

/**
 * Number of tries at the point webhook was triggered
 * @member {Number} tries
 */
WebhookLogMetadata.prototype['tries'] = undefined;

/**
 * @member {module:Model/WebhookLogMetadataRequest} request
 */
WebhookLogMetadata.prototype['request'] = undefined;

/**
 * @member {module:Model/WebhookLogMetadataResponse} response
 */
WebhookLogMetadata.prototype['response'] = undefined;

/**
 * ID of app on which webhook event was triggered
 * @member {String} app_id
 */
WebhookLogMetadata.prototype['app_id'] = undefined;

/**
 * ID of API key in use at point of webhook event
 * @member {String} api_key_id
 */
WebhookLogMetadata.prototype['api_key_id'] = undefined;var _default = exports["default"] =






WebhookLogMetadata;
