// transferzero-sdk/dist/Model/WebhookDefinition.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports["default"] = void 0;












var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ValidationErrorDescription = _interopRequireDefault(require("./ValidationErrorDescription"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };}function _typeof(o) {"@babel/helpers - typeof";return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) {return typeof o;} : function (o) {return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o;}, _typeof(o);}function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError("Cannot call a class as a function");}}function _defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ("value" in descriptor) descriptor.writable = true;Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);}}function _createClass(Constructor, protoProps, staticProps) {if (protoProps) _defineProperties(Constructor.prototype, protoProps);if (staticProps) _defineProperties(Constructor, staticProps);Object.defineProperty(Constructor, "prototype", { writable: false });return Constructor;}function _toPropertyKey(t) {var i = _toPrimitive(t, "string");return "symbol" == _typeof(i) ? i : i + "";}function _toPrimitive(t, r) {if ("object" != _typeof(t) || !t) return t;var e = t[Symbol.toPrimitive];if (void 0 !== e) {var i = e.call(t, r || "default");if ("object" != _typeof(i)) return i;throw new TypeError("@@toPrimitive must return a primitive value.");}return ("string" === r ? String : Number)(t);} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
 * The WebhookDefinition model module.
 * @module Model/WebhookDefinition
 * @version 1.35.1
 */var WebhookDefinition = /*#__PURE__*/function () {/**
   * Constructs a new <code>WebhookDefinition</code>.
   * Object that holds where to send webhooks, and what events should it be triggered with
   * @alias module:Model/WebhookDefinition
   * @param url {String} The URL to receive the webhook
   * @param events {Array.<String>} A list of events to subscribe to
   */function WebhookDefinition(url, events) {_classCallCheck(this, WebhookDefinition);WebhookDefinition.initialize(this, url, events);} /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */return _createClass(WebhookDefinition, null, [{ key: "initialize", value: function initialize(obj, url, events) {
      obj['url'] = url;
      obj['events'] = events;
    }

    /**
     * Constructs a <code>WebhookDefinition</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:Model/WebhookDefinition} obj Optional instance to populate.
     * @return {module:Model/WebhookDefinition} The populated <code>WebhookDefinition</code> instance.
     */ }, { key: "constructFromObject", value:
    function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WebhookDefinition();

        if (data.hasOwnProperty('url')) {
          obj['url'] = _ApiClient["default"].convertToType(data['url'], 'String');
        }
        if (data.hasOwnProperty('events')) {
          obj['events'] = _ApiClient["default"].convertToType(data['events'], ['String']);
        }
        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], Object);
        }
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('errors')) {
          obj['errors'] = _ApiClient["default"].convertToType(data['errors'], { 'String': [_ValidationErrorDescription["default"]] });
        }
      }
      return obj;
    } }]);}();




/**
 * The URL to receive the webhook
 * @member {String} url
 */
WebhookDefinition.prototype['url'] = undefined;

/**
 * A list of events to subscribe to
 * @member {Array.<String>} events
 */
WebhookDefinition.prototype['events'] = undefined;

/**
 * @member {Object} metadata
 */
WebhookDefinition.prototype['metadata'] = undefined;

/**
 * The ID of the webhook
 * @member {String} id
 */
WebhookDefinition.prototype['id'] = undefined;

/**
 * The fields that have some problems and don't pass validation
 * @member {Object.<String, Array.<module:Model/ValidationErrorDescription>>} errors
 */
WebhookDefinition.prototype['errors'] = undefined;var _default = exports["default"] =






WebhookDefinition;
