// transferzero-sdk/dist/Model/WebhookLogMetadataRequest.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports["default"] = void 0;












var _ApiClient = _interopRequireDefault(require("../ApiClient"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };}function _typeof(o) {"@babel/helpers - typeof";return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) {return typeof o;} : function (o) {return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o;}, _typeof(o);}function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError("Cannot call a class as a function");}}function _defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ("value" in descriptor) descriptor.writable = true;Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);}}function _createClass(Constructor, protoProps, staticProps) {if (protoProps) _defineProperties(Constructor.prototype, protoProps);if (staticProps) _defineProperties(Constructor, staticProps);Object.defineProperty(Constructor, "prototype", { writable: false });return Constructor;}function _toPropertyKey(t) {var i = _toPrimitive(t, "string");return "symbol" == _typeof(i) ? i : i + "";}function _toPrimitive(t, r) {if ("object" != _typeof(t) || !t) return t;var e = t[Symbol.toPrimitive];if (void 0 !== e) {var i = e.call(t, r || "default");if ("object" != _typeof(i)) return i;throw new TypeError("@@toPrimitive must return a primitive value.");}return ("string" === r ? String : Number)(t);} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
 * The WebhookLogMetadataRequest model module.
 * @module Model/WebhookLogMetadataRequest
 * @version 1.35.1
 */var WebhookLogMetadataRequest = /*#__PURE__*/function () {/**
   * Constructs a new <code>WebhookLogMetadataRequest</code>.
   * Metadata related to the initial request that triggered the webhook
   * @alias module:Model/WebhookLogMetadataRequest
   */function WebhookLogMetadataRequest() {_classCallCheck(this, WebhookLogMetadataRequest);WebhookLogMetadataRequest.initialize(this);} /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */return _createClass(WebhookLogMetadataRequest, null, [{ key: "initialize", value: function initialize(obj) {
    }

    /**
     * Constructs a <code>WebhookLogMetadataRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:Model/WebhookLogMetadataRequest} obj Optional instance to populate.
     * @return {module:Model/WebhookLogMetadataRequest} The populated <code>WebhookLogMetadataRequest</code> instance.
     */ }, { key: "constructFromObject", value:
    function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WebhookLogMetadataRequest();

        if (data.hasOwnProperty('headers')) {
          obj['headers'] = _ApiClient["default"].convertToType(data['headers'], { 'String': 'String' });
        }
        if (data.hasOwnProperty('body')) {
          obj['body'] = _ApiClient["default"].convertToType(data['body'], 'String');
        }
        if (data.hasOwnProperty('url')) {
          obj['url'] = _ApiClient["default"].convertToType(data['url'], 'String');
        }
      }
      return obj;
    } }]);}();




/**
 * @member {Object.<String, String>} headers
 */
WebhookLogMetadataRequest.prototype['headers'] = undefined;

/**
 * The body that was sent in the webhook
 * @member {String} body
 */
WebhookLogMetadataRequest.prototype['body'] = undefined;

/**
 * Webhook URL
 * @member {String} url
 */
WebhookLogMetadataRequest.prototype['url'] = undefined;var _default = exports["default"] =






WebhookLogMetadataRequest;
