// transferzero-sdk/dist/Model/WebhookLog.js
"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports["default"] = void 0;












var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _WebhookLogMetadata = _interopRequireDefault(require("./WebhookLogMetadata"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { "default": obj };}function _typeof(o) {"@babel/helpers - typeof";return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) {return typeof o;} : function (o) {return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o;}, _typeof(o);}function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError("Cannot call a class as a function");}}function _defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ("value" in descriptor) descriptor.writable = true;Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);}}function _createClass(Constructor, protoProps, staticProps) {if (protoProps) _defineProperties(Constructor.prototype, protoProps);if (staticProps) _defineProperties(Constructor, staticProps);Object.defineProperty(Constructor, "prototype", { writable: false });return Constructor;}function _toPropertyKey(t) {var i = _toPrimitive(t, "string");return "symbol" == _typeof(i) ? i : i + "";}function _toPrimitive(t, r) {if ("object" != _typeof(t) || !t) return t;var e = t[Symbol.toPrimitive];if (void 0 !== e) {var i = e.call(t, r || "default");if ("object" != _typeof(i)) return i;throw new TypeError("@@toPrimitive must return a primitive value.");}return ("string" === r ? String : Number)(t);} /**
 * TransferZero API
 * Reference documentation for the TransferZero API V1
 *
 * The version of the OpenAPI document: 1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */ /**
 * The WebhookLog model module.
 * @module Model/WebhookLog
 * @version 1.35.1
 */var WebhookLog = /*#__PURE__*/function () {/**
   * Constructs a new <code>WebhookLog</code>.
   * @alias module:Model/WebhookLog
   */function WebhookLog() {_classCallCheck(this, WebhookLog);WebhookLog.initialize(this);} /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */return _createClass(WebhookLog, null, [{ key: "initialize", value: function initialize(obj) {
    }

    /**
     * Constructs a <code>WebhookLog</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:Model/WebhookLog} obj Optional instance to populate.
     * @return {module:Model/WebhookLog} The populated <code>WebhookLog</code> instance.
     */ }, { key: "constructFromObject", value:
    function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WebhookLog();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('message')) {
          obj['message'] = _ApiClient["default"].convertToType(data['message'], 'String');
        }
        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _WebhookLogMetadata["default"].constructFromObject(data['metadata']);
        }
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'String');
        }
        if (data.hasOwnProperty('parent_id')) {
          obj['parent_id'] = _ApiClient["default"].convertToType(data['parent_id'], 'String');
        }
        if (data.hasOwnProperty('parent_type')) {
          obj['parent_type'] = _ApiClient["default"].convertToType(data['parent_type'], 'String');
        }
      }
      return obj;
    } }]);}();




/**
 * ID of the webhook log
 * @member {String} id
 */
WebhookLog.prototype['id'] = undefined;

/**
 * Message detailing webhook event
 * @member {String} message
 */
WebhookLog.prototype['message'] = undefined;

/**
 * @member {module:Model/WebhookLogMetadata} metadata
 */
WebhookLog.prototype['metadata'] = undefined;

/**
 * Date and time of response
 * @member {String} created_at
 */
WebhookLog.prototype['created_at'] = undefined;

/**
 * @member {String} parent_id
 */
WebhookLog.prototype['parent_id'] = undefined;

/**
 * @member {String} parent_type
 */
WebhookLog.prototype['parent_type'] = undefined;var _default = exports["default"] =






WebhookLog;
